---
title: "gamma_secretase_DIAN"
author: "Stephanie Schultz"
date: "04/06/2024"
output:
  word_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Load libraries
```{r}
#install.packages('vctrs')
library("ggsci");library(ggplot2); library(outliers); library(factoextra); library(dplyr); library(cowplot); library(lme4);library(lmerTest)


```


#read in cell-based datasheet and generate gsc
```{r}

#read in source df for cell-based measures
bc_data <- read.csv("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/revision_2/data_and_code//data_Liu_schultz_combined.csv", header=TRUE)

#create a composite based on petite et al of short (37,38,40) to long (42,43) abeta species
bc_data$gsec_composite <- (bc_data$X37_pg_ml + bc_data$X38_pg_ml +bc_data$X40_pg_ml)/ (bc_data$X42_pg_ml + bc_data$X43_pg_ml)



#percent relative to wildtype; WT value is 11.5862150 for lei et al sample, 11.6644414 for dian-obs sample
bc_data$g_gsec_composite_rel_wt <- ifelse (bc_data$paper == "2", bc_data$gsec_composite/11.664441*100,bc_data$gsec_composite/11.5862150 *100)

```

# calculate AAO GSC and plot Figure 1b, median (SD) for percent rel WT individual species
```{r}

#filter out lei et al WT
bc_data_unique_var_inclwt <- bc_data %>%  filter (variant_orig != "WT-PSEN1_mean")

# make sure abeta species are read as numeric
bc_data_unique_var_inclwt$X37_pg_ml <- as.numeric (bc_data_unique_var_inclwt$X37_pg_ml)
bc_data_unique_var_inclwt$X38_pg_ml <- as.numeric (bc_data_unique_var_inclwt$X38_pg_ml)
bc_data_unique_var_inclwt$X40_pg_ml <- as.numeric (bc_data_unique_var_inclwt$X40_pg_ml)
bc_data_unique_var_inclwt$X42_pg_ml <- as.numeric (bc_data_unique_var_inclwt$X42_pg_ml)
bc_data_unique_var_inclwt$X43_pg_ml <- as.numeric (bc_data_unique_var_inclwt$X43_pg_ml)

# calculate total for each variant and WT 
bc_data_unique_var_inclwt$sum <- bc_data_unique_var_inclwt$X37_pg_ml +bc_data_unique_var_inclwt$X38_pg_ml +bc_data_unique_var_inclwt$X40_pg_ml+bc_data_unique_var_inclwt$X42_pg_ml+ bc_data_unique_var_inclwt$X43_pg_ml

# calculate abeta peptide relative to total abeta produced
bc_data_unique_var_inclwt$abx_37_mean_perc_sum <- bc_data_unique_var_inclwt$X37_pg_ml/ bc_data_unique_var_inclwt$sum*100
bc_data_unique_var_inclwt$abx_38_mean_perc_sum <- bc_data_unique_var_inclwt$X38_pg_ml/ bc_data_unique_var_inclwt$sum*100
bc_data_unique_var_inclwt$abx_40_mean_perc_sum <- bc_data_unique_var_inclwt$X40_pg_ml/ bc_data_unique_var_inclwt$sum*100
bc_data_unique_var_inclwt$abx_42_mean_perc_sum <- bc_data_unique_var_inclwt$X42_pg_ml/ bc_data_unique_var_inclwt$sum*100
bc_data_unique_var_inclwt$abx_43_mean_perc_sum <- bc_data_unique_var_inclwt$X43_pg_ml/ bc_data_unique_var_inclwt$sum*100


# DIAN WT x37= 2235.050
# DIAN WT x38=  2693.27
# DIAN WT x40= 42161.38
# DIAN WT x42= 3957.98
# DIAN WT x43= 79.05000

# Liu WT x37 = 1446.073
# Liu WT x38 = 1456.00
# Liu WT x40 = 19051.68
# Liu WT x42 = 1868.10
# Liu WT x43 = 26.71667
# Liu WT 42/40= 0.09805434
# Liu WT gsec_composite= 11.5862150

bc_data$a_X37_pg_ml_rel_wt <- ifelse (bc_data$paper == "2", bc_data$X37_pg_ml/2235.050*100,bc_data$X37_pg_ml/1446.073 *100)
bc_data$b_X38_pg_ml_rel_wt <- ifelse (bc_data$paper == "2", bc_data$X38_pg_ml/2693.27*100,bc_data$X38_pg_ml/1456.00 *100)
bc_data$c_X40_pg_ml_rel_wt <- ifelse (bc_data$paper == "2", bc_data$X40_pg_ml/42161.38*100,bc_data$X40_pg_ml/19051.68 *100)
bc_data$d_X42_pg_ml_rel_wt <- ifelse (bc_data$paper == "2", bc_data$X42_pg_ml/3957.98*100,bc_data$X42_pg_ml/1868.10 *100)
bc_data$e_X43_pg_ml_rel_wt <- ifelse (bc_data$paper == "2", bc_data$X43_pg_ml/79.05000*100,bc_data$X43_pg_ml/26.71667 *100)

  
  bc_data_unique_var_rmwt <- bc_data %>%  filter (variant_rename != "WT")

 median(bc_data_unique_var_rmwt$a_X37_pg_ml_rel_wt)
  sd(bc_data_unique_var_rmwt$a_X37_pg_ml_rel_wt)
  IQR(bc_data_unique_var_rmwt$a_X37_pg_ml_rel_wt)
   quantile(bc_data_unique_var_rmwt$a_X37_pg_ml_rel_wt, c(0:4/4))

    
 median(bc_data_unique_var_rmwt$b_X38_pg_ml_rel_wt)
  sd(bc_data_unique_var_rmwt$b_X38_pg_ml_rel_wt)
    IQR(bc_data_unique_var_rmwt$b_X38_pg_ml_rel_wt)
quantile(bc_data_unique_var_rmwt$b_X38_pg_ml_rel_wt, c(0:4/4))


 median(bc_data_unique_var_rmwt$c_X40_pg_ml_rel_wt)
  sd(bc_data_unique_var_rmwt$c_X40_pg_ml_rel_wt)
      IQR(bc_data_unique_var_rmwt$c_X40_pg_ml_rel_wt)
quantile(bc_data_unique_var_rmwt$c_X40_pg_ml_rel_wt, c(0:4/4))

  
#Grouping together in vitro Aβ production for all 161 variants (Figure 1B and appendix pg 11-17), median (sd) levels relative to wildtype PSEN1 for Aβ37, 38, and 40 were 46.49% (44.4), 51.49% (51.4), and 85.13% (64.2). 
  

 median(bc_data_unique_var_rmwt$d_X42_pg_ml_rel_wt)
  sd(bc_data_unique_var_rmwt$d_X42_pg_ml_rel_wt)
        IQR(bc_data_unique_var_rmwt$d_X42_pg_ml_rel_wt)
quantile(bc_data_unique_var_rmwt$d_X42_pg_ml_rel_wt, c(0:4/4))

 median(bc_data_unique_var_rmwt$e_X43_pg_ml_rel_wt)
  sd(bc_data_unique_var_rmwt$e_X43_pg_ml_rel_wt)
        IQR(bc_data_unique_var_rmwt$e_X43_pg_ml_rel_wt)
quantile(bc_data_unique_var_rmwt$e_X43_pg_ml_rel_wt, c(0:4/4))

  
#Additionally, relative levels of Aβ42 and 43 across variants were higher compared to that observed #with wildtype PSEN1 (median (sd) = 182.41% (221.2) and 158.96% (1174.4), respectively; Figure 1B).

cor.test(bc_data_unique_var_inclwt$AAO_MutAAO, bc_data_unique_var_inclwt$g_gsec_composite_rel_wt)


#Next, we assessed relationships between the relative concentrations of Aβ species, using the GSC, produced by each variant and AAO. Lower GSC values (Pearson’s correlation [degrees of freedom], p-value: r[159] = 0.58, p < 0.0001) were associated with earlier AAO (Figure 1C). 


# calculate a gsc-dependent AAO

#restrict to just Lei et al variants (non-dian)
bc_data_unique_var_lei <- bc_data_unique_var_inclwt %>%  filter (paper == "1")

# regression between gsc and AAO
summary(lm(AAO_parental ~ g_gsec_composite_rel_wt, bc_data_unique_var_lei))

#Call:
#lm(formula = AAO_parental ~ g_gsec_composite_rel_wt, data = bc_data_unique_var_lei)

#Residuals:
#     Min       1Q   Median       3Q      Max 
#-25.2939  -4.3982  -0.4263   4.9959  16.5497 

#Coefficients:
#                        Estimate Std. Error t value Pr(>|t|)    
#(Intercept)             28.13757    1.73355  16.231  < 2e-16 ***
#g_gsec_composite_rel_wt  0.26896    0.02971   9.052 9.43e-15 ***
#---
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

#Residual standard error: 7.501 on 103 degrees of freedom
#Multiple R-squared:  0.4431,	Adjusted R-squared:  0.4377 
#F-statistic: 81.95 on 1 and 103 DF,  p-value: 9.431e-15



# take intercept and slope and use it to predict AAO in dian-obs variants
bc_data_unique_var_inclwt$AAO_GSC <- 28.13757 + 0.26896*bc_data_unique_var_inclwt$g_gsec_composite_rel_wt

bc_data_unique_var_inclwt_dian <- bc_data_unique_var_inclwt %>% filter (paper =="2")

cor.test(bc_data_unique_var_inclwt_dian$AAO_MutAAO, bc_data_unique_var_inclwt_dian$AAO_GSC)


#bc_mEYO_merge$AAO_GSC <- 28.13757 + 0.26896*bc_mEYO_merge$g_gsec_composite_rel_wt


#We further examined the potential utility of using the GSC to predict an approximate AAO for novel pathogenic variants and/or when a clinical history is not available. There was high correlation between the history-derived AAO and the GSC-derived AAO (AAOgsc) in the subset of variants represented in DIAN (r [54] = 0.59, p < 0.0001; appendix pg 11-15).


bc_data_unique_var_rmwt_dian <- bc_data_unique_var_rmwt %>% filter (paper == "2")

# calculate a gsc-dependent AAO

#restrict to just DIAN variants
bc_data_unique_var_DIAN <- bc_data_unique_var_rmwt_dian %>%  filter (paper == "2")
# 
# # regression between gsc and AAO
# summary(lm(AAO_parental ~ g_gsec_composite_rel_wt, bc_data_unique_var_rmwt_dian))
# 
# Call:
# lm(formula = AAO_parental ~ g_gsec_composite_rel_wt, data = bc_data_unique_var_rmwt_dian)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -17.0395  -4.2682  -0.6297   5.2935  16.9761 
# 
# Coefficients:
#                         Estimate Std. Error t value Pr(>|t|)    
# (Intercept)             34.61113    2.30141  15.039  < 2e-16 ***
# g_gsec_composite_rel_wt  0.27077    0.05533   4.894 9.32e-06 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 7.229 on 54 degrees of freedom
# Multiple R-squared:  0.3073,	Adjusted R-squared:  0.2944 
# F-statistic: 23.95 on 1 and 54 DF,  p-value: 9.321e-06



# take intercept and slope and use it to predict AAO in dian-obs variants
bc_data_unique_var_inclwt_dian$AAO_GSC_dian  <- 34.61113 + 0.27077*bc_data_unique_var_inclwt_dian$g_gsec_composite_rel_wt


#bc_mEYO_merge$AAO_GSC_dian  <- 34.61113 + 0.27077*bc_mEYO_merge$g_gsec_composite_rel_wt

cor.test(bc_data_unique_var_inclwt_dian$AAO_MutAAO, bc_data_unique_var_inclwt_dian$AAO_GSC_dian)

#We further examined the potential utility of using the GSC to predict an approximate AAO for novel pathogenic variants and/or when a clinical history is not available. There was high correlation between the history-derived AAO and the GSC-derived AAO (AAOgsc) in the subset of variants represented in DIAN (r [54] = 0.59, p < 0.0001; appendix pg 11-15).


bc_data_unique_var_rmwt_dian$x42_40_rel_wt <- (bc_data_unique_var_rmwt_dian$X42_pg_ml/bc_data_unique_var_rmwt_dian$X40_pg_ml) / 0.09387691 * 100


bc_data_unique_var_rmwt_dian$log_42_40_rel_wt <- log10 (bc_data_unique_var_rmwt_dian$X42_pg_ml/bc_data_unique_var_rmwt_dian$X40_pg_ml) / -1.027441 * 100

bc_data_unique_var_rmwt_dian$x37_42_rel_wt <- (bc_data_unique_var_rmwt_dian$X37_pg_ml/bc_data_unique_var_rmwt_dian$X42_pg_ml) / 0.5646946 * 100



bc_data_unique_var_rmwt_dian$log_37_42_rel_wt <- log10(bc_data_unique_var_rmwt_dian$X37_pg_ml/bc_data_unique_var_rmwt_dian$X42_pg_ml) / -0.2481864 * 100


summary(lm(formula = AAO_parental ~ log_42_40_rel_wt+ log_37_42_rel_wt , data = bc_data_unique_var_rmwt_dian))




bc_data_unique_var_rmwt_dian$AAO_log4240_log3742_dian <- 40.30422 + 0.16797*bc_data_unique_var_rmwt_dian$log_42_40_rel_wt -0.01326*bc_data_unique_var_rmwt_dian$log_37_42_rel_wt



# 
# 
# bc_mEYO_merge$x42_40_rel_wt <- (bc_mEYO_merge$X42_pg_ml/bc_mEYO_merge$X40_pg_ml) / 0.09387691 * 100
# 
# 
# bc_mEYO_merge$log_42_40_rel_wt <- log10 (bc_mEYO_merge$X42_pg_ml/bc_mEYO_merge$X40_pg_ml) / -1.027441 * 100
# 
# bc_mEYO_merge$x37_42_rel_wt <- (bc_mEYO_merge$X37_pg_ml/bc_mEYO_merge$X42_pg_ml) / 0.5646946 * 100
# 
# 
# 
# bc_mEYO_merge$log_37_42_rel_wt <- log10(bc_mEYO_merge$X37_pg_ml/bc_mEYO_merge$X42_pg_ml) / -0.2481864 * 100
# 
# 
# 
# bc_mEYO_merge$AAO_log4240_log3742_dian <- 40.30422 + 0.16797*bc_mEYO_merge$log_42_40_rel_wt -0.01326*bc_mEYO_merge$log_37_42_rel_wt
# 



cor.test(bc_data_unique_var_rmwt_dian$AAO_parental, bc_data_unique_var_rmwt_dian$AAO_log4240_log3742_dian)

bc_data_unique_var_rmwt_dian$AAO_GSC_AAO_MutAAO_diff <-bc_data_unique_var_rmwt_dian$AAO_log4240_log3742_dian - bc_data_unique_var_rmwt_dian$AAO_parental



# extract codon number
require(stringr)
#to get the digits from variant_rename
regexp <- "[[:digit:]]+"
bc_data_unique_var_inclwt$codon_number <- as.numeric(str_extract(bc_data_unique_var_inclwt$variant_rename, regexp))

# for exon 9 deletion, place it after codon 290
bc_data_unique_var_inclwt$codon_number2 <- ifelse(bc_data_unique_var_inclwt$codon_number > 9, bc_data_unique_var_inclwt$codon_number, 291) 

# code WT as 0 so it is first
bc_data_unique_var_inclwt <- bc_data_unique_var_inclwt %>% mutate(codon_number2 = ifelse(is.na(bc_data_unique_var_inclwt$codon_number2), "0", bc_data_unique_var_inclwt$codon_number2)) 

bc_data_unique_var_inclwt <- bc_data_unique_var_inclwt %>% 
  arrange (as.numeric(codon_number2)) %>%
  mutate(codon_order = order(as.numeric(codon_number2)))

# I think this creating a new oder for variant name based on codon number
bc_data_unique_var_inclwt$variant_rename <- factor(bc_data_unique_var_inclwt$variant_rename, levels = bc_data_unique_var_inclwt$variant_rename[order(bc_data_unique_var_inclwt$codon_order)])


# convert wide to long df
library(tidyr)
bc_data_unique_var_inclwt_long <- bc_data_unique_var_inclwt %>% 
  pivot_longer(
    cols = `abx_37_mean_perc_sum`:`abx_43_mean_perc_sum`, 
    names_to = "variable",
    values_to = "value")

# break df into two so that you can plot in two separate bar charts
bc_data_unique_var_inclwt_long_1st_half <- bc_data_unique_var_inclwt_long %>% filter (codon_order <=80)
bc_data_unique_var_inclwt_long_2nd_half <- bc_data_unique_var_inclwt_long %>% filter (codon_order >80)


# bar chart 1
stacked_barchart_1 <- ggplot(bc_data_unique_var_inclwt_long_1st_half, aes(x =variant_rename, y = value, fill = variable), show.legend=F) + geom_col() +  labs(y ="% total", x ="",fill = (expression(bold(paste("A", beta,"peptide"))))) +theme(axis.text.x = element_text(angle = 90, vjust = 0, hjust=1, size = 20, face="bold")) + theme(axis.text.y = element_text(size = 25, face ="bold"))+ theme(axis.title.y = element_text(size = 25, face ="bold"))  +scale_fill_manual(values=c( '#02778a',"#82a2bf", '#d4d2e5','#b87caf','#a30348'), labels = c((expression(bold(paste("A", beta,"37")))), (expression(bold(paste("A", beta,"38")))), (expression(bold(paste("A", beta,"40")))), (expression(bold(paste("A", beta,"42")))), (expression(bold(paste("A", beta,"43"))))))+
  theme(legend.text = element_text(size=40, family="Helvetica", face="bold"))+  theme(legend.position = c("")) 

# bar chart 2
stacked_barchart_2 <- ggplot(bc_data_unique_var_inclwt_long_2nd_half, aes(x =variant_rename, y = value, fill = variable), show.legend=F) + geom_col() +  labs(y ="% total", x ="",fill = (expression(bold(paste("A", beta," peptide"))))) +theme(axis.text.x = element_text(angle = 90, vjust = 0, hjust=1, size = 20, face="bold")) + theme(axis.text.y = element_text(size = 25, face ="bold"))+ theme(axis.title.y = element_text(size = 25, face ="bold"))  +scale_fill_manual(values=c( '#02778a',"#82a2bf", '#d4d2e5','#b87caf','#a30348'), labels = c((expression(bold(paste("A", beta,"37")))), (expression(bold(paste("A", beta,"38")))), (expression(bold(paste("A", beta,"40")))), (expression(bold(paste("A", beta,"42")))), (expression(bold(paste("A", beta,"43"))))))+  theme(legend.title = element_text(size=30, family="Helvetica", face="bold"))+
  theme(legend.text = element_text(size=30, family="Helvetica", face="bold"))+  theme(legend.position = c(0.9, 0.7))+  
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent"))

# plot together
stacked_barchat_group <- plot_grid(stacked_barchart_1,stacked_barchart_2, ncol =1)

# save
dev.new()
pdf("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/stacked_barchat_group.pdf", width=25, height=25)
stacked_barchat_group
dev.off()
```

# g-sec composite compared with AAO; figure 2b
```{r}


cor.test(bc_data_unique_var_rmwt$AAO_MutAAO, bc_data_unique_var_rmwt$g_gsec_composite_rel_wt)

#Next, we assessed relationships between the relative concentrations of Aβ species, using the GSC, produced by each variant and AAO. Lower GSC values (Pearson’s correlation [degrees of freedom], p-value: r[159] = 0.58, p < 0.0001) were associated with earlier AAO (Figure 1C). 
library(ggbreak)

max(bc_data_unique_var_rmwt$g_gsec_composite_rel_wt)


bc_data_unique_var_rmwt$paper <- as.factor(bc_data_unique_var_rmwt$paper)
print(Liu_and_dian_gsec_composite_v_AAO <-bc_data_unique_var_rmwt %>%
        ggplot( aes(x=g_gsec_composite_rel_wt, y=AAO_MutAAO, color = paper)) +
        geom_smooth ( method = "lm", size = 3,se=TRUE, color="black")+ 
        geom_point(size=6,show.legend=T, aes(color=paper, shape = paper)) + scale_color_manual(values=c( 'cornflowerblue', '#800000'), label = c( "Non-DIAN-Obs PSEN1 variants", "DIAN-Obs PSEN1 variants")) +
        labs( x=expression(bold(paste(gamma~"-secretase activity composite (% rel to wt)"))), y=expression(bold(paste("Mutation Age of Symptom Onset (years)"))), color ="Dataset", shape= "Dataset" )+
        scale_shape_manual(values = c(17,15), label = c("Non-DIAN-Obs PSEN1 variants", "DIAN-Obs PSEN1 variants"))+ 
        scale_x_continuous(breaks = c(0,20,40,60,80,100), labels = c("0", "20", "40","60" ,"80","100"), limits = c(0,115))+
        scale_y_continuous(breaks = c(0, 20,40,60,80,100), labels = c("0", "20", "40","60" ,"80","100"), limits = c(0,80))+
        
       # scale_y_break(breaks = c(1, 10), space=.5, ticklabels = c(0, 10,20, 40, 60, 80))+
      #   annotate("text", size = 15, x=75, y=25, label= "R-squared = 0.34",color = "black", fontface="bold")+
      # annotate("text", size = 15, x=75, y=22, label= "p = 6.92 e-16",color = "black", fontface="bold")+
       theme_bw() +
        theme(axis.text = element_text(size = 45))+
        theme(legend.title=element_text(size = 40, face="bold")) +
        theme(legend.text=element_text(size= 35, face = "bold",family="Helvetica")) +
        theme(axis.title.y = element_text(size = 45)) + 
        theme(axis.title.x = element_text(size = 45, face="bold")) +
        theme(legend.position = c(0.4,.90))+ 
         theme(axis.text.y.right = element_blank(), axis.line.y.right = element_blank(), axis.ticks.y.right = element_blank())+
        theme(plot.margin = margin(2, 1, 1, 1, "cm")))


dev.new()
  pdf("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/Revision_3/figure1c_raw.pdf", width=15, height=15)
Liu_and_dian_gsec_composite_v_AAO
dev.off()

```



```{r}

library(stringr); library(viridis)

regexp <- "[[:digit:]]+"
bc_data$variant_codon <- as.numeric(str_extract(bc_data$variant_rename,regexp))

bc_data$variant_codon 

Formatting_Scatter = #All of this just controls the physical look of the image
  theme_bw() +
  theme(axis.text = element_text(size = 40, face="bold"))+
  theme(legend.title=element_blank()) +
  theme(legend.text=element_text(size= 5, family="Helvetica")) +
  theme(axis.title.y = element_text(size = 20, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(size = 20, face="bold")) +
  theme(legend.position = "bottom" )
cPlot=ggplot(bc_data, aes(x=g_gsec_composite_rel_wt, y=as.factor(variant_codon), group = as.factor(variant_codon), label=variant_rename))
line = geom_smooth ( method = "lm", size = 2,se=TRUE, color = "black")
Point = geom_point(size=15,show.legend=T, aes(color=g_gsec_composite_rel_wt))
color=scale_color_viridis(  discrete=FALSE,direction=1, option = "C") #set colors and labels
#color= scale_color_manual(values=c( 'black', '#2b8cbe')) #set colors and labels
xlab =xlab(expression(bold(paste(AAO))))
text=geom_text(hjust=-0.2, vjust=-0.2, size=10,face="bold", angle=45,aes(color=g_gsec_composite_rel_wt))
title = labs(title = "") 
aoo_ab_composite2=cPlot+text +title+xlab + Formatting_Scatter  + line+color + Point  
print(aoo_ab_composite2)

# Extract the legend. Returns a gtable
leg <- get_legend(aoo_ab_composite2)

library(ggpubr)
# Convert to a ggplot and print
legend <- as_ggplot(leg)

dev.new()
  pdf("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/Revision_3/line_chart_psen1fig2c.pdf", width=25, height=70)
aoo_ab_composite2
dev.off()

#write.csv(bc_data, "/Users/sws59/Desktop/bc_data.csv")
```

# load and merge DIAN data
```{r eval=FALSE, include=FALSE}

source("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/Revision_1/DF15_Load_and_Subset_Script_july2022.R")

max$j_MutationType_2 <- ifelse(max$j_MutationType_2 == "PSEN1_Phe175del", "PSEN1_Phe176del", max$j_MutationType_2)

educ <- cbind.data.frame (max$newid15, max$EDUC.x)
educ1 <-educ %>% filter(`max$EDUC.x` > -1 )
educ1$EDUC_s <-educ1$`max$EDUC.x`
educ1$newid15 <-educ1$`max$newid15`
highest_educ1 <- aggregate(EDUC_s~newid15, educ1, FUN = "max")
base2 <- max
base_edit <- merge(highest_educ1, base2, by = c("newid15"))

RACE <- cbind.data.frame (max$newid15, max$RACE.x)
RACE1 <-RACE %>% filter(`max$RACE.x` > -1 )
RACE1$RACE1_s <-RACE1$`max$RACE.x`
RACE1$newid15 <-RACE1$`max$newid15`
highest_race1 <- aggregate(RACE1_s~newid15, RACE1, FUN = "max")
base_edit_race <- merge(highest_race1, base_edit, by = c("newid15"))



RACE_oth <- cbind.data.frame (max$newid15, max$RACEX.x)
RACE_oth1 <-RACE_oth %>% filter(`max$RACEX.x` > -1 )
RACE_oth1$RACE_oth1_s <-RACE_oth1$`max$RACEX.x`
RACE_oth1$newid15 <-RACE_oth1$`max$newid15`
highest_RACE_oth1 <- aggregate(RACE_oth1_s~newid15, RACE_oth1, FUN = "max")
base_edit_race <- merge(highest_RACE_oth1, base_edit_race, by = c("newid15"), all.y = TRUE)


#merge together biochemical data with mean parental age of onset 
PARENTAO_list<- aggregate(PARENTAO~j_MutationType_2, base_edit_race, mean)
max_bc_data_merge <- merge(bc_data_unique_var_inclwt, PARENTAO_list, by = c("j_MutationType_2"))

mEYO_ONSET_MEAN_list<- aggregate(mEYO_ONSET_MEAN~j_MutationType_2, base_edit_race, mean)
codon_number_list<- aggregate(codon_number~j_MutationType_2, base_edit_race, mean)

bc_mEYO_merge <- merge(max_bc_data_merge, mEYO_ONSET_MEAN_list, by = c("j_MutationType_2"), all.x = TRUE)
bc_mEYO_merge<- merge(bc_mEYO_merge, codon_number_list, by = c("j_MutationType_2"), all.x = TRUE)

base_edit_race$visit_date.x

dian_df_slim <- base_edit_race %>%
  dplyr::select(visit_date.x, j_MutationType_2,newid15,imagid.x,EDUC_s,RACE1_s,RACE_oth1_s,j_PIB_RSF_CORTMEAN,j_adj_HV,MMSE,fam_mutation_chg,Mutation,visitage.x,LUMIPULSE_CSF_AB42,LUMIPULSE_CSF_AB40,SEX,LUMIPULSE_CSF_pTau,master_famid_fact,PT205_D_T205_Y10_Y10_RATE_COR,PT217_D_T217_Y6_RATE_COR,PT181_D_T181_Y14_RATE_COR,PS202_Y10_RATE_COR,FDG_fSUVR_TOT_CTX_PRECUNEUS,MR_TOTV_INTRACRANIAL,MR_LV_HIPPOCAMPUS,MR_RV_HIPPOCAMPUS,CDRSUM,MEMUNITS,VISITAGEc.x,z_cent_bl_HV,z_cent_LUMIPULSE_CSF_ptau,z_cent_PIB_RSF_CORTMEAN,z_cent_LUMIPULSE_CSF_AB42,z_cent_FDG_POSTMID,z_cent_CDRSUM,
PIB_fSUVR_rsf_TOT_CTX_SSTSBANK ,
PIB_fSUVR_rsf_TOT_CTX_CAUDANTCNG ,
PIB_fSUVR_rsf_TOT_CTX_CAUDMIDFRN ,
PIB_fSUVR_rsf_TOT_CTX_CUNEUS ,
PIB_fSUVR_rsf_TOT_CTX_ENTORHINAL ,
PIB_fSUVR_rsf_TOT_CTX_FRNPOLE ,
PIB_fSUVR_rsf_TOT_CTX_FUSIFORM ,
PIB_fSUVR_rsf_TOT_CTX_INFERPRTL ,
PIB_fSUVR_rsf_TOT_CTX_INFERTMP ,
PIB_fSUVR_rsf_TOT_CTX_INSULA ,
PIB_fSUVR_rsf_TOT_CTX_ISTHMUSCNG ,
PIB_fSUVR_rsf_TOT_CTX_LATOCC ,
PIB_fSUVR_rsf_TOT_CTX_LATORBFRN ,
PIB_fSUVR_rsf_TOT_CTX_LINGUAL ,
PIB_fSUVR_rsf_TOT_CTX_MEDORBFRN ,
PIB_fSUVR_rsf_TOT_CTX_MIDTMP ,
PIB_fSUVR_rsf_TOT_CTX_PARACNTRL ,
PIB_fSUVR_rsf_TOT_CTX_PARAHPCMPL ,
PIB_fSUVR_rsf_TOT_CTX_PARSOPCLRS ,
PIB_fSUVR_rsf_TOT_CTX_PARSORBLS ,
PIB_fSUVR_rsf_TOT_CTX_PARSTRNGLS ,
PIB_fSUVR_rsf_TOT_CTX_PERICLCRN ,
PIB_fSUVR_rsf_TOT_CTX_POSTCNTRL ,
PIB_fSUVR_rsf_TOT_CTX_POSTCNG ,
PIB_fSUVR_rsf_TOT_CTX_PRECNTRL ,
PIB_fSUVR_rsf_TOT_CTX_PRECUNEUS ,
PIB_fSUVR_rsf_TOT_CTX_ROSANTCNG ,
PIB_fSUVR_rsf_TOT_CTX_ROSMIDFRN ,
PIB_fSUVR_rsf_TOT_CTX_SUPERFRN ,
PIB_fSUVR_rsf_TOT_CTX_SUPERPRTL ,
PIB_fSUVR_rsf_TOT_CTX_SUPERTMP ,
PIB_fSUVR_rsf_TOT_CTX_SUPRAMRGNL ,
PIB_fSUVR_rsf_TOT_CTX_TMPPOLE ,
PIB_fSUVR_rsf_TOT_CTX_TRANSTMP ,
PIB_fSUVR_rsf_TOT_HIPPOCAMPUS ,
PIB_fSUVR_rsf_TOT_AMYGDALA ,
PIB_fSUVR_rsf_TOT_PALLIDUM ,
PIB_fSUVR_rsf_TOT_THALAMUS_PRPR ,
PIB_fSUVR_rsf_TOT_PUTAMEN ,
PIB_fSUVR_rsf_TOT_CAUD ,FDG_fSUVR_rsf_TOT_CTX_SSTSBANK,
FDG_fSUVR_rsf_TOT_CTX_CAUDANTCNG,
FDG_fSUVR_rsf_TOT_CTX_CAUDMIDFRN,
FDG_fSUVR_rsf_TOT_CTX_CUNEUS,
FDG_fSUVR_rsf_TOT_CTX_ENTORHINAL,
FDG_fSUVR_rsf_TOT_CTX_FRNPOLE,
FDG_fSUVR_rsf_TOT_CTX_FUSIFORM,
FDG_fSUVR_rsf_TOT_CTX_INFERPRTL,
FDG_fSUVR_rsf_TOT_CTX_INFERTMP,
FDG_fSUVR_rsf_TOT_CTX_INSULA,
FDG_fSUVR_rsf_TOT_CTX_ISTHMUSCNG,
FDG_fSUVR_rsf_TOT_CTX_LATOCC,
FDG_fSUVR_rsf_TOT_CTX_LATORBFRN,
FDG_fSUVR_rsf_TOT_CTX_LINGUAL,
FDG_fSUVR_rsf_TOT_CTX_MEDORBFRN,
FDG_fSUVR_rsf_TOT_CTX_MIDTMP,
FDG_fSUVR_rsf_TOT_CTX_PARACNTRL,
FDG_fSUVR_rsf_TOT_CTX_PARAHPCMPL,
FDG_fSUVR_rsf_TOT_CTX_PARSOPCLRS,
FDG_fSUVR_rsf_TOT_CTX_PARSORBLS,
FDG_fSUVR_rsf_TOT_CTX_PARSTRNGLS,
FDG_fSUVR_rsf_TOT_CTX_PERICLCRN,
FDG_fSUVR_rsf_TOT_CTX_POSTCNTRL,
FDG_fSUVR_rsf_TOT_CTX_POSTCNG,
FDG_fSUVR_rsf_TOT_CTX_PRECNTRL,
FDG_fSUVR_rsf_TOT_CTX_PRECUNEUS,
FDG_fSUVR_rsf_TOT_CTX_ROSANTCNG,
FDG_fSUVR_rsf_TOT_CTX_ROSMIDFRN,
FDG_fSUVR_rsf_TOT_CTX_SUPERFRN,
FDG_fSUVR_rsf_TOT_CTX_SUPERPRTL,
FDG_fSUVR_rsf_TOT_CTX_SUPERTMP,
FDG_fSUVR_rsf_TOT_CTX_SUPRAMRGNL,
FDG_fSUVR_rsf_TOT_CTX_TMPPOLE,
FDG_fSUVR_rsf_TOT_CTX_TRANSTMP,
FDG_fSUVR_rsf_TOT_HIPPOCAMPUS,
FDG_fSUVR_rsf_TOT_AMYGDALA,
FDG_fSUVR_rsf_TOT_PALLIDUM,
FDG_fSUVR_rsf_TOT_THALAMUS_PRPR,
FDG_fSUVR_rsf_TOT_PUTAMEN,
FDG_fSUVR_rsf_TOT_CAUD,
j_adj_SSTSBANK,
j_adj_CAUDANTCNG,
j_adj_CAUDMIDFRN,
j_adj_CUNEUS,
j_adj_ENTORHINAL,
j_adj_FRNPOLE,
j_adj_FUSIFORM,
j_adj_INFRPRTL,
j_adj_INFRTMP,
j_adj_INSULA,
j_adj_ISTHMUSCNG,
j_adj_LATOCC,
j_adj_LATORBFRN,
j_adj_LINGUAL,
j_adj_MEDORBFRN,
j_adj_MIDTMP,
j_adj_PARACNTRL,
j_adj_PARAHPCMPL,
bl_PARAOPRCLRS,
j_adj_PARSORBLS,
j_adj_PARSTRNGLRS,
j_adj_PERICLCRN,
j_adj_POSTCNTRL,
j_adj_POSTCNG,
j_adj_PRECNTRL,
j_adj_PRECUNEUS,
j_adj_ROSANTCNG,
j_adj_ROSMIDFRN,
j_adj_SUPERFRN,
j_adj_SUPERPRTL,
j_adj_SUPERTMP,
j_adj_SUPRAMRGNL,
j_adj_TMPPOLE,
j_adj_TRANSTMP,
j_adj_HV,
j_adj_AMYGDALA,
j_adj_PALLIDUM,
j_adj_THALAMUS,
j_adj_PUTAMEN,
j_adj_CAUD,
time_wrt_base,
earliest_age,
APOE_binary,
CDRGLOB_f3,
dian_mutpar_eyo
)

dian_df_bc_df <- merge(bc_mEYO_merge,dian_df_slim, by = c("j_MutationType_2"))
#restrict to visits with pib, mri, and cognitive variables
dian_matched_bc_data_pib <-dian_df_bc_df %>% filter(j_PIB_RSF_CORTMEAN > -1 )
dian_matched_bc_data_pib_mri <-dian_matched_bc_data_pib %>% filter(j_adj_HV > -1 )
dian_matched_bc_data_pib_mri_mmse <-dian_matched_bc_data_pib_mri %>% filter(MMSE > -1 )
dian_matched_bc_data_pib_mri_mmse2 <-dian_matched_bc_data_pib_mri_mmse %>% filter(MMSE < 31 )


```

### cross-sectional analyses
```{r}

#want to make the dataset cross-sectional. Could choose first of last visit. I decided to choose the last visit so the chances of having elevated biomarker levels were greater. 
latest <- aggregate(visitage.x~newid15, dian_matched_bc_data_pib_mri_mmse2, FUN = "max") #choose last visit 
dian_matched_bc_data_pib_mri_mmse2_latest <- merge(latest, dian_matched_bc_data_pib_mri_mmse2, by = c("newid15", "visitage.x")) #restrict longitudinal dataset to rows for pertaining to individuals last visit

#really only interested in MC data
dian_matched_bc_data_pib_mri_mmse2_latest_MC <-dian_matched_bc_data_pib_mri_mmse2_latest %>% filter(Mutation == 1 )

dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s_dichot <- ifelse(dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s == "White", "a_white","b_Other")


dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s_primary <- ifelse(dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s == "Other (specify)", dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE_oth1_s ,dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s)



race_lookup <- dian_matched_bc_data_pib_mri_mmse2_latest_MC %>% filter (dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s_primary == "none" |dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s_primary == "Unknown")
dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s_primary_s <- ifelse(dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s_primary == "Unknown" | dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s_primary == "none" |
dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s_primary ==  "Croatian" |
  dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s_primary ==  "North African" |
  dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s_primary ==  "Arab" , "None, Unknown, or Other", 
ifelse( dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s_primary == "Latin American" | dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s_primary ==   "LATIN" |
dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s_primary ==   "Latin" | dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s_primary ==  "Hispanic" |                           dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s_primary ==  "Mexican American", 
"Latin America" ,
ifelse(dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s_primary == "Australian aboriginal" | dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s_primary == "Aboriginal" |
         dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s_primary == "Australian Indigenous" , "Australian Aboriginal or Indigenous",dian_matched_bc_data_pib_mri_mmse2_latest_MC$RACE1_s_primary )))

write.csv(dian_matched_bc_data_pib_mri_mmse2, "dian_matched_bc_data_pib_mri_mmse2.csv")


#Characteristic
#Value (N = 190)
#Age, years
#40.0 (10.3)
#EYO, years
#-5.8 (10.4)
#AAO, years
#46.8 (7.8)
#APOE4+, % (N)
#29 (55)
#Female, % (N)
#58 (110)
#Education, years
#14.4 (2.9)
#CDR (0, 0.5, 1+), % (N)
#58 (110), 23 (44), 19 (36)


#LN_paper_ids <- as.data.frame(dian_matched_bc_data_pib_mri_mmse2_latest_MC$newid15)
#write.csv (LN_paper_ids, "LN_paper_ids.csv")
table0 <- dian_matched_bc_data_pib_mri_mmse2_latest_MC %>% select(visitage.x, SEX, APOE_binary,EDUC_s, dian_mutpar_eyo,PARENTAO, CDRGLOB_f3,RACE1_s_primary_s) %>%
  mutate(CDRGLOB_f3 = factor(CDRGLOB_f3, labels = c("CDR 0", "CDR 0.5", "CDR 1+"))) %>%
  mutate(SEX = factor(SEX, labels = c("Male", "Female")))%>%
  mutate(APOE_binary = factor(APOE_binary, labels = c("Carrier", "Non-Carrier")))

library(gtsummary)

 table1 <- gtsummary::tbl_summary(table0,
                      digits = all_continuous() ~ 1,
                      missing = "no" ,
                      label = list(visitage.x ~ "Age (yrs)",
                                   SEX = "Sex",
                                   RACE1_s_primary_s = "Race",
                                   APOE_binary = "APOE4",
                                   dian_mutpar_eyo = "EYO (yrs)",
                                   PARENTAO = "AAO (yrs)",
                                   EDUC_s = "Education (yrs)",
                                   CDRGLOB_f3 = "CDR Global")) %>%
  modify_header(label = "**Variable**") %>% 
  modify_caption("**Table 1. Cross-sectional Participant Characteristics**") %>%
  # update the column header
  bold_labels()

 library(flextable); library(gt)
 table1_flex <-table1 %>% as_flex_table() 
 save_as_image(table1_flex, "table1_flex.png", expand = 100, res = 1000)

 as_gt(table1) %>% gt::as_latex()

  as_gt(table1) %>% gtsave("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/Revision_3/tab_1.docx")


summary(lmer(j_PIB_RSF_CORTMEAN ~  SEX + visitage.x + g_gsec_composite_rel_wt + (1|master_famid_fact), dian_matched_bc_data_pib_mri_mmse2_latest_MC))

#In the DIAN Observation study sample (n = 190, representing 56 unique PSEN1 variants), lower GSC (decreased -secretase activity relative to wildtype) was associated with higher levels of mean cortical Aβ burden (B[SE] = -0.03[0.01], p < 0.0001; Figure 2A and appendix pg 18) as assessed by A PET, after controlling for demographic factors. 

summary(lmer(j_adj_HV ~  SEX+ visitage.x + g_gsec_composite_rel_wt + (1|master_famid_fact), dian_matched_bc_data_pib_mri_mmse2_latest_MC))
summary(lmer(FDG_fSUVR_TOT_CTX_PRECUNEUS ~  SEX + visitage.x + g_gsec_composite_rel_wt + (1|master_famid_fact), dian_matched_bc_data_pib_mri_mmse2_latest_MC))

#We next examined associations between the variant-level GSC and commonly used imaging biomarkers of neurodegeneration. Both MRI-based hippocampal volume (B[SE] = 37.35[6.3], p < 0.0001; Figure 2B and appendix pg 18) and precuneus FDG-PET signal (B[SE] = 0.004[0.001], p = 0.001; Figure 2C and appendix pg 18) were highly associated with the GSC, after controlling for demographic factors. 


dian_matched_bc_data_pib_mri_mmse2_latest_MC$LUMIPULSE_CSF_AB42_40 <- dian_matched_bc_data_pib_mri_mmse2_latest_MC$LUMIPULSE_CSF_AB42/dian_matched_bc_data_pib_mri_mmse2_latest_MC$LUMIPULSE_CSF_AB40
dian_matched_bc_data_pib_mri_mmse2_latest_MC_csf <-dian_matched_bc_data_pib_mri_mmse2_latest_MC %>% filter(LUMIPULSE_CSF_pTau >0 )

dian_matched_bc_data_pib_mri_mmse2_latest_MC$PT217_D_T217_Y6_RATE_COR_log <- log10(dian_matched_bc_data_pib_mri_mmse2_latest_MC$PT217_D_T217_Y6_RATE_COR)

summary(lmer(LUMIPULSE_CSF_AB42_40 ~  SEX +visitage.x + g_gsec_composite_rel_wt + (1|master_famid_fact), dian_matched_bc_data_pib_mri_mmse2_latest_MC_csf))

summary(lmer(log10(LUMIPULSE_CSF_pTau) ~  SEX + visitage.x + g_gsec_composite_rel_wt + (1|master_famid_fact), dian_matched_bc_data_pib_mri_mmse2_latest_MC_csf))
summary(lmer(PT217_D_T217_Y6_RATE_COR_log ~  SEX+ visitage.x + g_gsec_composite_rel_wt + (1|master_famid_fact), dian_matched_bc_data_pib_mri_mmse2_latest_MC))

#Using established CSF measures of Alzheimer’s disease pathology, we observed that variants with lower GSC were associated with lower CSF Aβ 42/40 (B[SE] = 5.3e-04[1.4e-04], p = 0.0004; Figure 2G and appendix pg 19), higher CSF ELISA log10(phosphorylated-tauT181) levels (B[SE] = -0.007[0.002], p = 0.0003; Figure 2H and appendix pg 19), and higher CSF IP-MS log10(pT217/T217) levels (B[SE] = -0.009[0.002], p =0.0007; Figure 2I and appendix pg 19).


summary(lmer(MMSE ~ visitage.x+ SEX + EDUC_s + g_gsec_composite_rel_wt + (1|master_famid_fact), dian_matched_bc_data_pib_mri_mmse2_latest_MC))
summary(lmer(CDRSUM ~  visitage.x+ SEX + EDUC_s + g_gsec_composite_rel_wt + (1|master_famid_fact), dian_matched_bc_data_pib_mri_mmse2_latest_MC))


dian_matched_bc_data_pib_mri_mmse2_latest_MC_memunits <- filter (dian_matched_bc_data_pib_mri_mmse2_latest_MC,MEMUNITS < 40 )
summary(lmer(MEMUNITS ~ visitage.x+ SEX + EDUC_s + g_gsec_composite_rel_wt + (1|master_famid_fact), dian_matched_bc_data_pib_mri_mmse2_latest_MC_memunits))

#The cell-derived GSC levels were associated with MMSE (B[SE] = 0.05[.02], p = 0.002; Figure 2J and appendix pg 20), CDR-SB (B[SE] = -0.05[0.02], p = 0.003; Figure 2K and appendix pg 20), and logical memory-delayed scores (B[SE] = 0.09[.03], p = 0.0006; Figure 2L and appendix pg 20). 

print(p)


t <- corrplot(res3$r, type="upper", order="hclust", 
         p.mat = res3$P, sig.level = 0.01, insig = "blank")



```


#plotting cross-sectional analyses
```{r}


dian_matched_bc_data_pib_mri_mmse2_latest_MC$j_PIB_RSF_CORTMEAN_adj_age <- resid(lm(dian_matched_bc_data_pib_mri_mmse2_latest_MC$j_PIB_RSF_CORTMEAN ~ dian_matched_bc_data_pib_mri_mmse2_latest_MC$visitage.x + dian_matched_bc_data_pib_mri_mmse2_latest_MC$SEX, na.action = na.exclude)) + mean(dian_matched_bc_data_pib_mri_mmse2_latest_MC$j_PIB_RSF_CORTMEAN, na.rm = TRUE)

dian_matched_bc_data_pib_mri_mmse2_latest_MC$HV_avg <- 0.5*(dian_matched_bc_data_pib_mri_mmse2_latest_MC$MR_LV_HIPPOCAMPUS+dian_matched_bc_data_pib_mri_mmse2_latest_MC$MR_RV_HIPPOCAMPUS)

dian_matched_bc_data_pib_mri_mmse2_latest_MC$j_adj_HV_adj_age <- resid(lm(dian_matched_bc_data_pib_mri_mmse2_latest_MC$HV_avg ~ dian_matched_bc_data_pib_mri_mmse2_latest_MC$MR_TOTV_INTRACRANIAL + dian_matched_bc_data_pib_mri_mmse2_latest_MC$visitage.x + dian_matched_bc_data_pib_mri_mmse2_latest_MC$SEX, na.action = na.exclude)) + mean(dian_matched_bc_data_pib_mri_mmse2_latest_MC$HV_avg, na.rm = TRUE)

dian_matched_bc_data_pib_mri_mmse2_latest_MC_fdg <- filter(dian_matched_bc_data_pib_mri_mmse2_latest_MC, FDG_fSUVR_TOT_CTX_PRECUNEUS > -99 )
dian_matched_bc_data_pib_mri_mmse2_latest_MC_fdg$FDG_fSUVR_TOT_CTX_PRECUNEUS_adj_age <- resid(lm(dian_matched_bc_data_pib_mri_mmse2_latest_MC_fdg$FDG_fSUVR_TOT_CTX_PRECUNEUS ~ dian_matched_bc_data_pib_mri_mmse2_latest_MC_fdg$visitage.x + dian_matched_bc_data_pib_mri_mmse2_latest_MC_fdg$SEX, na.action = na.exclude)) + mean(dian_matched_bc_data_pib_mri_mmse2_latest_MC_fdg$FDG_fSUVR_TOT_CTX_PRECUNEUS, na.rm = TRUE)

dian_matched_bc_data_pib_mri_mmse2_latest_MC_csf$csfab42_40_adj_age <- resid(lm(dian_matched_bc_data_pib_mri_mmse2_latest_MC_csf$LUMIPULSE_CSF_AB42_40 ~ dian_matched_bc_data_pib_mri_mmse2_latest_MC_csf$visitage.x + dian_matched_bc_data_pib_mri_mmse2_latest_MC_csf$SEX, na.action = na.exclude)) + mean(dian_matched_bc_data_pib_mri_mmse2_latest_MC_csf$LUMIPULSE_CSF_AB42_40, na.rm = TRUE)

dian_matched_bc_data_pib_mri_mmse2_latest_MC_csf$log_LUMIPULSE_CSF_pTau_adj_age <- resid(lm(log10(dian_matched_bc_data_pib_mri_mmse2_latest_MC_csf$LUMIPULSE_CSF_pTau) ~ dian_matched_bc_data_pib_mri_mmse2_latest_MC_csf$visitage.x + dian_matched_bc_data_pib_mri_mmse2_latest_MC_csf$SEX, na.action = na.exclude)) + mean(log10(dian_matched_bc_data_pib_mri_mmse2_latest_MC_csf$LUMIPULSE_CSF_pTau), na.rm = TRUE)

dian_matched_bc_data_pib_mri_mmse2_latest_MC_ipmscsf <-dian_matched_bc_data_pib_mri_mmse2_latest_MC %>% filter(PT217_D_T217_Y6_RATE_COR >0 )
dian_matched_bc_data_pib_mri_mmse2_latest_MC_ipmscsf$log_LUMIPULSE_CSF_ipms_pTau217_adj_age <- resid(lm(log10(dian_matched_bc_data_pib_mri_mmse2_latest_MC_ipmscsf$PT217_D_T217_Y6_RATE_COR) ~ dian_matched_bc_data_pib_mri_mmse2_latest_MC_ipmscsf$visitage.x + dian_matched_bc_data_pib_mri_mmse2_latest_MC_ipmscsf$SEX, na.action = na.exclude)) + mean(log10(dian_matched_bc_data_pib_mri_mmse2_latest_MC_ipmscsf$PT217_D_T217_Y6_RATE_COR), na.rm = TRUE)

dian_matched_bc_data_pib_mri_mmse2_latest_MC$cdr_age_adj <- resid(lm(dian_matched_bc_data_pib_mri_mmse2_latest_MC$CDRSUM ~ dian_matched_bc_data_pib_mri_mmse2_latest_MC$visitage.x + dian_matched_bc_data_pib_mri_mmse2_latest_MC$SEX + dian_matched_bc_data_pib_mri_mmse2_latest_MC$EDUC_s, na.action = na.exclude)) + mean(dian_matched_bc_data_pib_mri_mmse2_latest_MC$CDRSUM, na.rm = TRUE)

 dian_matched_bc_data_pib_mri_mmse2_latest_MC$MMSE_age_adj <- resid(lm(dian_matched_bc_data_pib_mri_mmse2_latest_MC$MMSE ~ dian_matched_bc_data_pib_mri_mmse2_latest_MC$visitage.x + dian_matched_bc_data_pib_mri_mmse2_latest_MC$SEX + dian_matched_bc_data_pib_mri_mmse2_latest_MC$EDUC_s, na.action = na.exclude)) + mean(dian_matched_bc_data_pib_mri_mmse2_latest_MC$MMSE, na.rm = TRUE) 

dian_matched_bc_data_pib_mri_mmse2_latest_MC_memunits$MEMUNITS_age_adj <- resid(lm(dian_matched_bc_data_pib_mri_mmse2_latest_MC_memunits$MEMUNITS ~ dian_matched_bc_data_pib_mri_mmse2_latest_MC_memunits$visitage.x + dian_matched_bc_data_pib_mri_mmse2_latest_MC_memunits$SEX + dian_matched_bc_data_pib_mri_mmse2_latest_MC_memunits$EDUC_s, na.action = na.exclude)) + mean(dian_matched_bc_data_pib_mri_mmse2_latest_MC_memunits$MEMUNITS, na.rm = TRUE)


Formatting_Scatter = #All of this just controls the physical look of the image
  theme_bw() +
  theme(axis.text = element_text(size = 20, face="bold"))+
  theme(legend.title=element_blank()) +
  theme(legend.text=element_text(size= 35, family="Helvetica")) +
  theme(axis.title.y = element_text(size = 25, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(size = 25, face="bold")) +
  theme(plot.margin = unit(c(2,2,2,2), "lines")) +
  theme(legend.position = "bottom" )


cPlot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC, aes(x=g_gsec_composite_rel_wt, y=j_PIB_RSF_CORTMEAN_adj_age))
line = geom_smooth ( method = "lm", size=2,se=TRUE, color = "black") 
Point = geom_point(size=4,show.legend=F, color = "black", position = position_jitter(w = 3, h = 0)  ) 
xlab =xlab(expression(bold(paste(gamma~"-secretase activity composite (% rel to wt)"))))
ylab= ylab( expression(bold("Cortical PiB PET (SUVR; Covariate-adj.)"))) 
ylim = ylim(0,7)
pib_abcomposite_adjage=cPlot + ylim+ Point +line + xlab +ylab + Formatting_Scatter  
print(pib_abcomposite_adjage)

cPlot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC, aes(x=g_gsec_composite_rel_wt, y=j_adj_HV_adj_age))
line = geom_smooth ( method = "lm", size=2,se=TRUE, color = "black") 
Point = geom_point(size=4,show.legend=F, color = "black" , position = position_jitter(w = 3, h = 0) ) 
text = annotate("text", size = 8, x=1.3 ,y=-3700, fontface=2,label= "B (SE) = 1999.53 (281.63)",color = "black")
text1 = annotate("text", size = 8, x=1.3, y=-4200, fontface=2,label= "p = 2.13e-10",color = "black")
#ybreak =   scale_y_break(breaks = c(1, 1000), space=.5) #ticklabels = c(0,1, 1500, 2000, 3000, 4000, 5000)
ycont = scale_y_continuous(limits = c(1500, 5500), breaks  = c( 1500, 2000, 3000, 4000, 5000), labels = c("0", "2000", "3000", "4000", "5000"))
xlab =xlab(expression(bold(paste(gamma~"-secretase activity composite (% rel to wt)"))))
ylab= ylab(bquote(bold('Hippocampal Volume ('~ mm^3~"; Covariate-adj.)")))
hcv_abcomposite_adjage=cPlot + Point +line+xlab +ylab + ycont+Formatting_Scatter 
print(hcv_abcomposite_adjage)

cPlot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC_fdg, aes(x=g_gsec_composite_rel_wt, y=FDG_fSUVR_TOT_CTX_PRECUNEUS_adj_age))
line = geom_smooth ( method = "lm", size=2,se=TRUE, color = "black") 
Point = geom_point(size=4,show.legend=F, color = "black" , position = position_jitter(w = 3, h = 0) ) 
text = annotate("text", size = 8, x=1.3 ,y=-3700, fontface=2,label= "B (SE) = 1999.53 (281.63)",color = "black")
ycont = scale_y_continuous(limits = c(0.5,1.75), breaks  = c( .5, .75, 1.25, 1.75), labels = c("0", "0.75", "1.25", "1.75"))
text1 = annotate("text", size = 8, x=1.3, y=-4200, fontface=2,label= "p = 2.13e-10",color = "black")
xlab =xlab(expression(bold(paste(gamma~"-secretase activity composite (% rel to wt)"))))
ylab= ylab(bquote(bold('Precuneus Glucose Metabolism (SUVR; Covariate-adj.)')))
fdg_abcomposite_adjage=cPlot + ycont+ Point +line+ xlab +ylab + Formatting_Scatter 
print(fdg_abcomposite_adjage)

dian_matched_bc_data_pib_mri_mmse2_latest_MC_csf$csfab42_40_adj_age
cPlot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC_csf, aes(x=g_gsec_composite_rel_wt, y=csfab42_40_adj_age))
ylim = ylim(0,.15)
line = geom_smooth ( method = "lm", size=2,se=TRUE, color = "black") 
Point = geom_point(size=4,show.legend=F, color = "black", position = position_jitter(w = 3, h = 0)  ) 
xlab =xlab(expression(bold(paste(gamma~"-secretase activity composite (% rel to wt)"))))
ylab= ylab( expression(bold(paste("CSF A", beta, "42/40 (Covariate-adj.)"))))
csfab42_40_abcomposite_adjage=cPlot + ylim+ Point +line + xlab +ylab + Formatting_Scatter  
print(csfab42_40_abcomposite_adjage)

cPlot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC_csf, aes(x=g_gsec_composite_rel_wt, y=log_LUMIPULSE_CSF_pTau_adj_age))
ycont = scale_y_continuous(limits = c(0.75,3), breaks  = c( .75, 1, 2, 3), labels = c("0",  "1", "2", "3"))
line = geom_smooth ( method = "lm", size=2,se=TRUE, color = "black") 
Point = geom_point(size=4,show.legend=F, color = "black", position = position_jitter(w = 3, h = 0)  ) 
xlab =xlab(expression(bold(paste(gamma~"-secretase activity composite (% rel to wt)"))))
ylab= ylab( expression(bold(paste(log[10]~CSF~p-"tau"~181~"(Covariate-adj.)"))))
csf_ptau_abcomposite=cPlot + ycont+ Point +line + xlab +ylab + Formatting_Scatter  
print(csf_ptau_abcomposite)

cPlot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC_ipmscsf, aes(x=g_gsec_composite_rel_wt, y=log_LUMIPULSE_CSF_ipms_pTau217_adj_age))
line = geom_smooth ( method = "lm", size=2,se=TRUE, color = "black") 
Point = geom_point(size=4,show.legend=F, color = "black", position = position_jitter(w = 3, h = 0)  ) 
xlab =xlab(expression(bold(paste(gamma~"-secretase activity composite (% rel to wt)"))))
ylab= ylab( expression(bold(paste(log[10]~IP-MS~CSF~p-"tau"~217~"(Covariate-adj.)"))))
msipcsf_ptau217_abcomposite=cPlot +  Point +line + xlab +ylab + Formatting_Scatter  
print(msipcsf_ptau217_abcomposite)

cPlot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC, aes(x=g_gsec_composite_rel_wt, y=cdr_age_adj))
line = geom_smooth ( method = "lm", size=2,se=TRUE, color = "black") 
Point = geom_point(size=4,show.legend=F, color = "black" ,position = position_jitter(w = 3, h = 0)) 
text = annotate("text", size = 8, x=.4, y=10, fontface=2,label= "B (SE) = 6.15 (1.35)",color = "black")
text1 = annotate("text", size = 8, x=.4, y=7.5, fontface=2,label= "p = 1.67e-05",color = "black")
xlab =xlab(expression(bold(paste(gamma~"-secretase activity composite (% rel to wt)"))))
ylab= ylab( expression(bold("CDR-SB (Covariate-adj.)")))
cdr_abcomposite_adjage=cPlot +  Point +line+ xlab +ylab + Formatting_Scatter 
print(cdr_abcomposite_adjage)

cPlot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC, aes(x=g_gsec_composite_rel_wt, y=MMSE_age_adj))
line = geom_smooth ( method = "lm", size=2,se=TRUE, color = "black") 
Point = geom_point(size=4,show.legend=F, color = "black",position = position_jitter(w = 3, h = 0) ) 
text = annotate("text", size = 8, x=.4, y=10, fontface=2,label= "B (SE) = 6.15 (1.35)",color = "black")
text1 = annotate("text", size = 8, x=.4, y=7.5, fontface=2,label= "p = 1.67e-05",color = "black")
xlab =xlab(expression(bold(paste(gamma~"-secretase activity composite (% rel to wt)"))))
ylab= ylab( expression(bold("MMSE (Covariate-adj.)")))
ylim = ylim(0, 35)
mmse_abcomposite_adjage=cPlot + ylim+ Point +line+ xlab +ylab + Formatting_Scatter 
print(mmse_abcomposite_adjage)

cPlot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC_memunits, aes(x=g_gsec_composite_rel_wt, y=MEMUNITS_age_adj))
line = geom_smooth ( method = "lm", size=2,se=TRUE, color = "black") 
Point = geom_point(size=4,show.legend=F, color = "black" , position = position_jitter(w =3, h = 0) ) 
text = annotate("text", size = 8, x=1.3 ,y=-3700, fontface=2,label= "B (SE) = 1999.53 (281.63)",color = "black")
text1 = annotate("text", size = 8, x=1.3, y=-4200, fontface=2,label= "p = 2.13e-10",color = "black")
xlab =xlab(expression(bold(paste(gamma~"-secretase activity composite (% rel to wt)"))))
ylab= ylab(bquote(bold('Logical Memory Delayed (Covariate-adj.)')))
memunits_abcomposite_adjage=cPlot +  Point +line+ xlab +ylab + Formatting_Scatter 
print(memunits_abcomposite_adjage)






Figure_2_ac<-plot_grid(pib_abcomposite_adjage, hcv_abcomposite_adjage,fdg_abcomposite_adjage
,nrow = 1)
dev.new()
svg("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/Revision_3/Figure_3_ac.svg", width=35, height=10)
Figure_2_ac
dev.off()


Figure_2_df<-plot_grid(csfab42_40_abcomposite_adjage,csf_ptau_abcomposite, msipcsf_ptau217_abcomposite,nrow = 1)

dev.new()
svg("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/Revision_3/Figure_3_df.svg", width=35, height=10)
Figure_2_df
dev.off()



Figure_2_gi<-plot_grid(mmse_abcomposite_adjage,cdr_abcomposite_adjage,memunits_abcomposite_adjage,nrow = 1)

dev.new()
svg("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/revision_3/Figure_3_gi.", width=35, height=10)
Figure_2_gi
dev.off()


```


#### Exploratory regional analyses and ROI figure ####
```{r}
#install.packages("extrafont")

library(ggseg); library(viridis); library(cowplot);library(extrafont)
#font_import()
loadfonts(device="pdf")       #Register fonts for Windows bitmap output
fonts()                


## ROI Figure 

aparc_figure <- ggseg(mapping = aes(fill = region),hemi = "left",
colour = "black") +
scale_fill_brain("dk") + theme(legend.justification = c(1, 0),legend.position = "right" , legend.text = element_text(size = 30, family ="Arial"))+
guides(fill = guide_legend(ncol = 2))


aseg_roi <- read.csv("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/Revision_1/aseg_roi.csv")
aseg_roi$region_value <- as.factor(aseg_roi$region_value)

aseg_figure<-ggseg(.data=aseg_roi, atlas = aseg, colour="grey15", hemi = "right",mapping=aes(fill=region_value ), show.legend=T) +  scale_fill_discrete(na.value="#d3d3d3") + theme(legend.justification = c(1, 0),legend.position = "right" , legend.text = element_text(size = 30, family ="Arial"))


roi_figure<-plot_grid(aparc_figure,aseg_figure, nrow = 1, labels = c("A",  "B"),  label_size=20, hjust=0)

dev.new()
pdf("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/revision_2/Figures//roi_figure010224.pdf", width=40, height=20)
roi_figure
dev.off()

tert <- cbind(dian_matched_bc_data_pib_mri_mmse2_latest_MC$g_gsec_composite_rel_wt, dian_matched_bc_data_pib_mri_mmse2_latest_MC$tert_ab_composite)

vTert = quantile(dian_matched_bc_data_pib_mri_mmse2_latest_MC$g_gsec_composite_rel_wt, c(0:3/3))

# classify values
dian_matched_bc_data_pib_mri_mmse2_latest_MC$tert_ab_composite = with(dian_matched_bc_data_pib_mri_mmse2_latest_MC, 
                                                                      cut(g_gsec_composite_rel_wt, 
                                                                          vTert, 
                                                                          include.lowest = T, 
                                                                          labels = c("Low", "Medium", "High")))

regional_df_lo_high <-dian_matched_bc_data_pib_mri_mmse2_latest_MC %>%
  filter(tert_ab_composite != "Medium")

summary(lmer(j_PIB_RSF_CORTMEAN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))

#aparc
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_SSTSBANK ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_CAUDANTCNG ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_CAUDMIDFRN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_CUNEUS ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_ENTORHINAL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_FRNPOLE ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_FUSIFORM ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_INFERPRTL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_INFERTMP ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_INSULA ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_ISTHMUSCNG ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_LATOCC ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_LATORBFRN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_LINGUAL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_MEDORBFRN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_MIDTMP ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_PARACNTRL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_PARAHPCMPL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_PARSOPCLRS ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_PARSORBLS ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_PARSTRNGLS ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_PERICLCRN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_POSTCNTRL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_POSTCNG ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_PRECNTRL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_PRECUNEUS ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_ROSANTCNG ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_ROSMIDFRN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_SUPERFRN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_SUPERPRTL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_SUPERTMP ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_SUPRAMRGNL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_TMPPOLE ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CTX_TRANSTMP ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))

#aseg
summary(lmer(PIB_fSUVR_rsf_TOT_HIPPOCAMPUS ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_AMYGDALA ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_PALLIDUM ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_THALAMUS_PRPR ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_PUTAMEN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(PIB_fSUVR_rsf_TOT_CAUD ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))



library(ggseg); library(viridis)

pib_aparc <- read.csv("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/Revision_1/aparc_pib_dec26csv.csv")

pib_aparc_plot_t<-ggseg(.data=pib_aparc, atlas = dk, colour="grey15", hemi = "left",mapping=aes(fill=t_value_lo_v_hi_lme ), show.legend=F) + theme_void()+  scale_fill_viridis(option="C", discrete=FALSE, limits= c(1.5, 7.9), na.value="#d3d3d3")
print(pib_aparc_plot_t)

pib_aseg <- read.csv("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/Revision_1/aseg_pib_dec26.csv")

pib_aseg_plot_t<-ggseg(.data=pib_aseg, atlas = aseg, colour="grey15", hemi = "left",mapping=aes(fill=t_value_lo_v_hi_lme ), show.legend=F) + theme_void()+  scale_fill_viridis(option="C", discrete=FALSE, limits= c(1.5, 7.9), na.value="#d3d3d3")
print(pib_aseg_plot_t)


#aparc
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_SSTSBANK ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_CAUDANTCNG ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_CAUDMIDFRN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_CUNEUS ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_ENTORHINAL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_FRNPOLE ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_FUSIFORM ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_INFERPRTL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_INFERTMP ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_INSULA ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_ISTHMUSCNG ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_LATOCC ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_LATORBFRN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_LINGUAL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_MEDORBFRN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_MIDTMP ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_PARACNTRL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_PARAHPCMPL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_PARSOPCLRS ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_PARSORBLS ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_PARSTRNGLS ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_PERICLCRN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_POSTCNTRL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_POSTCNG ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_PRECNTRL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_PRECUNEUS ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_ROSANTCNG ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_ROSMIDFRN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_SUPERFRN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_SUPERPRTL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_SUPERTMP ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_SUPRAMRGNL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_TMPPOLE ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CTX_TRANSTMP ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))

#aseg
summary(lmer(FDG_fSUVR_rsf_TOT_HIPPOCAMPUS ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_AMYGDALA ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_PALLIDUM ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_THALAMUS_PRPR ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_PUTAMEN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(FDG_fSUVR_rsf_TOT_CAUD ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))



library(ggseg); library(viridis)
fdg_aparc <- read.csv("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/Revision_1/aparc_fdg_dec26.csv")

fdg_aparc_sig <- fdg_aparc %>% filter (pvalue_lo_v_hi_lme < 0.05)
fdg_aparc_plot_t<-ggseg(.data=fdg_aparc_sig, atlas = dk, colour="grey15", hemi = "left",mapping=aes(fill=t_value_lo_v_hi_lme ), show.legend=F) + theme_void()+  scale_fill_viridis(option="C", discrete=FALSE, limits= c(1.5, 7.9), na.value="#d3d3d3")
print(fdg_aparc_plot_t)



fdg_aseg <- read.csv("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/Revision_1/aseg_fdg_dec26.csv")

fdg_aseg_sig <- fdg_aseg %>% filter (pvalue_lo_v_hi_lme < 0.05)

fdg_aseg_plot_t<-ggseg(.data=fdg_aseg_sig, atlas = aseg, colour="grey15", hemi = "left",mapping=aes(fill=t_value_lo_v_hi_lme ), show.legend=F) + theme_void()+  scale_fill_viridis(option="C", discrete=FALSE, limits= c(1.5, 7.9), na.value="#d3d3d3")
  
print(fdg_aseg_plot_t)








#aparc
summary(lmer(j_adj_SSTSBANK ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_CAUDANTCNG ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_CAUDMIDFRN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_CUNEUS ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_ENTORHINAL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_FRNPOLE ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_FUSIFORM ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_INFRPRTL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_INFRTMP ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_INSULA ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_ISTHMUSCNG ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_LATOCC ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_LATORBFRN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_LINGUAL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_MEDORBFRN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_MIDTMP ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_PARACNTRL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_PARAHPCMPL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(bl_PARAOPRCLRS ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_PARSORBLS ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_PARSTRNGLRS ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_PERICLCRN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_POSTCNTRL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_POSTCNG ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_PRECNTRL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_PRECUNEUS ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_ROSANTCNG ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_ROSMIDFRN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_SUPERFRN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_SUPERPRTL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_SUPERTMP ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_SUPRAMRGNL ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_TMPPOLE ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_TRANSTMP ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))

#aseg
summary(lmer(j_adj_HV ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_AMYGDALA ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_PALLIDUM ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_THALAMUS ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_PUTAMEN ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))
summary(lmer(j_adj_CAUD ~  SEX + visitage.x + tert_ab_composite + (1|master_famid_fact), regional_df_lo_high))



library(ggseg); library(viridis)
mri_aparc <- read.csv("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/Revision_1/aparc_mri_dec26.csv")

mri_aparc_sig <- mri_aparc %>% filter (pvalue_lo_v_hi_lme < 0.05)
mri_aparc_plot_t<-ggseg(.data=mri_aparc_sig, atlas = dk, colour="grey15", hemi = "left",mapping=aes(fill=t_value_lo_v_hi_lme ), show.legend=F) + theme_void()+  scale_fill_viridis(option="C", discrete=FALSE, limits= c(1.5, 7.9), na.value="#d3d3d3")
print(mri_aparc_plot_t)



mri_aseg <- read.csv("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/Revision_1/aseg_mri_dec26.csv")

mri_aseg_sig <- mri_aseg %>% filter (pvalue_lo_v_hi_lme < 0.05)

mri_aseg_plot_t<-ggseg(.data=mri_aseg_sig, atlas = aseg, colour="grey15", hemi = "left",mapping=aes(fill=t_value_lo_v_hi_lme ), show.legend=F) + theme_void()+  scale_fill_viridis(option="C", discrete=FALSE, limits= c(1.5, 7.9), na.value="#d3d3d3")
print(mri_aseg_plot_t)



roi_fig_2<-plot_grid(pib_aparc_plot_t,pib_aseg_plot_t, mri_aparc_plot_t,mri_aseg_plot_t, fdg_aparc_plot_t, fdg_aseg_plot_t, nrow = 3, label_size=30)

dev.new()
pdf("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/Revision_1/explor_roi_figure.pdf", width=28, height=20)
roi_fig_2
dev.off()



pib_aseg_plot_t<-ggseg(.data=pib_aseg, atlas = aseg, colour="grey15", hemi = "left",mapping=aes(fill=t_value_lo_v_hi_lme ), show.legend=T) + theme_void()+  scale_fill_viridis(option="C", discrete=FALSE, limits= c(1.5, 7.9), na.value="#d3d3d3")
print(pib_aseg_plot_t)


dev.new()
pdf("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/Revision_1/roi_legend.pdf", width=10, height=10)
pib_aseg_plot_t
dev.off()


```


# figure 3
```{r}

max_bc_data_MC <-dian_df_bc_df %>% filter(Mutation == 1 )

# classify values
max_bc_data_MC$tert_ab_composite = with(max_bc_data_MC, 
                                                                      cut(g_gsec_composite_rel_wt, 
                                                                          vTert, 
                                                                          include.lowest = T, 
                                                                          labels = c("Low", "Medium", "High")))

max_bc_data_pib <-max_bc_data_MC %>% filter(j_PIB_RSF_CORTMEAN > 0 )
max_bc_data_pib <-max_bc_data_pib %>% filter(visitage.x > 0 )
max_bc_data_pibLong <-max_bc_data_pib %>% group_by(newid15) %>%
  filter(n() > 1)

max_bc_data_hcv <-max_bc_data_MC %>% filter(j_adj_HV > 0 )
max_bc_data_hcv <-max_bc_data_hcv %>% filter(visitage.x > 0 )
max_bc_data_hcvLong <-max_bc_data_hcv %>% group_by(newid15) %>%
  filter(n() > 1)

max_bc_data_mmse <-max_bc_data_MC %>% filter(MMSE > -1 )
max_bc_data_mmse <-max_bc_data_mmse %>% filter(MMSE < 75)
max_bc_data_mmse <-max_bc_data_mmse %>% filter(visitage.x > 0 )
max_bc_data_mmseLong <-max_bc_data_mmse %>% group_by(newid15) %>%
  filter(n() > 1)



max_bc_data_MEMUNITS <-max_bc_data_MC %>% filter(MEMUNITS > -1 )
max_bc_data_MEMUNITS <-max_bc_data_MEMUNITS %>% filter(MEMUNITS < 75)
max_bc_data_MEMUNITS <-max_bc_data_MEMUNITS %>% filter(visitage.x > 0 )
max_bc_data_MEMUNITSLong <-max_bc_data_MEMUNITS %>% group_by(newid15) %>%
  filter(n() > 1)


summary(lmer(j_PIB_RSF_CORTMEAN~  earliest_age*time_wrt_base+ SEX*time_wrt_base +g_gsec_composite_rel_wt*time_wrt_base + (1+time_wrt_base|newid15)+ (1|master_famid_fact),max_bc_data_pibLong))
summary(lmer(j_adj_HV~  earliest_age*time_wrt_base+ SEX*time_wrt_base +g_gsec_composite_rel_wt*time_wrt_base + (1+time_wrt_base|newid15)+ (1|master_famid_fact),max_bc_data_hcvLong))
summary(lmer(MMSE~  earliest_age*time_wrt_base+ SEX*time_wrt_base +g_gsec_composite_rel_wt*time_wrt_base + (1+time_wrt_base|newid15)+ (1|master_famid_fact),max_bc_data_mmseLong))
summary(lmer(MEMUNITS~  earliest_age*time_wrt_base+ SEX*time_wrt_base +g_gsec_composite_rel_wt*time_wrt_base + (1+time_wrt_base|newid15)+ (1|master_famid_fact),max_bc_data_MEMUNITSLong))


#extracting slope, using time from base, and only in those with 2+ visits
#could try with putting in covariate for mutation status, or calculating separately for each group.
LMM_time_pib <-(lmer(j_PIB_RSF_CORTMEAN  ~ time_wrt_base + (1+time_wrt_base|newid15), max_bc_data_pibLong))
beta <- coef(LMM_time_pib)$newid15
beta2<-beta$time_wrt_base
extracted1 <- tibble::rownames_to_column(beta, "newid15") # Apply rownames_to_column
extracted1
max_bc_data_pibLong_rate<- merge(extracted1, max_bc_data_pibLong, by = "newid15")
max_bc_data_pibLong_rate$cortmeannpib_rate <-max_bc_data_pibLong_rate$time_wrt_base.x
summary(max_bc_data_pibLong_rate$cortmeannpib_rate) # mean slope in this sample is .06

latest <- aggregate(visitage.x~newid15, max_bc_data_pibLong_rate, FUN = "min")
max_bc_data_pibLong_rate_latest <- merge(latest, max_bc_data_pibLong_rate, by = c("newid15", "visitage.x"))
rm(latest)

summary(lmer(cortmeannpib_rate~   visitage.x +SEX + g_gsec_composite_rel_wt + (1|master_famid_fact),max_bc_data_pibLong_rate_latest))



LMM_time_hcv <-(lmer(j_adj_HV  ~ 1 +time_wrt_base + (time_wrt_base|newid15), max_bc_data_hcvLong))
beta <- coef(LMM_time_hcv)$newid15
beta2<-beta$time_wrt_base
extracted1 <- tibble::rownames_to_column(beta, "newid15") # Apply rownames_to_column
extracted1
max_bc_data_hcvLong_rate<- merge(extracted1, max_bc_data_hcvLong, by = "newid15")
max_bc_data_hcvLong_rate$HCV_rate <-max_bc_data_hcvLong_rate$time_wrt_base.x
summary(max_bc_data_hcvLong_rate$HCV_rate) # mean slope in this sample is .06


latest <- aggregate(visitage.x~newid15, max_bc_data_hcvLong_rate, FUN = "max")
max_bc_data_hcvLong_rate_latest <- merge(latest, max_bc_data_hcvLong_rate, by = c("newid15", "visitage.x"))
rm(latest)

summary(lmer(HCV_rate~   visitage.x +SEX + g_gsec_composite_rel_wt + (1|master_famid_fact),max_bc_data_hcvLong_rate_latest))


LMM_time_mmse <-(lmer(MMSE  ~ 1 +time_wrt_base + (time_wrt_base|newid15), max_bc_data_mmseLong))
beta <- coef(LMM_time_mmse)$newid15
beta2<-beta$time_wrt_base
extracted1 <- tibble::rownames_to_column(beta, "newid15") # Apply rownames_to_column
extracted1
max_bc_data_mmseLong_rate<- merge(extracted1, max_bc_data_mmseLong, by = "newid15")
max_bc_data_mmseLong_rate$mmse_rate <-max_bc_data_mmseLong_rate$time_wrt_base.x
summary(max_bc_data_mmseLong_rate$mmse_rate) # mean slope in this sample is .06


latest <- aggregate(visitage.x~newid15, max_bc_data_mmseLong_rate, FUN = "max")
max_bc_data_mmseLong_rate_latest <- merge(latest, max_bc_data_mmseLong_rate, by = c("newid15", "visitage.x"))
rm(latest)

summary(lmer(mmse_rate~   visitage.x +SEX + g_gsec_composite_rel_wt + (1|master_famid_fact),max_bc_data_mmseLong_rate_latest))


LMM_time_MEMUNITS <-(lmer(MEMUNITS  ~ 1 +time_wrt_base + (time_wrt_base|newid15), max_bc_data_MEMUNITSLong))
beta <- coef(LMM_time_MEMUNITS)$newid15
beta2<-beta$time_wrt_base
extracted1 <- tibble::rownames_to_column(beta, "newid15") # Apply rownames_to_column
extracted1
max_bc_data_MEMUNITSLong_rate<- merge(extracted1, max_bc_data_MEMUNITSLong, by = "newid15")
max_bc_data_MEMUNITSLong_rate$memunits_rate <-max_bc_data_MEMUNITSLong_rate$time_wrt_base.x
summary(max_bc_data_MEMUNITSLong_rate$memunits_rate) # mean slope in this sample is .06

latest <- aggregate(visitage.x~newid15, max_bc_data_MEMUNITSLong_rate, FUN = "min")
max_bc_data_MEMUNITSLong_rate_latest <- merge(latest, max_bc_data_MEMUNITSLong_rate, by = c("newid15", "visitage.x"))
rm(latest)


summary(lmer(memunits_rate~   visitage.x +SEX + EDUC_s+g_gsec_composite_rel_wt + (1|master_famid_fact),max_bc_data_MEMUNITSLong_rate_latest))




#We next assessed how longitudinal rates of change in core biomarker, clinical, and cognitive measures of interest (Figure 3A, C, E, and G) may be related to cell-based measures of -secretase activity, summarized using the GSC. Variants with lower (more pathogenic) GSCs were associated with faster increase in β-amyloid PET signal (B[SE] = -7.5e-04[3e-04], p = 0.005; Figure 3B and appendix pg 21), as well as more rapid decreases in hippocampal volume (B[SE] = 4.19[0.8], p < 0.0001; Figure 3D and appendix pg 21), MMSE (B[SE] = 0.02[0.01], p = 0.006; Figure 3F and appendix pg 21), and logical memory-delayed (B[SE] = 0.004[0.001], p = 0.0003; Figure 3H and appendix pg 21).




Formatting_Scatter = #All of this just controls the physical look of the image
  theme(axis.title = element_text( color="black", face="bold", size=20)) + 
  theme_bw() +
  theme(legend.title=element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size =25, face="bold")) +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size =20, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size =20, face="bold")) +
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent"))+  
  theme(legend.text = element_text(size=25, family="Helvetica", face="bold"))+
  theme(legend.position = c(0.15,.8))+ 
  theme(axis.text = element_text(size = 15))

cPlot=ggplot(max_bc_data_pibLong, aes(x=visitage.x, y=j_PIB_RSF_CORTMEAN, group=tert_ab_composite, col=tert_ab_composite))
line= geom_line(data=max_bc_data_pibLong, size=1.5, linetype=1, aes(x=visitage.x, y=j_PIB_RSF_CORTMEAN, group=newid15),show.legend=T, alpha=.8) #Line for carriers
Point = geom_point(size=2,aes(shape =tert_ab_composite, color=tert_ab_composite),show.legend=T, alpha=.8) #modify the point parameters
#Color=scale_color_carto_d( palette = "Earth", labels= labels)
Color = scale_color_manual( values = c("#a51626","black", "#3c9ab2"  ), labels=c('Lowest', 'Middle','Highest'))#"#E8702A"
Shape=scale_shape_manual(values=c(15, 16, 17), labels=c('Lowest', 'Middle','Highest')) #set colors and labels
Labels=labs(x = "Age", y= "PiB PET (SUVR)") #, title="testing") #Add in X and Y axis labels
legend=labs(colour=expression(bold(paste("Gamma-secretase \ncomposite tertile"))), shape = expression(bold(paste("Gamma-secretase \ncomposite tertile"))))
cx= scale_x_continuous(breaks = c(), labels = c(""))
long_putamenpib_age_tert_AB_composite=cPlot + line+ Point +Shape  + Color +legend + Labels + Formatting_Scatter+cx
print(long_putamenpib_age_tert_AB_composite)



cortmeannpib_rate_adj_age <-lm(cortmeannpib_rate ~  visitage.x + SEX, max_bc_data_pibLong_rate_latest)
max_bc_data_pibLong_rate_latest$cortmeannpib_rate_adj_age <- cortmeannpib_rate_adj_age$residuals


max_bc_data_pibLong_rate_latest$cortmeannpib_rate_adj_age <- resid(lm(max_bc_data_pibLong_rate_latest$cortmeannpib_rate ~ max_bc_data_pibLong_rate_latest$visitage.x + max_bc_data_pibLong_rate_latest$SEX, na.action = na.exclude)) + mean(max_bc_data_pibLong_rate_latest$cortmeannpib_rate, na.rm = TRUE)


library(nord)
Formatting_Scatter = #All of this just controls the physical look of the image
  theme(axis.title = element_text( color="black", face="bold", size=20)) + 
  theme_bw() +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size =20)) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size =20)) +
  theme(legend.title = element_text(size=25, family="Helvetica", face="bold"))+  
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent"))+  
  theme(legend.text = element_text(size=25, family="Helvetica", face="bold"))+
  theme(legend.position = c(0.85,.85))+ 
  theme(axis.text = element_text(size = 15))


cPlot=ggplot(max_bc_data_pibLong_rate_latest, aes(x=g_gsec_composite_rel_wt, y=cortmeannpib_rate_adj_age))
line = geom_smooth ( method = "lm", size=3,se=T, color="black")
Point = geom_point(size=3, aes( color=visitage.x,),position = position_jitter(w = 3, h = 0)) #modify the point parameters
xlab =xlab(expression(bold(paste(gamma~"-secretase activity composite (% rel to wt)"))))
ylab= ylab( expression(bold("Rate of Change in PiB PET (Cov.-adj.)")))
legend=labs(col="Baseline Age", shape = expression(bold(paste("A", beta, " composite tertile"))))
Color=scale_color_nord( palette = "lumina", discrete=FALSE,reverse=TRUE) #set colors and labels
pib_cortmean_rate_AB_composite_age=cPlot + Point+ line+Color +Shape  +ylab+xlab + Formatting_Scatter + legend
print(pib_cortmean_rate_AB_composite_age)

Formatting_Scatter = #All of this just controls the physical look of the image
  theme(axis.title = element_text( color="black", face="bold", size=20)) + 
  theme_bw() +
  theme(legend.title=element_blank()) +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size =20, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size =20, face="bold")) +
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent"))+  
  theme(legend.text = element_text(size=20, family="Helvetica", face="bold"))+
  theme(legend.position = c(0.8,.15))+ 
  theme(axis.text = element_text(size = 15))



cPlot=ggplot(max_bc_data_hcvLong, aes(x=visitage.x, y=j_adj_HV, group=tert_ab_composite, col=tert_ab_composite))
line= geom_line(data=max_bc_data_hcvLong, size=1.5, linetype=1, aes(x=visitage.x, y=j_adj_HV, group=newid15),show.legend=F, alpha=.8) #Line for carriers
Point = geom_point(size=2,aes(color=tert_ab_composite, shape=tert_ab_composite),show.legend=F, alpha=.8) #modify the point parameters
Color = scale_color_manual( values = c("#a51626","black", "#3c9ab2"  ), labels=c('Low', 'Middle','High'))#"#E8702A"
Shape=scale_shape_manual(values=c(15, 16, 17), labels=c('Low', 'Middle','High')) #set colors and labels
ylab= ylab(bquote(bold('Hippocampal Volume'~ (mm^3))))
xlab=xlab("Age")
cx= scale_x_continuous(breaks = c(), labels = c(""))
long_HCV_age_tert_AB_composite=cPlot + line+ Point +Shape  + Color  + ylab+xlab + Formatting_Scatter+cx
print(long_HCV_age_tert_AB_composite)


HCV_rate_adj_age <-lm(HCV_rate ~  visitage.x + SEX, max_bc_data_hcvLong_rate_latest)
max_bc_data_hcvLong_rate_latest$HCV_rate_adj_age <- HCV_rate_adj_age$residuals


max_bc_data_hcvLong_rate_latest$HCV_rate_adj_age <- resid(lm(max_bc_data_hcvLong_rate_latest$HCV_rate ~ max_bc_data_hcvLong_rate_latest$visitage.x + max_bc_data_hcvLong_rate_latest$SEX, na.action = na.exclude)) + mean(max_bc_data_hcvLong_rate_latest$HCV_rate, na.rm = TRUE)



Formatting_Scatter = #All of this just controls the physical look of the image
  theme(axis.title = element_text( color="black", face="bold", size=20)) + 
  theme_bw() +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size =20, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size =20, face="bold")) +
  theme(legend.title = element_text(size=15, family="Helvetica", face="bold"))+  
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent"))+  
  theme(legend.text = element_text(size=15, family="Helvetica", face="bold"))+
  theme(axis.text = element_text(size = 15))


cPlot=ggplot(max_bc_data_hcvLong_rate_latest, aes(x=g_gsec_composite_rel_wt, y=HCV_rate_adj_age), show.legend=F)
line = geom_smooth ( method = "lm", size=3,se=T, color="black")
Point = geom_point(size=3, aes(color=visitage.x), show.legend=F,position = position_jitter(w = 3, h = 0)) #modify the point parameters
xlab =xlab(expression(bold(paste(gamma~"-secretase activity composite (% rel to wt)"))))
ylab= ylab( expression(bold("Rate of Change in HV (Cov.-adj.)")))
Color=scale_color_nord( palette = "lumina", discrete=FALSE,reverse=TRUE) #set colors and labels
hcv_rate_AB_composite_age=cPlot + Point+ line + Color+ Shape + ylab+xlab + Formatting_Scatter 
print(hcv_rate_AB_composite_age)





Formatting_Scatter = #All of this just controls the physical look of the image
  theme(axis.title = element_text( color="black", face="bold", size=20)) + 
  theme_bw() +
  theme(legend.title=element_blank()) +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size =20, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size =20, face="bold")) +
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent"))+  
  theme(legend.text = element_text(size=20, family="Helvetica", face="bold"))+
  theme(legend.position = c(0.75,.15))+ 
  theme(axis.text = element_text(size = 15))

max_bc_data_mmseLong$tert_ab_composite <- as.factor(max_bc_data_mmseLong$tert_ab_composite )
cPlot=ggplot(max_bc_data_mmseLong, aes(x=visitage.x, y=MMSE, group=tert_ab_composite, col=tert_ab_composite))
line= geom_line(data=max_bc_data_mmseLong, size=1.5, linetype=1, aes(x=visitage.x, y=MMSE, group=newid15),show.legend=F, alpha=.8) #Line for carriers
Point = geom_point(size=2,aes(color=tert_ab_composite, shape =tert_ab_composite),show.legend=F, alpha=.8) #modify the point parameters
Color = scale_color_manual( values = c("#a51626","black", "#3c9ab2"  ), labels=c('Low', 'Middle','High'))#"#E8702A"
Shape=scale_shape_manual(values=c(15, 16, 17), labels=c('Low', 'Middle','High')) #set colors and labels
ylab= ylab("MMSE")
xlab=xlab("Age")
cx= scale_x_continuous(breaks = c(), labels = c(""))
long_mmse_age_tert_AB_composite=cPlot + line+ Point  +cx + Color + Shape + xlab+ylab + Formatting_Scatter
print(long_mmse_age_tert_AB_composite)

mmse_rate_adj_age <-lm(mmse_rate ~  visitage.x + SEX + EDUC_s, max_bc_data_mmseLong_rate_latest)
max_bc_data_mmseLong_rate_latest$mmse_rate_adj_age <- mmse_rate_adj_age$residuals


max_bc_data_mmseLong_rate_latest$mmse_rate_adj_age <- resid(lm(max_bc_data_mmseLong_rate_latest$mmse_rate ~ max_bc_data_mmseLong_rate_latest$visitage.x + max_bc_data_mmseLong_rate_latest$SEX, na.action = na.exclude)) + mean(max_bc_data_mmseLong_rate_latest$mmse_rate, na.rm = TRUE)



Formatting_Scatter = #All of this just controls the physical look of the image
  theme_bw() +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size =20, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size =20, face="bold"))+
  theme(legend.title = element_text(size=15, family="Helvetica", face="bold"))+  
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent"))+  
  theme(legend.text = element_text(size=15, family="Helvetica", face="bold"))+
  theme(legend.position = c(0.2,.15))+ 
  theme(axis.text = element_text(size = 15))

cPlot=ggplot(max_bc_data_mmseLong_rate_latest, aes(x=g_gsec_composite_rel_wt, y=mmse_rate_adj_age), show.legend=F)
line = geom_smooth ( method = "lm", size=3,se=T, color="black")
Point = geom_point(size=3, aes(color=visitage.x), show.legend=F,position = position_jitter(w = 3, h = 0)) #modify the point parameters
xlab =xlab(expression(bold(paste(gamma~"-secretase activity composite (% rel to wt)"))))
ylab= ylab( expression(bold("Rate of Change in MMSE (Cov.-adj.)")))
Color=scale_color_nord( palette = "lumina", discrete=FALSE,reverse=TRUE) #set colors and labels
mmse_rate_AB_composite_age=cPlot + Point+ line + Color+ Shape + ylab+xlab + Formatting_Scatter 
print(mmse_rate_AB_composite_age)



max_bc_data_MEMUNITS <-max_bc_data_MC %>% filter(MEMUNITS > -1 )
max_bc_data_MEMUNITS <-max_bc_data_MEMUNITS %>% filter(MEMUNITS < 75)
max_bc_data_MEMUNITSLong <-max_bc_data_MEMUNITS %>% group_by(newid15) %>%
  filter(n() > 1)


Formatting_Scatter = #All of this just controls the physical look of the image
  theme(axis.title = element_text( color="black", face="bold", size=20)) + 
  theme_bw() +
  theme(legend.title=element_blank()) +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size =20, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size =20, face="bold")) +
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent"))+  
  theme(legend.text = element_text(size=20, family="Helvetica", face="bold"))+
  theme(legend.position = c(0.80,.15))+ 
  theme(axis.text = element_text(size = 15))


cPlot=ggplot(max_bc_data_MEMUNITSLong, aes(x=visitage.x, y=MEMUNITS, group=tert_ab_composite, col=tert_ab_composite))
line= geom_line(data=max_bc_data_MEMUNITSLong, size=1.5, linetype=1, aes(x=visitage.x, y=MEMUNITS, group=newid15),show.legend=F, alpha=.8) #Line for carriers
Point = geom_point(size=2,aes(color=tert_ab_composite, shape=tert_ab_composite),show.legend=F, alpha=.8) #modify the point parameters
Color = scale_color_manual( values = c("#a51626","black", "#3c9ab2"  ), labels=c('Low', 'Middle','High'))#"#E8702A"
Shape=scale_shape_manual(values=c(15, 16, 17), labels=c('Low', 'Middle','High')) #set colors and labels
ylab= ylab("Logical Memory Delayed")
xlab=xlab("Age")
cx= scale_x_continuous(breaks = c(), labels = c(""))
long_memunits_age_tert_AB_composite=cPlot + line+ Point   + Color + Shape + ylab+xlab + Formatting_Scatter+cx
print(long_memunits_age_tert_AB_composite)


memunits_rate_adj_age <-lm(memunits_rate ~  visitage.x + SEX + EDUC_s, max_bc_data_MEMUNITSLong_rate_latest)
max_bc_data_MEMUNITSLong_rate_latest$memunits_rate_adj_age <- memunits_rate_adj_age$residuals


max_bc_data_MEMUNITSLong_rate_latest$memunits_rate_adj_age <- resid(lm(max_bc_data_MEMUNITSLong_rate_latest$memunits_rate ~ max_bc_data_MEMUNITSLong_rate_latest$visitage.x + max_bc_data_MEMUNITSLong_rate_latest$SEX, na.action = na.exclude)) + mean(max_bc_data_MEMUNITSLong_rate_latest$memunits_rate, na.rm = TRUE)


Formatting_Scatter = #All of this just controls the physical look of the image
  theme(axis.title = element_text( color="black", face="bold", size=20)) + 
  theme_bw() +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size =20)) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size =20)) +
  theme(legend.title = element_text(size=15, family="Helvetica", face="bold"))+  
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent"))+  
  theme(axis.text = element_text(size = 15))


cPlot=ggplot(max_bc_data_MEMUNITSLong_rate_latest, aes(x=g_gsec_composite_rel_wt, y=memunits_rate_adj_age), show.legend=F)
line = geom_smooth ( method = "lm", size=3,se=T, color="black")
Point = geom_point(size=3, aes(color=visitage.x), show.legend=F,position = position_jitter(w = 3, h = 0)) #modify the point parameters
xlab =xlab(expression(bold(paste(gamma~"-secretase activity composite (% rel to wt)"))))
ylab= ylab( expression(bold("Rate of Change in Log. Mem. (Cov.-adj.)")))
Color=scale_color_nord( palette = "lumina", discrete=FALSE,reverse=TRUE) #set colors and labels
memunits_rate_AB_composite_age=cPlot + Point+ line + Color+ Shape + ylab+xlab + Formatting_Scatter 
print(memunits_rate_AB_composite_age)





longitudinal<-plot_grid(
  long_putamenpib_age_tert_AB_composite,pib_cortmean_rate_AB_composite_age,
  long_HCV_age_tert_AB_composite, hcv_rate_AB_composite_age, long_mmse_age_tert_AB_composite, mmse_rate_AB_composite_age
  ,long_memunits_age_tert_AB_composite,memunits_rate_AB_composite_age,
  
  labels = c("A.", "B.", "C.", "D.", 
             "E.", "F.", "G.", "H.") 
  ,nrow = 4,  label_size=20, hjust=0)
dev.new()
pdf("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/revision_2/Figures/Figure_3_040624.pdf", width=25, height=30)
longitudinal
dev.off()

```


#figure 4 ac
```{r}


dian_matched_bc_data_pib_mri_mmse2_latest_MC_lo <- subset (dian_matched_bc_data_pib_mri_mmse2_latest_MC, tert_ab_composite =="Low")
dian_matched_bc_data_pib_mri_mmse2_latest_MC_med <- subset (dian_matched_bc_data_pib_mri_mmse2_latest_MC, tert_ab_composite =="Medium")
dian_matched_bc_data_pib_mri_mmse2_latest_MC_hi <- subset (dian_matched_bc_data_pib_mri_mmse2_latest_MC, tert_ab_composite =="High")
COLS = c("#0571af", "#499364","#8c7aba", "#e08726",  "#e5c218", "black")
print(tert_lo_age <- ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC_lo) + 
        geom_smooth(aes(VISITAGEc.x,z_cent_bl_HV,colour="#0571af",fill="#0571af"), method = "loess",span = 1, alpha = 0.3, lwd = 2.0, se=F) +
        geom_smooth(aes(VISITAGEc.x,z_cent_LUMIPULSE_CSF_ptau,colour="#499364",fill="#499364"), method = "loess",span = 1, alpha = 0.3, lwd = 2.0, se=F) +
        geom_smooth(aes(VISITAGEc.x,z_cent_PIB_RSF_CORTMEAN,colour="#e08726",fill="#e08726"), method = "loess",span = 1, alpha = 0.3, lwd = 2.0, se=F) +
        geom_smooth(aes(VISITAGEc.x,z_cent_LUMIPULSE_CSF_AB42,colour="#e5c218",fill="#e5c218"), method = "loess",span = 1, alpha = 0.3, lwd = 2.0, se=F) +
        geom_smooth(aes(VISITAGEc.x,z_cent_FDG_POSTMID,colour="#8c7aba",fill="#8c7aba"), method = "loess",span = 1, alpha = 0.1, lwd = 2.0, se=F) +
        geom_smooth(aes(VISITAGEc.x,z_cent_CDRSUM,colour="black",fill="black"), method = "loess",span = 1, alpha = 0.1, lwd = 2.0, se=F) +
          scale_color_manual(name="Measure", values = COLS, labels = c("Hipocampal Volume", "CSF pTau", "Glucose metabolism","AB deposition", "CSF AB42", "CDR-SB") )+
        scale_fill_manual(name="Measure", values = COLS, labels = c("Hipocampal Volume", "CSF pTau", "Glucose metabolism","AB deposition","CSF AB42", "CDR-SB"))+
        labs( x="Age", y=expression(bold(paste("Standardized Difference"))), title = expression(bold(paste("Lowest A", beta, " Composite Tertile"))))+
        theme_bw() +
        coord_cartesian(ylim = c(-3, 3)) +
        annotate("rect", xmin = 20 , xmax = 60, ymin = 1, ymax = 3, alpha = .1, fill = "grey33")+
        annotate("rect", xmin = 20 , xmax = 60, ymin = -3, ymax = -1, alpha = .1, fill = "grey33")+            
        scale_x_continuous(breaks = c(20,30,40,50,60), labels = c("20","30" , "40", "50" ,"60"))+
        scale_y_continuous(breaks = c(-3,-2,-1,0,1,2,3), labels = c("-3","-2", "-1", "0", "1" ,"2", "3"))+
        theme(axis.text = element_text(size = 20))+
        theme(plot.title = element_text(hjust = 0.5))+
        theme(title = element_text(size = 20))+
        theme(legend.title=element_text(size = 15, face="bold")) +
        theme(legend.text=element_text(size= 15, face = "bold",family="Helvetica")) +
        theme(axis.title.y = element_text(size = 25, face="bold")) + 
        theme(axis.title.x = element_text(size = 25, face="bold")) +
        theme(legend.position = "blank"))


print(tert_med_age <- ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC_med) + 
        geom_smooth(aes(VISITAGEc.x,z_cent_LUMIPULSE_CSF_AB42,colour="#e5c218",fill="#e5c218"), method = "loess",span = 1, alpha = 0.3, lwd = 2.0, se=F) +
        geom_smooth(aes(VISITAGEc.x,z_cent_FDG_POSTMID,colour="#8c7aba",fill="#8c7aba"), method = "loess",span = 1, alpha = 0.1, lwd = 2.0, se=F) +
        geom_smooth(aes(VISITAGEc.x,z_cent_CDRSUM,colour="black",fill="black"), method = "loess",span = 1, alpha = 0.1, lwd = 2.0, se=F) +
        geom_smooth(aes(VISITAGEc.x,z_cent_bl_HV,colour="#0571af",fill="#0571af"), method = "loess",span = 1, alpha = 0.3, lwd = 2.0, se=F) +
        geom_smooth(aes(VISITAGEc.x,z_cent_LUMIPULSE_CSF_ptau,colour="#499364",fill="#499364"), method = "loess",span = 1, alpha = 0.3, lwd = 2.0, se=F) +
        geom_smooth(aes(VISITAGEc.x,z_cent_PIB_RSF_CORTMEAN,colour="#e08726",fill="#e08726"), method = "loess",span = 1, alpha = 0.3, lwd = 2.0, se=F) +
         scale_color_manual(name="Measure", values = COLS, labels = c("Hipocampal Volume", "CSF pTau", "Glucose metabolism","AB deposition", "CSF AB42", "CDR-SB") )+
        scale_fill_manual(name="Measure", values = COLS, labels = c("Hipocampal Volume", "CSF pTau", "Glucose metabolism","AB deposition","CSF AB42", "CDR-SB"))+
        labs( x="Age", y=expression(bold(paste("Standardized difference"))), title = expression(bold(paste("Middle A", beta, " Composite Tertile"))))+
        theme_bw() +
        annotate("rect", xmin = 20 , xmax = 65, ymin = 1, ymax = 3, alpha = .1, fill = "grey33")+
        annotate("rect", xmin = 20 , xmax = 65, ymin = -3, ymax = -1, alpha = .1, fill = "grey33")+            
        scale_x_continuous(breaks = c(20,30,40,50,60), labels = c("20","30" , "40", "50" ,"60"))+
        scale_y_continuous(breaks = c(-3,-2,-1,0,1,2,3), labels = c("-3","-2", "-1", "0", "1" ,"2", "3"))+
        coord_cartesian(ylim = c(-3, 3)) +
        theme(plot.title = element_text(hjust = 0.5))+
        theme(title = element_text(size = 20))+
        theme(axis.text = element_text(size = 20))+
        theme(legend.title=element_text(size = 15, face="bold")) +
        theme(legend.text=element_text(size= 15, face = "bold",family="Helvetica")) +
        theme(axis.title.y = element_text(size = 25, face="bold")) + 
        theme(axis.title.x = element_text(size = 25, face="bold")) +
        theme(legend.position = "blank"))

print(tert_hi_age<-  ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC_hi) + 
        geom_smooth(aes(VISITAGEc.x,z_cent_bl_HV,colour="#0571af",fill="#0571af"), method = "loess",span = 1, alpha = 0.3, lwd = 2.0, se=F) +
        geom_smooth(aes(VISITAGEc.x,z_cent_LUMIPULSE_CSF_ptau,colour="#499364",fill="#499364"), method = "loess",span = 1, alpha = 0.3, lwd = 2.0, se=F) +
        geom_smooth(aes(VISITAGEc.x,z_cent_PIB_RSF_CORTMEAN,colour="#e08726",fill="#e08726"), method = "loess",span = 1, alpha = 0.3, lwd = 2.0, se=F) +
        geom_smooth(aes(VISITAGEc.x,z_cent_LUMIPULSE_CSF_AB42,colour="#e5c218",fill="#e5c218"), method = "loess",span = 1, alpha = 0.3, lwd = 2.0, se=F) +
        geom_smooth(aes(VISITAGEc.x,z_cent_FDG_POSTMID,colour="#8c7aba",fill="#8c7aba"), method = "loess",span = 1, alpha = 0.1, lwd = 2.0, se=F) +
        geom_smooth(aes(VISITAGEc.x,z_cent_CDRSUM,colour="black",fill="black"), method = "loess",span = 1, alpha = 0.1, lwd = 2.0, se=F) +
           scale_color_manual(name="Measure", values = COLS, labels = c("Hipocampal Volume", "CSF pTau", "Glucose metabolism","AB deposition", "CSF AB42", "CDR-SB") )+
        scale_fill_manual(name="Measure", values = COLS, labels = c("Hipocampal Volume", "CSF pTau", "Glucose metabolism","AB deposition","CSF AB42", "CDR-SB"))+
        labs( x="Age", y=expression(bold(paste("Standardized difference"))), title = expression(bold(paste("Highest A", beta, " Composite Tertile"))))+
        theme_bw() +
        coord_cartesian(ylim = c(-3, 3)) +
        annotate("rect", xmin = 20 , xmax = 70, ymin = 1, ymax = 3, alpha = .1, fill = "grey33")+
        annotate("rect", xmin = 20 , xmax = 70, ymin = -3, ymax = -1, alpha = .1, fill = "grey33")+            
        scale_x_continuous(breaks = c(20,30,40,50,60, 70), labels = c("20","30" , "40", "50" ,"60", "70"))+
        scale_y_continuous(breaks = c(-3,-2,-1,0,1,2,3), labels = c("-3","-2", "-1", "0", "1" ,"2", "3"))+
        theme(axis.text = element_text(size = 20))+
        theme(plot.title = element_text(hjust = 0.5))+
        theme(title = element_text(size = 20))+
        theme(legend.title=element_text(size = 15, face="bold")) +
        theme(legend.text=element_text(size= 15, face = "bold",family="Helvetica")) +
        theme(axis.title.y = element_text(size = 25, face="bold")) + 
        theme(axis.title.x = element_text(size = 25, face="bold")) +
        theme(legend.position = "blank"))

library(cowplot)
jackcurves_bl_abtert_combined_age<-plot_grid( tert_lo_age, tert_med_age, tert_hi_age  ,nrow = 1, label_size=30)


dev.new()
pdf("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/revision_2/Figures/jackcurves_bl_abtert_combined_age.pdf", width=40, height=10)
jackcurves_bl_abtert_combined_age
dev.off()

```


# figure 4di
```{r}

library(rstan); library(rstanarm); library(dplyr); library(psych); library(viridis); library(Hmisc); library(lme4); library(lmerTest)

splinefit=rcspline.eval(dian_matched_bc_data_pib_mri_mmse2_latest_MC$visitage.x, nk=3, norm=2, pc=FALSE, inclx=TRUE) #generate the cubic spline fit
colnames (splinefit) <- c("age_linear", "age_cubic") #add labels to the two columns
dian_matched_bc_data_pib_mri_mmse2_latest_MC = cbind(dian_matched_bc_data_pib_mri_mmse2_latest_MC, splinefit) # Add the spline fits back to the main data

stan_tert_zcdr <- stan_glm(z_cent_CDRSUM ~ g_gsec_composite_rel_wt*age_linear + g_gsec_composite_rel_wt*age_cubic, 
                          data = dian_matched_bc_data_pib_mri_mmse2_latest_MC, family = gaussian(), 
                          algorithm = "sampling", adapt_delta = 0.99, seed=123456,
                          prior = cauchy(0,2.5), prior_intercept = cauchy(0,2.5),
                          chains = 8, cores = 1,  iter = 10000, thin = 10)
# chains is how many instances are being run in parallel, iterations means how many iterations are done within each chain.
# Thin indicates how many of the interations to retain. In this case I have it set to 1/10. This was their recommended default.
summary(stan_tert_zcdr) 


parameter_estimates_rate <- rstan::extract(stan_tert_zcdr$stanfit) # Extract the model estimates from STAN
factorweights_rate <-cbind(parameter_estimates_rate$alpha, parameter_estimates_rate$beta) # This specifically pulls out the intercept (alpha) and beta (beta) parameters

age.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$visitage.x),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$visitage.x),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments
#age_cubic.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments


hv_25=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1)); #generate blank matrices for outputs for noncarriers
hv_50=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
hv_75=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
diff_log=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));

contrasts_25_hv=matrix(0,nrow = length(age.1), ncol = 6); contrasts_50_hv=matrix(0,nrow = length(age.1), ncol = 6);  contrasts_75_hv=matrix(0,nrow = length(age.1), ncol = 6) #Making a blank matrix for contrasts. ncol=6 as there are six model parameters as coded

for (j in 1:length(age.1)) 
{ # We take all of the values in the dummy matrix (age.1) and use the fit from our real data to get  the cubic term values for each dummy entry
  
  tempfit=rcspline.eval(age.1[j],attr(splinefit,'knots'),norm=2, pc=FALSE, inclx=TRUE) 
  
  #Contrast 
  contrasts = matrix(
    c(1,1,1,#intercept
      25,50,75, #cytoplasmic group
      tempfit[1,1],tempfit[1,1],tempfit[1,1],#eyo linear
      tempfit[1,2],tempfit[1,2],tempfit[1,2],#eyo cubic
      tempfit[1,1]*25,tempfit[1,1]*50,tempfit[1,1]*75, #eyolinear*CY group
      tempfit[1,2]*25,tempfit[1,2]*50,tempfit[1,2]*75),#eyo cubic*CY group
    
    nrow=3, 
    ncol=6)
  contrasts_25_hv[j,]=contrasts[1,]; contrasts_50_hv[j,]=contrasts[2,]; contrasts_75_hv[j,]=contrasts[3,] #Set up matrix of contrasts for each age point separately for carriers and noncarriers to be used below
}



for (i in 1:nrow(factorweights_rate)) {weights = factorweights_rate[i,] #for every iteration select the weights

for (j in 1:length(age.1)) # for every age value in our output what would the value be using this iterations model fits?
{
  contrasts=rbind(contrasts_25_hv[j,], contrasts_50_hv[j,], contrasts_75_hv[j,]) #select the appropriate contrast from the matrix made above
  
  weights_rate = contrasts %*% weights #Multiple beta weights by the contrast to get one value per group for this age and this iteration
  hv_25[i,j]=weights_rate[1,]; hv_50[i,j]=weights_rate[2,] ; hv_75[i,j]=weights_rate[3,]  #Write the points out into a matrix for each group
  diff_log[i,j]=weights_rate[3,]-weights_rate[1,] #difference is equal to carriers - noncarriers. This is  used to determine the actual ge of deviation 
  
}
} 

lines_HV_25=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_50=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_75=matrix(0, ncol=3, nrow=length(age.1));diff_lines=matrix(0, ncol=3, nrow=length(age.1));#generate blank matrices we will use for lines

for (i in 1:ncol(hv_25)) # For every age point (i)
{ 
  temp_hv_25=quantile(hv_25[,i], probs = c(0.05, .5, .95)); temp_hv_25=unname(temp_hv_25) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_50=quantile(hv_50[,i], probs = c(0.05, .5, .95)); temp_hv_50=unname(temp_hv_50) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_75=quantile(hv_75[,i], probs = c(0.05, .5, .95)); temp_hv_75=unname(temp_hv_75) #get the 0.05, median, and 99.5 percentile values for each age
  temp_diff=quantile(diff_log[,i], probs = c(0.05, .5, .95)); temp_diff=unname(temp_diff) #get the 0.05, median, and 99.5 percentile values for each age
  
  lines_HV_25[i,1]=temp_hv_25[1];  lines_HV_25[i,2]=temp_hv_25[2];  lines_HV_25[i,3]=temp_hv_25[3] #Write out the 99% Credible intervals and median for each EYO (i) 
  lines_HV_50[i,1]=temp_hv_50[1];  lines_HV_50[i,2]=temp_hv_50[2];  lines_HV_50[i,3]=temp_hv_50[3] # Carriers
  lines_HV_75[i,1]=temp_hv_75[1];  lines_HV_75[i,2]=temp_hv_75[2];  lines_HV_75[i,3]=temp_hv_75[3] # Carriers
  diff_lines[i,1]=temp_diff[1]; diff_lines[i,2]=temp_diff[2]; diff_lines[i,3]=temp_diff[3]; # Difference line. Where the 0.05% (lower) value is >0 the groups are different
  
  }

# The "lines" files now have three colums. Each column is a line that represents the 0.05, median (.5), and 99.5 values from the distribution
# Note that we are actually looking at the upper, median, and lower bounds of the difference distribution rather comparing where boundaries for the noncarriers and carriers touch 

# Add age term and column names to matrices for plotting
flipped_eyo_hv_quant=t(age.1); flipped_eyo_hv_quant=t(flipped_eyo_hv_quant); combined_lines_cdr_perc_rel_wt_mean=cbind(flipped_eyo_hv_quant,lines_HV_25, lines_HV_50, lines_HV_75); # Combined the matrices for noncarriers and carriers and have the age values
colnames(combined_lines_cdr_perc_rel_wt_mean)<- c("age","lower_lo_cdr", "median_lo_cdr", "upper_lo_cdr", "lower_med_cdr", "median_med_cdr", "upper_med_cdr", "lower_hi_cdr", "median_hi_cdr", "upper_hi_cdr") #combine carriers and noncarriers
combined_lines_cdr_perc_rel_wt_mean=as.data.frame(combined_lines_cdr_perc_rel_wt_mean) #Make it a data frame or none of the plotting works
diff_lines=cbind(flipped_eyo_hv_quant,diff_lines); colnames(diff_lines)<- c("age","lower", "median", "upper") #separate matrix for difference, honest this could be combined with the previous one

#980043","#88419d", "#9ebcda" 
##### Plot Images #####


##### Plot Images #####
Plot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC, aes(x=age_linear, y=z_cent_CDRSUM)) #scatter plot of the actual data
Color= scale_colour_gradient2(low = ("#980043"),mid = "#88419d",high = ("#9ebcda"), midpoint = .6)
Point = geom_point(size=3,aes(alpha=.1),show.legend=F) # change the size of the dots and have the shapes map carrier and noncarrier designations
cx= scale_x_continuous(breaks = c(30, 40, 50,60), labels = c("30","40","50", "60"))
Ribbon1=geom_ribbon(data = combined_lines_cdr_perc_rel_wt_mean, inherit.aes = FALSE, aes(x = age, ymin = lower_hi_cdr, ymax = upper_hi_cdr), fill = "#9ebcda", alpha = 0.2) # This plots upper and lower bounds
Ribbon2=geom_ribbon(data = combined_lines_cdr_perc_rel_wt_mean, inherit.aes = FALSE, aes(x = age, ymin = lower_med_cdr, ymax = upper_med_cdr), fill = "#88419d", alpha = 0.2)
Ribbon3=geom_ribbon(data = combined_lines_cdr_perc_rel_wt_mean, inherit.aes = FALSE, aes(x = age, ymin = lower_lo_cdr, ymax = upper_lo_cdr), fill = "#980043", alpha = 0.2)
Line_int = geom_line(data=combined_lines_cdr_perc_rel_wt_mean, size=2,aes(x=age.1, y=median_lo_cdr ),colour = "#980043",show.legend=F) # median line for noncarriers. Linetype=2 makes it dashed
Line_lo = geom_line(data=combined_lines_cdr_perc_rel_wt_mean, size=2, aes(x=age.1, y=median_med_cdr),show.legend=F,colour = "#88419d") # median line for carriers
Line_NC = geom_line(data=combined_lines_cdr_perc_rel_wt_mean, size=2, aes(x=age.1, y=median_hi_cdr ),show.legend=F,colour = "#9ebcda") # median line for noncarriers. Linetype=2 makes it dashed
Labels=labs(x = "Age", y= "Standardized CDR-SB ") #Add in X and Y axis labels


Formatting_Scatter = #All of this just controls the physical look of the image
  # theme(axis.title = element_text(family = "Helvetica", color="#666666", face="bold", size=30)) + 
  theme_bw() +
  theme(axis.text = element_text(size = 20), axis.title.x=element_blank()) +
  theme(legend.title=element_blank()) +
  theme(legend.text=element_text(size=15, family="Helvetica")) +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size = 25, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size = 25, face="bold")) +
  theme(legend.position = "blank" )+
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent")) #Make background of legends transparent

plot_CDRSUM_per_rel_wt_mean=Plot +Color +cx  + Line_int + Line_lo +Line_NC +Ribbon1 +Ribbon2+ Ribbon3 + Labels +Formatting_Scatter # Add all of the graph elements together
print(plot_CDRSUM_per_rel_wt_mean)


plot_CDRSUM_per_rel_wt_25_75=Plot +Color +cx  + Line_int  +Line_NC +Ribbon1 + Ribbon3 + Labels +Formatting_Scatter # Add all of the graph elements together
print(plot_CDRSUM_per_rel_wt_25_75)


#### hcv ####
stan_tert_zHV <- stan_glm(z_cent_bl_HV ~ g_gsec_composite_rel_wt*age_linear + g_gsec_composite_rel_wt*age_cubic, 
                           data = dian_matched_bc_data_pib_mri_mmse2_latest_MC, family = gaussian(), 
                           algorithm = "sampling", adapt_delta = 0.99, seed=123456,
                           prior = cauchy(0,2.5), prior_intercept = cauchy(0,2.5),
                           chains = 8, cores = 1,  iter = 10000, thin = 10)
# chains is how many instances are being run in parallel, iterations means how many iterations are done within each chain.
# Thin indicates how many of the interations to retain. In this case I have it set to 1/10. This was their recommended default.
summary(stan_tert_zHV)



parameter_estimates_rate <- rstan::extract(stan_tert_zHV$stanfit) # Extract the model estimates from STAN
factorweights_rate <-cbind(parameter_estimates_rate$alpha, parameter_estimates_rate$beta) # This specifically pulls out the intercept (alpha) and beta (beta) parameters

age.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_linear),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_linear),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments
#age_cubic.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments


hv_25=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1)); #generate blank matrices for outputs for noncarriers
hv_50=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
hv_75=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
diff_log=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));

contrasts_25_hv=matrix(0,nrow = length(age.1), ncol = 6); contrasts_50_hv=matrix(0,nrow = length(age.1), ncol = 6);  contrasts_75_hv=matrix(0,nrow = length(age.1), ncol = 6) #Making a blank matrix for contrasts. ncol=6 as there are six model parameters as coded

for (j in 1:length(age.1)) 
{ # We take all of the values in the dummy matrix (age.1) and use the fit from our real data to get  the cubic term values for each dummy entry
  
  tempfit=rcspline.eval(age.1[j],attr(splinefit,'knots'),norm=2, pc=FALSE, inclx=TRUE) 
  
  #Contrast 
  contrasts = matrix(
    c(1,1,1,#intercept
      25,50,75, #cytoplasmic group
      tempfit[1,1],tempfit[1,1],tempfit[1,1],#eyo linear
      tempfit[1,2],tempfit[1,2],tempfit[1,2],#eyo cubic
      tempfit[1,1]*25,tempfit[1,1]*50,tempfit[1,1]*75, #eyolinear*CY group
      tempfit[1,2]*25,tempfit[1,2]*50,tempfit[1,2]*75),#eyo cubic*CY group
    
    nrow=3, 
    ncol=6)
  contrasts_25_hv[j,]=contrasts[1,]; contrasts_50_hv[j,]=contrasts[2,]; contrasts_75_hv[j,]=contrasts[3,] #Set up matrix of contrasts for each age point separately for carriers and noncarriers to be used below
}



for (i in 1:nrow(factorweights_rate)) {weights = factorweights_rate[i,] #for every iteration select the weights

for (j in 1:length(age.1)) # for every age value in our output what would the value be using this iterations model fits?
{
  contrasts=rbind(contrasts_25_hv[j,], contrasts_50_hv[j,], contrasts_75_hv[j,]) #select the appropriate contrast from the matrix made above
  
  weights_rate = contrasts %*% weights #Multiple beta weights by the contrast to get one value per group for this age and this iteration
  hv_25[i,j]=weights_rate[1,]; hv_50[i,j]=weights_rate[2,] ; hv_75[i,j]=weights_rate[3,]  #Write the points out into a matrix for each group
  diff_log[i,j]=weights_rate[3,]-weights_rate[1,] #difference is equal to carriers - noncarriers. This is  used to determine the actual ge of deviation 
  
}
} 

lines_HV_25=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_50=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_75=matrix(0, ncol=3, nrow=length(age.1));diff_lines=matrix(0, ncol=3, nrow=length(age.1));#generate blank matrices we will use for lines

for (i in 1:ncol(hv_25)) # For every age point (i)
{ 
  temp_hv_25=quantile(hv_25[,i], probs = c(0.05, .5, .95)); temp_hv_25=unname(temp_hv_25) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_50=quantile(hv_50[,i], probs = c(0.05, .5, .95)); temp_hv_50=unname(temp_hv_50) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_75=quantile(hv_75[,i], probs = c(0.05, .5, .95)); temp_hv_75=unname(temp_hv_75) #get the 0.05, median, and 99.5 percentile values for each age
  temp_diff=quantile(diff_log[,i], probs = c(0.05, .5, .95)); temp_diff=unname(temp_diff) #get the 0.05, median, and 99.5 percentile values for each age
  
  lines_HV_25[i,1]=temp_hv_25[1];  lines_HV_25[i,2]=temp_hv_25[2];  lines_HV_25[i,3]=temp_hv_25[3] #Write out the 99% Credible intervals and median for each EYO (i) 
  lines_HV_50[i,1]=temp_hv_50[1];  lines_HV_50[i,2]=temp_hv_50[2];  lines_HV_50[i,3]=temp_hv_50[3] # Carriers
  lines_HV_75[i,1]=temp_hv_75[1];  lines_HV_75[i,2]=temp_hv_75[2];  lines_HV_75[i,3]=temp_hv_75[3] # Carriers
  diff_lines[i,1]=temp_diff[1]; diff_lines[i,2]=temp_diff[2]; diff_lines[i,3]=temp_diff[3]; # Difference line. Where the 0.05% (lower) value is >0 the groups are different
  
}

# The "lines" files now have three colums. Each column is a line that represents the 0.05, median (.5), and 99.5 values from the distribution
# Note that we are actually looking at the upper, median, and lower bounds of the difference distribution rather comparing where boundaries for the noncarriers and carriers touch 

# Add age term and column names to matrices for plotting
flipped_eyo_hv_quant=t(age.1); flipped_eyo_hv_quant=t(flipped_eyo_hv_quant); combined_lines_hv_per_rel_wt_mean=cbind(flipped_eyo_hv_quant,lines_HV_25, lines_HV_50, lines_HV_75); # Combined the matrices for noncarriers and carriers and have the age values
colnames(combined_lines_hv_per_rel_wt_mean)<- c("age","lower_lo_cdr", "median_lo_cdr", "upper_lo_cdr", "lower_med_cdr", "median_med_cdr", "upper_med_cdr", "lower_hi_cdr", "median_hi_cdr", "upper_hi_cdr") #combine carriers and noncarriers
combined_lines_hv_per_rel_wt_mean=as.data.frame(combined_lines_hv_per_rel_wt_mean) #Make it a data frame or none of the plotting works
diff_lines=cbind(flipped_eyo_hv_quant,diff_lines); colnames(diff_lines)<- c("age","lower", "median", "upper") #separate matrix for difference, honest this could be combined with the previous one

#980043","#88419d", "#9ebcda" 
##### Plot Images #####


##### Plot Images #####
Plot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC, aes(x=age_linear, y=z_cent_bl_HV)) #scatter plot of the actual data
Point = geom_point(size=3,aes(alpha=.1),show.legend=F) # change the size of the dots and have the shapes map carrier and noncarrier designations
Color= scale_colour_gradient2(low = ("#980043"),mid = "#88419d",high = ("#9ebcda"), midpoint = .6)
cx= scale_x_continuous(breaks = c(30, 40, 50,60), labels = c("30","40","50", "60"))
Ribbon1=geom_ribbon(data = combined_lines_hv_per_rel_wt_mean, inherit.aes = FALSE, aes(x = age, ymin = lower_hi_cdr, ymax = upper_hi_cdr), fill = "#9ebcda", alpha = 0.2) # This plots upper and lower bounds
Ribbon2=geom_ribbon(data = combined_lines_hv_per_rel_wt_mean, inherit.aes = FALSE, aes(x = age, ymin = lower_med_cdr, ymax = upper_med_cdr), fill = "#88419d", alpha = 0.2)
Ribbon3=geom_ribbon(data = combined_lines_hv_per_rel_wt_mean, inherit.aes = FALSE, aes(x = age, ymin = lower_lo_cdr, ymax = upper_lo_cdr), fill = "#980043", alpha = 0.2)
Line_int = geom_line(data=combined_lines_hv_per_rel_wt_mean, size=2,aes(x=age.1, y=median_lo_cdr ),colour = "#980043",show.legend=F) # median line for noncarriers. Linetype=2 makes it dashed
Line_lo = geom_line(data=combined_lines_hv_per_rel_wt_mean, size=2, aes(x=age.1, y=median_med_cdr),show.legend=F,colour = "#88419d") # median line for carriers
Line_NC = geom_line(data=combined_lines_hv_per_rel_wt_mean, size=2, aes(x=age.1, y=median_hi_cdr ),show.legend=F,colour = "#9ebcda") # median line for noncarriers. Linetype=2 makes it dashed
Labels=labs(x = "Age", y= "Standardized HV") #Add in X and Y axis labels


Formatting_Scatter = #All of this just controls the physical look of the image
  # theme(axis.title = element_text(family = "Helvetica", color="#666666", face="bold", size=30)) + 
  theme_bw() +
  theme(axis.text = element_text(size = 20), axis.title.x=element_blank()) +
  theme(legend.title=element_blank()) +
  theme(legend.text=element_text(size=15, family="Helvetica")) +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size = 25, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size = 25, face="bold")) +
  theme(legend.position = "blank" )+
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent")) #Make background of legends transparent

plot_HV_per_rel_wt=Plot+Color  +cx  + Line_int + Line_lo +Line_NC +Ribbon1 +Ribbon2+ Ribbon3 + Labels +Formatting_Scatter # Add all of the graph elements together
print(plot_HV_per_rel_wt)

plot_HV_per_rel_wt_25_75=Plot+Color  +cx  + Line_int  +Line_NC +Ribbon1 + Ribbon3 + Labels +Formatting_Scatter # Add all of the graph elements together
print(plot_HV_per_rel_wt_25_75)



stan_tert_zpib <- stan_glm(z_cent_PIB_RSF_CORTMEAN ~ g_gsec_composite_rel_wt*age_linear + g_gsec_composite_rel_wt*age_cubic, 
                          data = dian_matched_bc_data_pib_mri_mmse2_latest_MC, family = gaussian(), 
                          algorithm = "sampling", adapt_delta = 0.99, seed=123456,
                          prior = cauchy(0,2.5), prior_intercept = cauchy(0,2.5),
                          chains = 8, cores = 1,  iter = 10000, thin = 10)
# chains is how many instances are being run in parallel, iterations means how many iterations are done within each chain.
# Thin indicates how many of the interations to retain. In this case I have it set to 1/10. This was their recommended default.
summary(stan_tert_zpib)


#### % rel to wt #### 


parameter_estimates_rate <- rstan::extract(stan_tert_zpib$stanfit) # Extract the model estimates from STAN
factorweights_rate <-cbind(parameter_estimates_rate$alpha, parameter_estimates_rate$beta) # This specifically pulls out the intercept (alpha) and beta (beta) parameters

age.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_linear),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_linear),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments
#age_cubic.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments


hv_25=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1)); #generate blank matrices for outputs for noncarriers
hv_50=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
hv_75=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
diff_log=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));

contrasts_25_hv=matrix(0,nrow = length(age.1), ncol = 6); contrasts_50_hv=matrix(0,nrow = length(age.1), ncol = 6);  contrasts_75_hv=matrix(0,nrow = length(age.1), ncol = 6) #Making a blank matrix for contrasts. ncol=6 as there are six model parameters as coded

for (j in 1:length(age.1)) 
{ # We take all of the values in the dummy matrix (age.1) and use the fit from our real data to get  the cubic term values for each dummy entry
  
  tempfit=rcspline.eval(age.1[j],attr(splinefit,'knots'),norm=2, pc=FALSE, inclx=TRUE) 
  
  #Contrast 
  contrasts = matrix(
    c(1,1,1,#intercept
      25,50,75, #cytoplasmic group
      tempfit[1,1],tempfit[1,1],tempfit[1,1],#eyo linear
      tempfit[1,2],tempfit[1,2],tempfit[1,2],#eyo cubic
      tempfit[1,1]*25,tempfit[1,1]*50,tempfit[1,1]*75, #eyolinear*CY group
      tempfit[1,2]*25,tempfit[1,2]*50,tempfit[1,2]*75),#eyo cubic*CY group
    
    nrow=3, 
    ncol=6)
  contrasts_25_hv[j,]=contrasts[1,]; contrasts_50_hv[j,]=contrasts[2,]; contrasts_75_hv[j,]=contrasts[3,] #Set up matrix of contrasts for each age point separately for carriers and noncarriers to be used below
}



for (i in 1:nrow(factorweights_rate)) {weights = factorweights_rate[i,] #for every iteration select the weights

for (j in 1:length(age.1)) # for every age value in our output what would the value be using this iterations model fits?
{
  contrasts=rbind(contrasts_25_hv[j,], contrasts_50_hv[j,], contrasts_75_hv[j,]) #select the appropriate contrast from the matrix made above
  
  weights_rate = contrasts %*% weights #Multiple beta weights by the contrast to get one value per group for this age and this iteration
  hv_25[i,j]=weights_rate[1,]; hv_50[i,j]=weights_rate[2,] ; hv_75[i,j]=weights_rate[3,]  #Write the points out into a matrix for each group
  diff_log[i,j]=weights_rate[3,]-weights_rate[1,] #difference is equal to carriers - noncarriers. This is  used to determine the actual ge of deviation 
  
}
} 

lines_HV_25=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_50=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_75=matrix(0, ncol=3, nrow=length(age.1));diff_lines=matrix(0, ncol=3, nrow=length(age.1));#generate blank matrices we will use for lines

for (i in 1:ncol(hv_25)) # For every age point (i)
{ 
  temp_hv_25=quantile(hv_25[,i], probs = c(0.05, .5, .95)); temp_hv_25=unname(temp_hv_25) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_50=quantile(hv_50[,i], probs = c(0.05, .5, .95)); temp_hv_50=unname(temp_hv_50) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_75=quantile(hv_75[,i], probs = c(0.05, .5, .95)); temp_hv_75=unname(temp_hv_75) #get the 0.05, median, and 99.5 percentile values for each age
  temp_diff=quantile(diff_log[,i], probs = c(0.05, .5, .95)); temp_diff=unname(temp_diff) #get the 0.05, median, and 99.5 percentile values for each age
  
  lines_HV_25[i,1]=temp_hv_25[1];  lines_HV_25[i,2]=temp_hv_25[2];  lines_HV_25[i,3]=temp_hv_25[3] #Write out the 99% Credible intervals and median for each EYO (i) 
  lines_HV_50[i,1]=temp_hv_50[1];  lines_HV_50[i,2]=temp_hv_50[2];  lines_HV_50[i,3]=temp_hv_50[3] # Carriers
  lines_HV_75[i,1]=temp_hv_75[1];  lines_HV_75[i,2]=temp_hv_75[2];  lines_HV_75[i,3]=temp_hv_75[3] # Carriers
  diff_lines[i,1]=temp_diff[1]; diff_lines[i,2]=temp_diff[2]; diff_lines[i,3]=temp_diff[3]; # Difference line. Where the 0.05% (lower) value is >0 the groups are different
  
}

# Add age term and column names to matrices for plotting
flipped_eyo_hv_quant=t(age.1); flipped_eyo_hv_quant=t(flipped_eyo_hv_quant); combined_lines_pib_per_rel_wt_mean=cbind(flipped_eyo_hv_quant,lines_HV_25, lines_HV_50, lines_HV_75); # Combined the matrices for noncarriers and carriers and have the age values
colnames(combined_lines_pib_per_rel_wt_mean)<- c("age","lower_lo_cdr", "median_lo_cdr", "upper_lo_cdr", "lower_med_cdr", "median_med_cdr", "upper_med_cdr", "lower_hi_cdr", "median_hi_cdr", "upper_hi_cdr") #combine carriers and noncarriers
combined_lines_pib_per_rel_wt_mean=as.data.frame(combined_lines_pib_per_rel_wt_mean) #Make it a data frame or none of the plotting works
diff_lines=cbind(flipped_eyo_hv_quant,diff_lines); colnames(diff_lines)<- c("age","lower", "median", "upper") #separate matrix for difference, honest this could be combined with the previous one

##### Plot Images #####
Plot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC, aes(x=age_linear, y=z_cent_PIB_RSF_CORTMEAN)) #scatter plot of the actual data
Point = geom_point(size=3,aes(alpha=.1),show.legend=F) # change the size of the dots and have the shapes map carrier and noncarrier designations
Color= scale_colour_gradient2(low = ("#980043"),mid = "#88419d",high = ("#9ebcda"), midpoint = .6)
cx= scale_x_continuous(breaks = c(30, 40, 50,60), labels = c("30","40","50", "60"))
Ribbon1=geom_ribbon(data = combined_lines_pib_per_rel_wt_mean, inherit.aes = FALSE, aes(x = age, ymin = lower_hi_cdr, ymax = upper_hi_cdr), fill = "#9ebcda", alpha = 0.2) # This plots upper and lower bounds
Ribbon2=geom_ribbon(data = combined_lines_pib_per_rel_wt_mean, inherit.aes = FALSE, aes(x = age, ymin = lower_med_cdr, ymax = upper_med_cdr), fill = "#88419d", alpha = 0.2)
Ribbon3=geom_ribbon(data = combined_lines_pib_per_rel_wt_mean, inherit.aes = FALSE, aes(x = age, ymin = lower_lo_cdr, ymax = upper_lo_cdr), fill = "#980043", alpha = 0.2)
Line_int = geom_line(data=combined_lines_pib_per_rel_wt_mean, size=2,aes(x=age.1, y=median_lo_cdr ),colour = "#980043",show.legend=F) # median line for noncarriers. Linetype=2 makes it dashed
Line_lo = geom_line(data=combined_lines_pib_per_rel_wt_mean, size=2, aes(x=age.1, y=median_med_cdr),show.legend=F,colour = "#88419d") # median line for carriers
Line_NC = geom_line(data=combined_lines_pib_per_rel_wt_mean, size=2, aes(x=age.1, y=median_hi_cdr ),show.legend=F,colour = "#9ebcda") # median line for noncarriers. Linetype=2 makes it dashed
Labels=labs(x = "Age", y= "Standardized pib") #Add in X and Y axis labels


Formatting_Scatter = #All of this just controls the physical look of the image
  # theme(axis.title = element_text(family = "Helvetica", color="#666666", face="bold", size=30)) + 
  theme_bw() +
  theme(axis.text = element_text(size = 20), axis.title.x=element_blank()) +
  theme(legend.title=element_blank()) +
  theme(legend.text=element_text(size=15, family="Helvetica")) +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size = 25, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size = 25, face="bold")) +
  theme(legend.position = "blank" )+
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent")) #Make background of legends transparent

plot_pib_per_rel_wt_mean=Plot +Color +cx  + Line_int + Line_lo +Line_NC +Ribbon1 +Ribbon2+ Ribbon3 + Labels +Formatting_Scatter # Add all of the graph elements together
print(plot_pib_per_rel_wt_mean)


plot_pib_per_rel_wt_25_75=Plot +Color +cx  + Line_int  +Line_NC +Ribbon1 + Ribbon3 + Labels +Formatting_Scatter # Add all of the graph elements together
print(plot_pib_per_rel_wt_25_75)


stan_tert_zcsf42 <- stan_glm(z_cent_LUMIPULSE_CSF_AB42 ~ g_gsec_composite_rel_wt*age_linear + g_gsec_composite_rel_wt*age_cubic, 
                           data = dian_matched_bc_data_pib_mri_mmse2_latest_MC, family = gaussian(), 
                           algorithm = "sampling", adapt_delta = 0.99, seed=123456,
                           prior = cauchy(0,2.5), prior_intercept = cauchy(0,2.5),
                           chains = 8, cores = 1,  iter = 10000, thin = 10)
# chains is how many instances are being run in parallel, iterations means how many iterations are done within each chain.
# Thin indicates how many of the interations to retain. In this case I have it set to 1/10. This was their recommended default.
summary(stan_tert_zcsf42)

parameter_estimates_rate <- rstan::extract(stan_tert_zcsf42$stanfit) # Extract the model estimates from STAN
factorweights_rate <-cbind(parameter_estimates_rate$alpha, parameter_estimates_rate$beta) # This specifically pulls out the intercept (alpha) and beta (beta) parameters

age.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$visitage.x),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$visitage.x),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments
#age_cubic.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments


hv_25=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1)); #generate blank matrices for outputs for noncarriers
hv_50=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
hv_75=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));

contrasts_25_hv=matrix(0,nrow = length(age.1), ncol = 6); contrasts_50_hv=matrix(0,nrow = length(age.1), ncol = 6);  contrasts_75_hv=matrix(0,nrow = length(age.1), ncol = 6) #Making a blank matrix for contrasts. ncol=6 as there are six model parameters as coded

for (j in 1:length(age.1)) 
{ # We take all of the values in the dummy matrix (age.1) and use the fit from our real data to get  the cubic term values for each dummy entry
  
  tempfit=rcspline.eval(age.1[j],attr(splinefit,'knots'),norm=2, pc=FALSE, inclx=TRUE) 
  
  #Contrast 
  contrasts = matrix(
    c(1,1,1,#intercept
      25,50,75, #cytoplasmic group
      tempfit[1,1],tempfit[1,1],tempfit[1,1],#eyo linear
      tempfit[1,2],tempfit[1,2],tempfit[1,2],#eyo cubic
      tempfit[1,1]*25,tempfit[1,1]*50,tempfit[1,1]*75, #eyolinear*CY group
      tempfit[1,2]*25,tempfit[1,2]*50,tempfit[1,2]*75),#eyo cubic*CY group   
    nrow=3, 
    ncol=6)
  contrasts_25_hv[j,]=contrasts[1,]; contrasts_50_hv[j,]=contrasts[2,]; contrasts_75_hv[j,]=contrasts[3,] #Set up matrix of contrasts for each age point separately for carriers and noncarriers to be used below
}

for (i in 1:nrow(factorweights_rate)) {weights = factorweights_rate[i,] #for every iteration select the weights

for (j in 1:length(age.1)) # for every age value in our output what would the value be using this iterations model fits?
{
  contrasts=rbind(contrasts_25_hv[j,], contrasts_50_hv[j,], contrasts_75_hv[j,]) #select the appropriate contrast from the matrix made above
  
  weights_rate = contrasts %*% weights #Multiple beta weights by the contrast to get one value per group for this age and this iteration
  hv_25[i,j]=weights_rate[1,]; hv_50[i,j]=weights_rate[2,] ; hv_75[i,j]=weights_rate[3,]  #Write the points out into a matrix for each group
  
}
} 

lines_HV_25=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_50=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_75=matrix(0, ncol=3, nrow=length(age.1));#generate blank matrices we will use for lines

for (i in 1:ncol(hv_25)) # For every age point (i)
{ 
  temp_hv_25=quantile(hv_25[,i], probs = c(0.05, .5, .95)); temp_hv_25=unname(temp_hv_25) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_50=quantile(hv_50[,i], probs = c(0.05, .5, .95)); temp_hv_50=unname(temp_hv_50) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_75=quantile(hv_75[,i], probs = c(0.05, .5, .95)); temp_hv_75=unname(temp_hv_75) #get the 0.05, median, and 99.5 percentile values for each age
  
  lines_HV_25[i,1]=temp_hv_25[1];  lines_HV_25[i,2]=temp_hv_25[2];  lines_HV_25[i,3]=temp_hv_25[3] #Write out the 99% Credible intervals and median for each EYO (i) 
  lines_HV_50[i,1]=temp_hv_50[1];  lines_HV_50[i,2]=temp_hv_50[2];  lines_HV_50[i,3]=temp_hv_50[3] # Carriers
  lines_HV_75[i,1]=temp_hv_75[1];  lines_HV_75[i,2]=temp_hv_75[2];  lines_HV_75[i,3]=temp_hv_75[3] # Carriers
}

# The "lines" files now have three colums. Each column is a line that represents the 0.05, median (.5), and 99.5 values from the distribution
# Note that we are actually looking at the upper, median, and lower bounds of the difference distribution rather comparing where boundaries for the noncarriers and carriers touch 

# Add age term and column names to matrices for plotting
flipped_eyo_hv_quant=t(age.1); flipped_eyo_hv_quant=t(flipped_eyo_hv_quant); combined_lines_csf42_per_rel_wt_mean=cbind(flipped_eyo_hv_quant,lines_HV_25, lines_HV_50, lines_HV_75); # Combined the matrices for noncarriers and carriers and have the age values
colnames(combined_lines_csf42_per_rel_wt_mean)<- c("age","lower_lo_cdr", "median_lo_cdr", "upper_lo_cdr", "lower_med_cdr", "median_med_cdr", "upper_med_cdr", "lower_hi_cdr", "median_hi_cdr", "upper_hi_cdr") #combine carriers and noncarriers
combined_lines_csf42_per_rel_wt_mean=as.data.frame(combined_lines_csf42_per_rel_wt_mean) #Make it a data frame or none of the plotting works

Plot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC, aes(x=age_linear, y=z_cent_LUMIPULSE_CSF_AB42)) #scatter plot of the actual data
Point = geom_point(size=3,aes(alpha=.1),show.legend=F) # change the size of the dots and have the shapes map carrier and noncarrier designations
Color= scale_colour_gradient2(low = ("#980043"),mid = "#88419d",high = ("#9ebcda"), midpoint = .6)
cx= scale_x_continuous(breaks = c(30, 40, 50,60), labels = c("30","40","50", "60"))
Ribbon1=geom_ribbon(data = combined_lines_csf42_per_rel_wt_mean, inherit.aes = FALSE, aes(x = age, ymin = lower_hi_cdr, ymax = upper_hi_cdr), fill = "#9ebcda", alpha = 0.2) # This plots upper and lower bounds
Ribbon2=geom_ribbon(data = combined_lines_csf42_per_rel_wt_mean, inherit.aes = FALSE, aes(x = age, ymin = lower_med_cdr, ymax = upper_med_cdr), fill = "#88419d", alpha = 0.2)
Ribbon3=geom_ribbon(data = combined_lines_csf42_per_rel_wt_mean, inherit.aes = FALSE, aes(x = age, ymin = lower_lo_cdr, ymax = upper_lo_cdr), fill = "#980043", alpha = 0.2)
Line_int = geom_line(data=combined_lines_csf42_per_rel_wt_mean, size=2,aes(x=age.1, y=median_lo_cdr ),colour = "#980043",show.legend=F) # median line for noncarriers. Linetype=2 makes it dashed
Line_lo = geom_line(data=combined_lines_csf42_per_rel_wt_mean, size=2, aes(x=age.1, y=median_med_cdr),show.legend=F,colour = "#88419d") # median line for carriers
Line_NC = geom_line(data=combined_lines_csf42_per_rel_wt_mean, size=2, aes(x=age.1, y=median_hi_cdr ),show.legend=F,colour = "#9ebcda") # median line for noncarriers. Linetype=2 makes it dashed
Labels=labs(x = "Age", y= "Standardized CSF abeta42") #Add in X and Y axis labels

Formatting_Scatter = #All of this just controls the physical look of the image
  # theme(axis.title = element_text(family = "Helvetica", color="#666666", face="bold", size=30)) + 
  theme_bw() +
  theme(axis.text = element_text(size = 20), axis.title.x=element_blank()) +
  theme(legend.title=element_blank()) +
  theme(legend.text=element_text(size=15, family="Helvetica")) +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size = 25, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size = 25, face="bold")) +
  theme(legend.position = "blank" )+
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent")) #Make background of legends transparent

plot_csf42_per_rel_wt_25_75=Plot+Color  +cx  + Line_int  +Line_NC +Ribbon1 + Ribbon3 + Labels +Formatting_Scatter # Add all of the graph elements together
print(plot_csf42_per_rel_wt_25_75)

stan_tert_zfdg <- stan_glm(z_cent_FDG_POSTMID ~ g_gsec_composite_rel_wt*age_linear + g_gsec_composite_rel_wt*age_cubic, 
                             data = dian_matched_bc_data_pib_mri_mmse2_latest_MC, family = gaussian(), 
                             algorithm = "sampling", adapt_delta = 0.99, seed=123456,
                             prior = cauchy(0,2.5), prior_intercept = cauchy(0,2.5),
                             chains = 8, cores = 1,  iter = 10000, thin = 10)
# chains is how many instances are being run in parallel, iterations means how many iterations are done within each chain.
# Thin indicates how many of the interations to retain. In this case I have it set to 1/10. This was their recommended default.
summary(stan_tert_zfdg)

parameter_estimates_rate <- rstan::extract(stan_tert_zfdg$stanfit) # Extract the model estimates from STAN
factorweights_rate <-cbind(parameter_estimates_rate$alpha, parameter_estimates_rate$beta) # This specifically pulls out the intercept (alpha) and beta (beta) parameters

age.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$visitage.x),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$visitage.x),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments
#age_cubic.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments


hv_25=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1)); #generate blank matrices for outputs for noncarriers
hv_50=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
hv_75=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
diff_log=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));

contrasts_25_hv=matrix(0,nrow = length(age.1), ncol = 6); contrasts_50_hv=matrix(0,nrow = length(age.1), ncol = 6);  contrasts_75_hv=matrix(0,nrow = length(age.1), ncol = 6) #Making a blank matrix for contrasts. ncol=6 as there are six model parameters as coded

for (j in 1:length(age.1)) 
{ # We take all of the values in the dummy matrix (age.1) and use the fit from our real data to get  the cubic term values for each dummy entry
  
  tempfit=rcspline.eval(age.1[j],attr(splinefit,'knots'),norm=2, pc=FALSE, inclx=TRUE) 
  
  #Contrast 
  contrasts = matrix(
    c(1,1,1,#intercept
      25,50,75, #cytoplasmic group
      tempfit[1,1],tempfit[1,1],tempfit[1,1],#eyo linear
      tempfit[1,2],tempfit[1,2],tempfit[1,2],#eyo cubic
      tempfit[1,1]*25,tempfit[1,1]*50,tempfit[1,1]*75, #eyolinear*CY group
      tempfit[1,2]*25,tempfit[1,2]*50,tempfit[1,2]*75),#eyo cubic*CY group
    
    nrow=3, 
    ncol=6)
  contrasts_25_hv[j,]=contrasts[1,]; contrasts_50_hv[j,]=contrasts[2,]; contrasts_75_hv[j,]=contrasts[3,] #Set up matrix of contrasts for each age point separately for carriers and noncarriers to be used below
}

for (i in 1:nrow(factorweights_rate)) {weights = factorweights_rate[i,] #for every iteration select the weights

for (j in 1:length(age.1)) # for every age value in our output what would the value be using this iterations model fits?
{
  contrasts=rbind(contrasts_25_hv[j,], contrasts_50_hv[j,], contrasts_75_hv[j,]) #select the appropriate contrast from the matrix made above
  
  weights_rate = contrasts %*% weights #Multiple beta weights by the contrast to get one value per group for this age and this iteration
  hv_25[i,j]=weights_rate[1,]; hv_50[i,j]=weights_rate[2,] ; hv_75[i,j]=weights_rate[3,]  #Write the points out into a matrix for each group
  diff_log[i,j]=weights_rate[3,]-weights_rate[1,] #difference is equal to carriers - noncarriers. This is  used to determine the actual ge of deviation 
  
}
} 

lines_HV_25=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_50=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_75=matrix(0, ncol=3, nrow=length(age.1));diff_lines=matrix(0, ncol=3, nrow=length(age.1));#generate blank matrices we will use for lines

for (i in 1:ncol(hv_25)) # For every age point (i)
{ 
  temp_hv_25=quantile(hv_25[,i], probs = c(0.05, .5, .95)); temp_hv_25=unname(temp_hv_25) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_50=quantile(hv_50[,i], probs = c(0.05, .5, .95)); temp_hv_50=unname(temp_hv_50) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_75=quantile(hv_75[,i], probs = c(0.05, .5, .95)); temp_hv_75=unname(temp_hv_75) #get the 0.05, median, and 99.5 percentile values for each age
  temp_diff=quantile(diff_log[,i], probs = c(0.05, .5, .95)); temp_diff=unname(temp_diff) #get the 0.05, median, and 99.5 percentile values for each age
  
  lines_HV_25[i,1]=temp_hv_25[1];  lines_HV_25[i,2]=temp_hv_25[2];  lines_HV_25[i,3]=temp_hv_25[3] #Write out the 99% Credible intervals and median for each EYO (i) 
  lines_HV_50[i,1]=temp_hv_50[1];  lines_HV_50[i,2]=temp_hv_50[2];  lines_HV_50[i,3]=temp_hv_50[3] # Carriers
  lines_HV_75[i,1]=temp_hv_75[1];  lines_HV_75[i,2]=temp_hv_75[2];  lines_HV_75[i,3]=temp_hv_75[3] # Carriers
  diff_lines[i,1]=temp_diff[1]; diff_lines[i,2]=temp_diff[2]; diff_lines[i,3]=temp_diff[3]; # Difference line. Where the 0.05% (lower) value is >0 the groups are different
  
}

# Add age term and column names to matrices for plotting
flipped_eyo_hv_quant=t(age.1); flipped_eyo_hv_quant=t(flipped_eyo_hv_quant); combined_lines_fdg_per_rel_wt_mean=cbind(flipped_eyo_hv_quant,lines_HV_25, lines_HV_50, lines_HV_75); # Combined the matrices for noncarriers and carriers and have the age values
colnames(combined_lines_fdg_per_rel_wt_mean)<- c("age","lower_lo_cdr", "median_lo_cdr", "upper_lo_cdr", "lower_med_cdr", "median_med_cdr", "upper_med_cdr", "lower_hi_cdr", "median_hi_cdr", "upper_hi_cdr") #combine carriers and noncarriers
combined_lines_fdg_per_rel_wt_mean=as.data.frame(combined_lines_fdg_per_rel_wt_mean) #Make it a data frame or none of the plotting works
diff_lines=cbind(flipped_eyo_hv_quant,diff_lines); colnames(diff_lines)<- c("age","lower", "median", "upper") #separate matrix for difference, honest this could be combined with the previous one

##### Plot Images #####
Plot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC, aes(x=age_linear, y=z_cent_FDG_POSTMID)) #scatter plot of the actual data
Color= scale_colour_gradient2(low = ("#980043"),mid = "#88419d",high = ("#9ebcda"), midpoint = .6)
cx= scale_x_continuous(breaks = c(30, 40, 50,60), labels = c("30","40","50", "60"))
Ribbon1=geom_ribbon(data = combined_lines_fdg_per_rel_wt_mean, inherit.aes = FALSE, aes(x = age, ymin = lower_hi_cdr, ymax = upper_hi_cdr), fill = "#9ebcda", alpha = 0.2) # This plots upper and lower bounds
Ribbon3=geom_ribbon(data = combined_lines_fdg_per_rel_wt_mean, inherit.aes = FALSE, aes(x = age, ymin = lower_lo_cdr, ymax = upper_lo_cdr), fill = "#980043", alpha = 0.2)
Line_int = geom_line(data=combined_lines_fdg_per_rel_wt_mean, size=2,aes(x=age.1, y=median_lo_cdr ),colour = "#980043",show.legend=F) # median line for noncarriers. Linetype=2 makes it dashed
Line_NC = geom_line(data=combined_lines_fdg_per_rel_wt_mean, size=2, aes(x=age.1, y=median_hi_cdr ),show.legend=F,colour = "#9ebcda") # median line for noncarriers. Linetype=2 makes it dashed
Labels=labs(x = "Age", y= "Standardized FDG-PET") #Add in X and Y axis labels


Formatting_Scatter = #All of this just controls the physical look of the image
  # theme(axis.title = element_text(family = "Helvetica", color="#666666", face="bold", size=30)) + 
  theme_bw() +
  theme(axis.text = element_text(size = 20), axis.title.x=element_blank()) +
  theme(legend.title=element_blank()) +
  theme(legend.text=element_text(size=15, family="Helvetica")) +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size = 25, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size = 25, face="bold")) +
  theme(legend.position = "blank" )+
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent")) #Make background of legends transparent

plot_fdg_per_rel_wt_25_75=Plot +Color +cx  + Line_int  +Line_NC +Ribbon1 + Ribbon3 + Labels +Formatting_Scatter # Add all of the graph elements together
print(plot_fdg_per_rel_wt_25_75)

stan_tert_zcsftau <- stan_glm(z_cent_LUMIPULSE_CSF_ptau ~ g_gsec_composite_rel_wt*age_linear + g_gsec_composite_rel_wt*age_cubic, 
                           data = dian_matched_bc_data_pib_mri_mmse2_latest_MC, family = gaussian(), 
                           algorithm = "sampling", adapt_delta = 0.99, seed=123456,
                           prior = cauchy(0,2.5), prior_intercept = cauchy(0,2.5),
                           chains = 8, cores = 1,  iter = 10000, thin = 10)
# chains is how many instances are being run in parallel, iterations means how many iterations are done within each chain.
# Thin indicates how many of the interations to retain. In this case I have it set to 1/10. This was their recommended default.
summary(stan_tert_zcsftau)

parameter_estimates_rate <- rstan::extract(stan_tert_zcsftau$stanfit) # Extract the model estimates from STAN
factorweights_rate <-cbind(parameter_estimates_rate$alpha, parameter_estimates_rate$beta) # This specifically pulls out the intercept (alpha) and beta (beta) parameters

age.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$visitage.x),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$visitage.x),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments
#age_cubic.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments

hv_25=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1)); #generate blank matrices for outputs for noncarriers
hv_50=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
hv_75=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
diff_log=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));

contrasts_25_hv=matrix(0,nrow = length(age.1), ncol = 6); contrasts_50_hv=matrix(0,nrow = length(age.1), ncol = 6);  contrasts_75_hv=matrix(0,nrow = length(age.1), ncol = 6) #Making a blank matrix for contrasts. ncol=6 as there are six model parameters as coded

for (j in 1:length(age.1)) 
{ # We take all of the values in the dummy matrix (age.1) and use the fit from our real data to get  the cubic term values for each dummy entry
  
  tempfit=rcspline.eval(age.1[j],attr(splinefit,'knots'),norm=2, pc=FALSE, inclx=TRUE) 
  
  #Contrast 
  contrasts = matrix(
    c(1,1,1,#intercept
      25,50,75, #cytoplasmic group
      tempfit[1,1],tempfit[1,1],tempfit[1,1],#eyo linear
      tempfit[1,2],tempfit[1,2],tempfit[1,2],#eyo cubic
      tempfit[1,1]*25,tempfit[1,1]*50,tempfit[1,1]*75, #eyolinear*CY group
      tempfit[1,2]*25,tempfit[1,2]*50,tempfit[1,2]*75),#eyo cubic*CY group
    
    nrow=3, 
    ncol=6)
  contrasts_25_hv[j,]=contrasts[1,]; contrasts_50_hv[j,]=contrasts[2,]; contrasts_75_hv[j,]=contrasts[3,] #Set up matrix of contrasts for each age point separately for carriers and noncarriers to be used below
}

for (i in 1:nrow(factorweights_rate)) {weights = factorweights_rate[i,] #for every iteration select the weights

for (j in 1:length(age.1)) # for every age value in our output what would the value be using this iterations model fits?
{
  contrasts=rbind(contrasts_25_hv[j,], contrasts_50_hv[j,], contrasts_75_hv[j,]) #select the appropriate contrast from the matrix made above
  
  weights_rate = contrasts %*% weights #Multiple beta weights by the contrast to get one value per group for this age and this iteration
  hv_25[i,j]=weights_rate[1,]; hv_50[i,j]=weights_rate[2,] ; hv_75[i,j]=weights_rate[3,]  #Write the points out into a matrix for each group
  diff_log[i,j]=weights_rate[3,]-weights_rate[1,] #difference is equal to carriers - noncarriers. This is  used to determine the actual ge of deviation 
  
}
} 

lines_HV_25=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_50=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_75=matrix(0, ncol=3, nrow=length(age.1));diff_lines=matrix(0, ncol=3, nrow=length(age.1));#generate blank matrices we will use for lines

for (i in 1:ncol(hv_25)) # For every age point (i)
{ 
  temp_hv_25=quantile(hv_25[,i], probs = c(0.05, .5, .95)); temp_hv_25=unname(temp_hv_25) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_50=quantile(hv_50[,i], probs = c(0.05, .5, .95)); temp_hv_50=unname(temp_hv_50) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_75=quantile(hv_75[,i], probs = c(0.05, .5, .95)); temp_hv_75=unname(temp_hv_75) #get the 0.05, median, and 99.5 percentile values for each age
  temp_diff=quantile(diff_log[,i], probs = c(0.05, .5, .95)); temp_diff=unname(temp_diff) #get the 0.05, median, and 99.5 percentile values for each age
  
  lines_HV_25[i,1]=temp_hv_25[1];  lines_HV_25[i,2]=temp_hv_25[2];  lines_HV_25[i,3]=temp_hv_25[3] #Write out the 99% Credible intervals and median for each EYO (i) 
  lines_HV_50[i,1]=temp_hv_50[1];  lines_HV_50[i,2]=temp_hv_50[2];  lines_HV_50[i,3]=temp_hv_50[3] # Carriers
  lines_HV_75[i,1]=temp_hv_75[1];  lines_HV_75[i,2]=temp_hv_75[2];  lines_HV_75[i,3]=temp_hv_75[3] # Carriers
  diff_lines[i,1]=temp_diff[1]; diff_lines[i,2]=temp_diff[2]; diff_lines[i,3]=temp_diff[3]; # Difference line. Where the 0.05% (lower) value is >0 the groups are different
  
}

# Add age term and column names to matrices for plotting
flipped_eyo_hv_quant=t(age.1); flipped_eyo_hv_quant=t(flipped_eyo_hv_quant); combined_lines_csftau_per_rel_wt_mean=cbind(flipped_eyo_hv_quant,lines_HV_25, lines_HV_50, lines_HV_75); # Combined the matrices for noncarriers and carriers and have the age values
colnames(combined_lines_csftau_per_rel_wt_mean)<- c("age","lower_lo_cdr", "median_lo_cdr", "upper_lo_cdr", "lower_med_cdr", "median_med_cdr", "upper_med_cdr", "lower_hi_cdr", "median_hi_cdr", "upper_hi_cdr") #combine carriers and noncarriers
combined_lines_csftau_per_rel_wt_mean=as.data.frame(combined_lines_csftau_per_rel_wt_mean) #Make it a data frame or none of the plotting works
diff_lines=cbind(flipped_eyo_hv_quant,diff_lines); colnames(diff_lines)<- c("age","lower", "median", "upper") #separate matrix for difference, honest this could be combined with the previous one

##### Plot Images #####
Plot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC, aes(x=age_linear, y=z_cent_LUMIPULSE_CSF_ptau)) #scatter plot of the actual data
Color= scale_colour_gradient2(low = ("#980043"),mid = "#88419d",high = ("#9ebcda"), midpoint = .6)

#Color=scale_color_manual(values=c('black', '#00798c', '#A50026'), labels=c('NC', 'PSEN1 Cytoplasmic','PSEN1 Transmembrane')) #set colors and labels
Ribbon1=geom_ribbon(data = combined_lines_csftau_per_rel_wt_mean, inherit.aes = FALSE, aes(x = age, ymin = lower_hi_cdr, ymax = upper_hi_cdr), fill = "#9ebcda", alpha = 0.2) # This plots upper and lower bounds
Ribbon3=geom_ribbon(data = combined_lines_csftau_per_rel_wt_mean, inherit.aes = FALSE, aes(x = age, ymin = lower_lo_cdr, ymax = upper_lo_cdr), fill = "#980043", alpha = 0.2)
Line_int = geom_line(data=combined_lines_csftau_per_rel_wt_mean, size=2,aes(x=age.1, y=median_lo_cdr ),colour = "#980043",show.legend=F) # median line for noncarriers. Linetype=2 makes it dashed
Line_NC = geom_line(data=combined_lines_csftau_per_rel_wt_mean, size=2, aes(x=age.1, y=median_hi_cdr ),show.legend=F,colour = "#9ebcda") # median line for noncarriers. Linetype=2 makes it dashed
Labels=labs(x = "Age", y= "Standardized CSF pTau") #Add in X and Y axis labels
#Ticks=scale_x_continuous(breaks=seq(-35,25,5), limits=c(-35,22)) #Manually set x-axis tick spacing
cx= scale_x_continuous(breaks = c(30, 40, 50,60), labels = c("30","40","50", "60"))

Formatting_Scatter = #All of this just controls the physical look of the image
  # theme(axis.title = element_text(family = "Helvetica", color="#666666", face="bold", size=30)) + 
  theme_bw() +
  theme(axis.text = element_text(size = 20), axis.title.x=element_blank()) +
  theme(legend.title=element_blank()) +
  theme(legend.text=element_text(size=15, family="Helvetica")) +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size = 25, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size = 25, face="bold")) +
  theme(legend.position = "blank" )+
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent")) #Make background of legends transparent

plot_csftau_per_rel_wt_25_75=Plot  +cx +Color + Line_int  +Line_NC +Ribbon1 + Ribbon3 + Labels +Formatting_Scatter # Add all of the graph elements together
print(plot_csftau_per_rel_wt_25_75)



abcomposite_per_rel_wt_25_75_individual_biomarkers_age<-plot_grid(
  plot_pib_per_rel_wt_25_75,plot_csf42_per_rel_wt_25_75,
  plot_HV_per_rel_wt_25_75, plot_fdg_per_rel_wt_25_75,
  plot_csftau_per_rel_wt_25_75, plot_CDRSUM_per_rel_wt_25_75,
  
  labels = c("A.", "B.", "C.", "D.", 
             "E.", "F.") ,nrow = 3,  label_size=20, hjust=0)
dev.new()
pdf("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/revision_2/Figures//abcomposite_per_rel_wt_25_75_individual_biomarkers_age.pdf", width=30, height=40)
abcomposite_per_rel_wt_25_75_individual_biomarkers_age
dev.off()






####EYO and CDRSOB ####

dian_matched_bc_data_pib_mri_mmse2_latest_MC$dian_mutpar_eyo
splinefit=rcspline.eval(dian_matched_bc_data_pib_mri_mmse2_latest_MC$dian_mutpar_eyo, nk=3, norm=2, pc=FALSE, inclx=TRUE) #generate the cubic spline fit
colnames (splinefit) <- c("eyo_linear", "eyo_cubic") #add labels to the two columns
dian_matched_bc_data_pib_mri_mmse2_latest_MC = cbind(dian_matched_bc_data_pib_mri_mmse2_latest_MC, splinefit) # Add the spline fits back to the main data

stan_tert_zcdr_eyo <- stan_glm(z_cent_CDRSUM ~ g_gsec_composite_rel_wt*eyo_linear + g_gsec_composite_rel_wt*eyo_cubic, 
                           data = dian_matched_bc_data_pib_mri_mmse2_latest_MC, family = gaussian(), 
                           algorithm = "sampling", adapt_delta = 0.99, seed=123456,
                           prior = cauchy(0,2.5), prior_intercept = cauchy(0,2.5),
                           chains = 8, cores = 1,  iter = 10000, thin = 10)
# chains is how many instances are being run in parallel, iterations means how many iterations are done within each chain.
# Thin indicates how many of the interations to retain. In this case I have it set to 1/10. This was their recommended default.




parameter_estimates_rate <- extract(stan_tert_zcdr_eyo$stanfit) # Extract the model estimates from STAN
factorweights_rate <-cbind(parameter_estimates_rate$alpha, parameter_estimates_rate$beta) # This specifically pulls out the intercept (alpha) and beta (beta) parameters

age.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$eyo_linear),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$eyo_linear),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments
#age_cubic.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments


hv_25=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1)); #generate blank matrices for outputs for noncarriers
hv_50=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
hv_75=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
diff_log=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));

contrasts_25_hv=matrix(0,nrow = length(age.1), ncol = 6); contrasts_50_hv=matrix(0,nrow = length(age.1), ncol = 6);  contrasts_75_hv=matrix(0,nrow = length(age.1), ncol = 6) #Making a blank matrix for contrasts. ncol=6 as there are six model parameters as coded

for (j in 1:length(age.1)) 
{ # We take all of the values in the dummy matrix (age.1) and use the fit from our real data to get  the cubic term values for each dummy entry
  
  tempfit=rcspline.eval(age.1[j],attr(splinefit,'knots'),norm=2, pc=FALSE, inclx=TRUE) 
  
  #Contrast 
  contrasts = matrix(
    c(1,1,1,#intercept
      25,50,75, #cytoplasmic group
      tempfit[1,1],tempfit[1,1],tempfit[1,1],#eyo linear
      tempfit[1,2],tempfit[1,2],tempfit[1,2],#eyo cubic
      tempfit[1,1]*25,tempfit[1,1]*50,tempfit[1,1]*75, #eyolinear*CY group
      tempfit[1,2]*25,tempfit[1,2]*50,tempfit[1,2]*75),#eyo cubic*CY group
    
    nrow=3, 
    ncol=6)
  contrasts_25_hv[j,]=contrasts[1,]; contrasts_50_hv[j,]=contrasts[2,]; contrasts_75_hv[j,]=contrasts[3,] #Set up matrix of contrasts for each age point separately for carriers and noncarriers to be used below
}


#for (j in 1:length(age.1)) 
#{ # We take all of the values in the dummy matrix (age.1) and use the fit from our real data to get  the cubic term values for each dummy entry
  
#  tempfit=rcspline.eval(age.1[j],attr(splinefit,'knots'),norm=2, pc=FALSE, inclx=TRUE) 
  
#  #Contrast 
#  contrasts = matrix(
 #   c(1,1,1,#intercept
 #     10,50,90, #cytoplasmic group
 #     tempfit[1,1],tempfit[1,1],tempfit[1,1],#eyo linear
  #    tempfit[1,2],tempfit[1,2],tempfit[1,2],#eyo cubic
   #   tempfit[1,1]*10,tempfit[1,1]*50,tempfit[1,1]*90, #eyolinear*CY group
    #  tempfit[1,2]*10,tempfit[1,2]*50,tempfit[1,2]*90),#eyo cubic*CY group
    
   # nrow=3, 
   # ncol=6)
#  contrasts_25_hv[j,]=contrasts[1,]; contrasts_50_hv[j,]=contrasts[2,]; contrasts_75_hv[j,]=contrasts[3,] #Set up matrix of contrasts for each age point separately for carriers and noncarriers to be used below
#}

for (i in 1:nrow(factorweights_rate)) {weights = factorweights_rate[i,] #for every iteration select the weights

for (j in 1:length(age.1)) # for every age value in our output what would the value be using this iterations model fits?
{
  contrasts=rbind(contrasts_25_hv[j,], contrasts_50_hv[j,], contrasts_75_hv[j,]) #select the appropriate contrast from the matrix made above
  
  weights_rate = contrasts %*% weights #Multiple beta weights by the contrast to get one value per group for this age and this iteration
  hv_25[i,j]=weights_rate[1,]; hv_50[i,j]=weights_rate[2,] ; hv_75[i,j]=weights_rate[3,]  #Write the points out into a matrix for each group
  diff_log[i,j]=weights_rate[3,]-weights_rate[1,] #difference is equal to carriers - noncarriers. This is  used to determine the actual ge of deviation 
  
}
} 

lines_HV_25=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_50=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_75=matrix(0, ncol=3, nrow=length(age.1));diff_lines=matrix(0, ncol=3, nrow=length(age.1));#generate blank matrices we will use for lines

for (i in 1:ncol(hv_25)) # For every age point (i)
{ 
  temp_hv_25=quantile(hv_25[,i], probs = c(0.05, .5, .95)); temp_hv_25=unname(temp_hv_25) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_50=quantile(hv_50[,i], probs = c(0.05, .5, .95)); temp_hv_50=unname(temp_hv_50) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_75=quantile(hv_75[,i], probs = c(0.05, .5, .95)); temp_hv_75=unname(temp_hv_75) #get the 0.05, median, and 99.5 percentile values for each age
  temp_diff=quantile(diff_log[,i], probs = c(0.05, .5, .95)); temp_diff=unname(temp_diff) #get the 0.05, median, and 99.5 percentile values for each age
  
  lines_HV_25[i,1]=temp_hv_25[1];  lines_HV_25[i,2]=temp_hv_25[2];  lines_HV_25[i,3]=temp_hv_25[3] #Write out the 99% Credible intervals and median for each EYO (i) 
  lines_HV_50[i,1]=temp_hv_50[1];  lines_HV_50[i,2]=temp_hv_50[2];  lines_HV_50[i,3]=temp_hv_50[3] # Carriers
  lines_HV_75[i,1]=temp_hv_75[1];  lines_HV_75[i,2]=temp_hv_75[2];  lines_HV_75[i,3]=temp_hv_75[3] # Carriers
  diff_lines[i,1]=temp_diff[1]; diff_lines[i,2]=temp_diff[2]; diff_lines[i,3]=temp_diff[3]; # Difference line. Where the 0.05% (lower) value is >0 the groups are different
  
}

# The "lines" files now have three colums. Each column is a line that represents the 0.05, median (.5), and 99.5 values from the distribution
# Note that we are actually looking at the upper, median, and lower bounds of the difference distribution rather comparing where boundaries for the noncarriers and carriers touch 

# Add age term and column names to matrices for plotting
flipped_eyo_hv_quant=t(age.1); flipped_eyo_hv_quant=t(flipped_eyo_hv_quant); combined_lines_cdr_perc_rel_wt_mean_eyo=cbind(flipped_eyo_hv_quant,lines_HV_25, lines_HV_50, lines_HV_75); # Combined the matrices for noncarriers and carriers and have the age values
colnames(combined_lines_cdr_perc_rel_wt_mean_eyo)<- c("age","lower_lo_cdr", "median_lo_cdr", "upper_lo_cdr", "lower_med_cdr", "median_med_cdr", "upper_med_cdr", "lower_hi_cdr", "median_hi_cdr", "upper_hi_cdr") #combine carriers and noncarriers
combined_lines_cdr_perc_rel_wt_mean_eyo=as.data.frame(combined_lines_cdr_perc_rel_wt_mean_eyo) #Make it a data frame or none of the plotting works
diff_lines=cbind(flipped_eyo_hv_quant,diff_lines); colnames(diff_lines)<- c("age","lower", "median", "upper") #separate matrix for difference, honest this could be combined with the previous one

#980043","#88419d", "#9ebcda" 
##### Plot Images #####


##### Plot Images #####
Plot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC, aes(x=eyo_linear, y=z_cent_CDRSUM)) #scatter plot of the actual data
Color= scale_colour_gradient2(low = ("#980043"),mid = "#88419d",high = ("#9ebcda"), midpoint = .6)
Point = geom_point(size=3,aes(alpha=.1),show.legend=F) # change the size of the dots and have the shapes map carrier and noncarrier designations
cx= scale_x_continuous(breaks = c(30, 40, 50,60), labels = c("30","40","50", "60"))
Ribbon1=geom_ribbon(data = combined_lines_cdr_perc_rel_wt_mean_eyo, inherit.aes = FALSE, aes(x = age, ymin = lower_hi_cdr, ymax = upper_hi_cdr), fill = "#9ebcda", alpha = 0.2) # This plots upper and lower bounds
Ribbon2=geom_ribbon(data = combined_lines_cdr_perc_rel_wt_mean_eyo, inherit.aes = FALSE, aes(x = age, ymin = lower_med_cdr, ymax = upper_med_cdr), fill = "#88419d", alpha = 0.2)
Ribbon3=geom_ribbon(data = combined_lines_cdr_perc_rel_wt_mean_eyo, inherit.aes = FALSE, aes(x = age, ymin = lower_lo_cdr, ymax = upper_lo_cdr), fill = "#980043", alpha = 0.2)
Line_int = geom_line(data=combined_lines_cdr_perc_rel_wt_mean_eyo, size=2,aes(x=age.1, y=median_lo_cdr ),colour = "#980043",show.legend=F) # median line for noncarriers. Linetype=2 makes it dashed
Line_lo = geom_line(data=combined_lines_cdr_perc_rel_wt_mean_eyo, size=2, aes(x=age.1, y=median_med_cdr),show.legend=F,colour = "#88419d") # median line for carriers
Line_NC = geom_line(data=combined_lines_cdr_perc_rel_wt_mean_eyo, size=2, aes(x=age.1, y=median_hi_cdr ),show.legend=F,colour = "#9ebcda") # median line for noncarriers. Linetype=2 makes it dashed
Labels=labs(x = "EYO", y= "Standardized CDR-SB ") #Add in X and Y axis labels


Formatting_Scatter = #All of this just controls the physical look of the image
  # theme(axis.title = element_text(family = "Helvetica", color="#666666", face="bold", size=30)) + 
  theme_bw() +
  theme(axis.text = element_text(size = 20), axis.title.x=element_blank()) +
  theme(legend.title=element_blank()) +
  theme(legend.text=element_text(size=15, family="Helvetica")) +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size = 25, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size = 25, face="bold")) +
  theme(legend.position = "blank" )+
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent")) #Make background of legends transparent


plot_CDRSUM_per_rel_wt_25_75_eyo=Plot +Color + Line_int  +Line_NC +Ribbon1 + Ribbon3 + Labels +Formatting_Scatter # Add all of the graph elements together
print(plot_CDRSUM_per_rel_wt_25_75_eyo)



####eyo and HV ####

dian_matched_bc_data_pib_mri_mmse2_latest_MC$abcomposite_dian_rel_wt

summary(lm(z_cent_bl_HV ~ abcomposite_dian_rel_wt*eyo_linear2 + abcomposite_dian_rel_wt*eyo_cubic2,dian_matched_bc_data_pib_mri_mmse2_latest_MC))

stan_tert_zHV <- stan_glm(z_cent_bl_HV ~ g_gsec_composite_rel_wt*eyo_linear + g_gsec_composite_rel_wt*eyo_cubic, 
                          data = dian_matched_bc_data_pib_mri_mmse2_latest_MC, family = gaussian(), 
                          algorithm = "sampling", adapt_delta = 0.99, seed=123456,
                          prior = cauchy(0,2.5), prior_intercept = cauchy(0,2.5),
                          chains = 8, cores = 1,  iter = 10000, thin = 10)
# chains is how many instances are being run in parallel, iterations means how many iterations are done within each chain.
# Thin indicates how many of the interations to retain. In this case I have it set to 1/10. This was their recommended default.
summary(stan_tert_zHV)


parameter_estimates_rate <- extract(stan_tert_zHV$stanfit) # Extract the model estimates from STAN
factorweights_rate <-cbind(parameter_estimates_rate$alpha, parameter_estimates_rate$beta) # This specifically pulls out the intercept (alpha) and beta (beta) parameters

age.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$eyo_linear),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$eyo_linear),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments
#age_cubic.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments


hv_25=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1)); #generate blank matrices for outputs for noncarriers
hv_50=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
hv_75=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
diff_log=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));

contrasts_25_hv=matrix(0,nrow = length(age.1), ncol = 6); contrasts_50_hv=matrix(0,nrow = length(age.1), ncol = 6);  contrasts_75_hv=matrix(0,nrow = length(age.1), ncol = 6) #Making a blank matrix for contrasts. ncol=6 as there are six model parameters as coded

for (j in 1:length(age.1)) 
{ # We take all of the values in the dummy matrix (age.1) and use the fit from our real data to get  the cubic term values for each dummy entry
  
  tempfit=rcspline.eval(age.1[j],attr(splinefit,'knots'),norm=2, pc=FALSE, inclx=TRUE) 
  
  #Contrast 
  contrasts = matrix(
    c(1,1,1,#intercept
      25,50,75, #cytoplasmic group
      tempfit[1,1],tempfit[1,1],tempfit[1,1],#eyo linear
      tempfit[1,2],tempfit[1,2],tempfit[1,2],#eyo cubic
      tempfit[1,1]*25,tempfit[1,1]*50,tempfit[1,1]*75, #eyolinear*CY group
      tempfit[1,2]*25,tempfit[1,2]*50,tempfit[1,2]*75),#eyo cubic*CY group
    
    nrow=3, 
    ncol=6)
  contrasts_25_hv[j,]=contrasts[1,]; contrasts_50_hv[j,]=contrasts[2,]; contrasts_75_hv[j,]=contrasts[3,] #Set up matrix of contrasts for each age point separately for carriers and noncarriers to be used below
}



for (i in 1:nrow(factorweights_rate)) {weights = factorweights_rate[i,] #for every iteration select the weights

for (j in 1:length(age.1)) # for every age value in our output what would the value be using this iterations model fits?
{
  contrasts=rbind(contrasts_25_hv[j,], contrasts_50_hv[j,], contrasts_75_hv[j,]) #select the appropriate contrast from the matrix made above
  
  weights_rate = contrasts %*% weights #Multiple beta weights by the contrast to get one value per group for this age and this iteration
  hv_25[i,j]=weights_rate[1,]; hv_50[i,j]=weights_rate[2,] ; hv_75[i,j]=weights_rate[3,]  #Write the points out into a matrix for each group
  diff_log[i,j]=weights_rate[3,]-weights_rate[1,] #difference is equal to carriers - noncarriers. This is  used to determine the actual ge of deviation 
  
}
} 

lines_HV_25=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_50=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_75=matrix(0, ncol=3, nrow=length(age.1));diff_lines=matrix(0, ncol=3, nrow=length(age.1));#generate blank matrices we will use for lines

for (i in 1:ncol(hv_25)) # For every age point (i)
{ 
  temp_hv_25=quantile(hv_25[,i], probs = c(0.05, .5, .95)); temp_hv_25=unname(temp_hv_25) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_50=quantile(hv_50[,i], probs = c(0.05, .5, .95)); temp_hv_50=unname(temp_hv_50) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_75=quantile(hv_75[,i], probs = c(0.05, .5, .95)); temp_hv_75=unname(temp_hv_75) #get the 0.05, median, and 99.5 percentile values for each age
  temp_diff=quantile(diff_log[,i], probs = c(0.05, .5, .95)); temp_diff=unname(temp_diff) #get the 0.05, median, and 99.5 percentile values for each age
  
  lines_HV_25[i,1]=temp_hv_25[1];  lines_HV_25[i,2]=temp_hv_25[2];  lines_HV_25[i,3]=temp_hv_25[3] #Write out the 99% Credible intervals and median for each EYO (i) 
  lines_HV_50[i,1]=temp_hv_50[1];  lines_HV_50[i,2]=temp_hv_50[2];  lines_HV_50[i,3]=temp_hv_50[3] # Carriers
  lines_HV_75[i,1]=temp_hv_75[1];  lines_HV_75[i,2]=temp_hv_75[2];  lines_HV_75[i,3]=temp_hv_75[3] # Carriers
  diff_lines[i,1]=temp_diff[1]; diff_lines[i,2]=temp_diff[2]; diff_lines[i,3]=temp_diff[3]; # Difference line. Where the 0.05% (lower) value is >0 the groups are different
  
}

# The "lines" files now have three colums. Each column is a line that represents the 0.05, median (.5), and 99.5 values from the distribution
# Note that we are actually looking at the upper, median, and lower bounds of the difference distribution rather comparing where boundaries for the noncarriers and carriers touch 

# Add age term and column names to matrices for plotting
flipped_eyo_hv_quant=t(age.1); flipped_eyo_hv_quant=t(flipped_eyo_hv_quant); combined_lines_hv_per_rel_wt_mean_eyo=cbind(flipped_eyo_hv_quant,lines_HV_25, lines_HV_50, lines_HV_75); # Combined the matrices for noncarriers and carriers and have the age values
colnames(combined_lines_hv_per_rel_wt_mean_eyo)<- c("age","lower_lo_cdr", "median_lo_cdr", "upper_lo_cdr", "lower_med_cdr", "median_med_cdr", "upper_med_cdr", "lower_hi_cdr", "median_hi_cdr", "upper_hi_cdr") #combine carriers and noncarriers
combined_lines_hv_per_rel_wt_mean_eyo=as.data.frame(combined_lines_hv_per_rel_wt_mean_eyo) #Make it a data frame or none of the plotting works
diff_lines=cbind(flipped_eyo_hv_quant,diff_lines); colnames(diff_lines)<- c("age","lower", "median", "upper") #separate matrix for difference, honest this could be combined with the previous one

#980043","#88419d", "#9ebcda" 
##### Plot Images #####


##### Plot Images #####
Plot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC, aes(x=eyo_linear, y=z_cent_bl_HV)) #scatter plot of the actual data
#Point = geom_point(size=5,aes(color=alpha=.7),show.legend=F) # change the size of the dots and have the shapes map carrier and noncarrier designations
Color= scale_colour_gradient2(low = ("#980043"),mid = "#88419d",high = ("#9ebcda"), midpoint = .6)
cx= scale_x_continuous(breaks = c(-30, -20, -10,0,10,20), labels = c("-30", "-20", "-10","0","10","20"))
Ribbon1=geom_ribbon(data = combined_lines_hv_per_rel_wt_mean_eyo, inherit.aes = FALSE, aes(x = age, ymin = lower_hi_cdr, ymax = upper_hi_cdr), fill = "#9ebcda", alpha = 0.2) # This plots upper and lower bounds
Ribbon2=geom_ribbon(data = combined_lines_hv_per_rel_wt_mean_eyo, inherit.aes = FALSE, aes(x = age, ymin = lower_med_cdr, ymax = upper_med_cdr), fill = "#88419d", alpha = 0.2)
Ribbon3=geom_ribbon(data = combined_lines_hv_per_rel_wt_mean_eyo, inherit.aes = FALSE, aes(x = age, ymin = lower_lo_cdr, ymax = upper_lo_cdr), fill = "#980043", alpha = 0.2)
Line_int = geom_line(data=combined_lines_hv_per_rel_wt_mean_eyo, size=2,aes(x=age.1, y=median_lo_cdr ),colour = "#980043",show.legend=F) # median line for noncarriers. Linetype=2 makes it dashed
Line_lo = geom_line(data=combined_lines_hv_per_rel_wt_mean_eyo, size=2, aes(x=age.1, y=median_med_cdr),show.legend=F,colour = "#88419d") # median line for carriers
Line_NC = geom_line(data=combined_lines_hv_per_rel_wt_mean_eyo, size=2, aes(x=age.1, y=median_hi_cdr ),show.legend=F,colour = "#9ebcda") # median line for noncarriers. Linetype=2 makes it dashed
Labels=labs(x = "EYO", y= "Standardized HV") #Add in X and Y axis labels


Formatting_Scatter = #All of this just controls the physical look of the image
  # theme(axis.title = element_text(family = "Helvetica", color="#666666", face="bold", size=30)) + 
  theme_bw() +
  theme(axis.text = element_text(size = 20), axis.title.x=element_blank()) +
  theme(legend.title=element_blank()) +
  theme(legend.text=element_text(size=15, family="Helvetica")) +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size = 25, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size = 25, face="bold")) +
  theme(legend.position = "blank" )+
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent")) #Make background of legends transparent

plot_HV_per_rel_wt=Plot+Color  +cx  + Line_int + Line_lo +Line_NC +Ribbon1 +Ribbon2+ Ribbon3 + Labels +Formatting_Scatter # Add all of the graph elements together
print(plot_HV_per_rel_wt)

plot_HV_per_rel_wt_25_75_eyo=Plot+Color  +cx  + Line_int  +Line_NC +Ribbon1 + Ribbon3 + Labels +Formatting_Scatter # Add all of the graph elements together
print(plot_HV_per_rel_wt_25_75_eyo)




dian_matched_bc_data_pib_mri_mmse2_latest_MC$dian_mutpar_eyo
splinefit=rcspline.eval(dian_matched_bc_data_pib_mri_mmse2_latest_MC$dian_mutpar_eyo, nk=3, norm=2, pc=FALSE, inclx=TRUE) #generate the cubic spline fit
colnames (splinefit) <- c("eyo_linear", "eyo_cubic") #add labels to the two columns
dian_matched_bc_data_pib_mri_mmse2_latest_MC = cbind(dian_matched_bc_data_pib_mri_mmse2_latest_MC, splinefit) # Add the spline fits back to the main data



stan_tert_zpib <- stan_glm(z_cent_PIB_RSF_CORTMEAN ~ g_gsec_composite_rel_wt*eyo_linear + g_gsec_composite_rel_wt*eyo_cubic, 
                           data = dian_matched_bc_data_pib_mri_mmse2_latest_MC, family = gaussian(), 
                           algorithm = "sampling", adapt_delta = 0.99, seed=123456,
                           prior = cauchy(0,2.5), prior_intercept = cauchy(0,2.5),
                           chains = 8, cores = 1,  iter = 10000, thin = 10)
# chains is how many instances are being run in parallel, iterations means how many iterations are done within each chain.
# Thin indicates how many of the interations to retain. In this case I have it set to 1/10. This was their recommended default.
summary(stan_tert_zpib)


parameter_estimates_rate <- extract(stan_tert_zpib$stanfit) # Extract the model estimates from STAN
factorweights_rate <-cbind(parameter_estimates_rate$alpha, parameter_estimates_rate$beta) # This specifically pulls out the intercept (alpha) and beta (beta) parameters

age.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$eyo_linear),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$eyo_linear),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments
#age_cubic.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments


hv_25=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1)); #generate blank matrices for outputs for noncarriers
hv_50=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
hv_75=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
diff_log=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));

contrasts_25_hv=matrix(0,nrow = length(age.1), ncol = 6); contrasts_50_hv=matrix(0,nrow = length(age.1), ncol = 6);  contrasts_75_hv=matrix(0,nrow = length(age.1), ncol = 6) #Making a blank matrix for contrasts. ncol=6 as there are six model parameters as coded

for (j in 1:length(age.1)) 
{ # We take all of the values in the dummy matrix (age.1) and use the fit from our real data to get  the cubic term values for each dummy entry
  
  tempfit=rcspline.eval(age.1[j],attr(splinefit,'knots'),norm=2, pc=FALSE, inclx=TRUE) 
  
  #Contrast 
  contrasts = matrix(
    c(1,1,1,#intercept
      25,50,75, #cytoplasmic group
      tempfit[1,1],tempfit[1,1],tempfit[1,1],#eyo linear
      tempfit[1,2],tempfit[1,2],tempfit[1,2],#eyo cubic
      tempfit[1,1]*25,tempfit[1,1]*50,tempfit[1,1]*75, #eyolinear*CY group
      tempfit[1,2]*25,tempfit[1,2]*50,tempfit[1,2]*75),#eyo cubic*CY group
    
    nrow=3, 
    ncol=6)
  contrasts_25_hv[j,]=contrasts[1,]; contrasts_50_hv[j,]=contrasts[2,]; contrasts_75_hv[j,]=contrasts[3,] #Set up matrix of contrasts for each age point separately for carriers and noncarriers to be used below
}



for (i in 1:nrow(factorweights_rate)) {weights = factorweights_rate[i,] #for every iteration select the weights

for (j in 1:length(age.1)) # for every age value in our output what would the value be using this iterations model fits?
{
  contrasts=rbind(contrasts_25_hv[j,], contrasts_50_hv[j,], contrasts_75_hv[j,]) #select the appropriate contrast from the matrix made above
  
  weights_rate = contrasts %*% weights #Multiple beta weights by the contrast to get one value per group for this age and this iteration
  hv_25[i,j]=weights_rate[1,]; hv_50[i,j]=weights_rate[2,] ; hv_75[i,j]=weights_rate[3,]  #Write the points out into a matrix for each group
  diff_log[i,j]=weights_rate[3,]-weights_rate[1,] #difference is equal to carriers - noncarriers. This is  used to determine the actual ge of deviation 
  
}
} 

lines_HV_25=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_50=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_75=matrix(0, ncol=3, nrow=length(age.1));diff_lines=matrix(0, ncol=3, nrow=length(age.1));#generate blank matrices we will use for lines

for (i in 1:ncol(hv_25)) # For every age point (i)
{ 
  temp_hv_25=quantile(hv_25[,i], probs = c(0.05, .5, .95)); temp_hv_25=unname(temp_hv_25) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_50=quantile(hv_50[,i], probs = c(0.05, .5, .95)); temp_hv_50=unname(temp_hv_50) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_75=quantile(hv_75[,i], probs = c(0.05, .5, .95)); temp_hv_75=unname(temp_hv_75) #get the 0.05, median, and 99.5 percentile values for each age
  temp_diff=quantile(diff_log[,i], probs = c(0.05, .5, .95)); temp_diff=unname(temp_diff) #get the 0.05, median, and 99.5 percentile values for each age
  
  lines_HV_25[i,1]=temp_hv_25[1];  lines_HV_25[i,2]=temp_hv_25[2];  lines_HV_25[i,3]=temp_hv_25[3] #Write out the 99% Credible intervals and median for each EYO (i) 
  lines_HV_50[i,1]=temp_hv_50[1];  lines_HV_50[i,2]=temp_hv_50[2];  lines_HV_50[i,3]=temp_hv_50[3] # Carriers
  lines_HV_75[i,1]=temp_hv_75[1];  lines_HV_75[i,2]=temp_hv_75[2];  lines_HV_75[i,3]=temp_hv_75[3] # Carriers
  diff_lines[i,1]=temp_diff[1]; diff_lines[i,2]=temp_diff[2]; diff_lines[i,3]=temp_diff[3]; # Difference line. Where the 0.05% (lower) value is >0 the groups are different
  
}

# Add age term and column names to matrices for plotting
flipped_eyo_hv_quant=t(age.1); flipped_eyo_hv_quant=t(flipped_eyo_hv_quant); combined_lines_pib_per_rel_wt_mean_eyo=cbind(flipped_eyo_hv_quant,lines_HV_25, lines_HV_50, lines_HV_75); # Combined the matrices for noncarriers and carriers and have the age values
colnames(combined_lines_pib_per_rel_wt_mean_eyo)<- c("age","lower_lo_cdr", "median_lo_cdr", "upper_lo_cdr", "lower_med_cdr", "median_med_cdr", "upper_med_cdr", "lower_hi_cdr", "median_hi_cdr", "upper_hi_cdr") #combine carriers and noncarriers
combined_lines_pib_per_rel_wt_mean_eyo=as.data.frame(combined_lines_pib_per_rel_wt_mean_eyo) #Make it a data frame or none of the plotting works
diff_lines=cbind(flipped_eyo_hv_quant,diff_lines); colnames(diff_lines)<- c("age","lower", "median", "upper") #separate matrix for difference, honest this could be combined with the previous one

#980043","#88419d", "#9ebcda" 
##### Plot Images #####


##### Plot Images #####
Plot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC, aes(x=eyo_linear, y=z_cent_PIB_RSF_CORTMEAN)) #scatter plot of the actual data
#Point = geom_point(size=5,aes(color=alpha=.7),show.legend=F) # change the size of the dots and have the shapes map carrier and noncarrier designations
Color= scale_colour_gradient2(low = ("#980043"),mid = "#88419d",high = ("#9ebcda"), midpoint = .6)

cx= scale_x_continuous(breaks = c(30, 40, 50,60), labels = c("30","40","50", "60"))
Ribbon1=geom_ribbon(data = combined_lines_pib_per_rel_wt_mean_eyo, inherit.aes = FALSE, aes(x = age, ymin = lower_hi_cdr, ymax = upper_hi_cdr), fill = "#9ebcda", alpha = 0.2) # This plots upper and lower bounds
Ribbon2=geom_ribbon(data = combined_lines_pib_per_rel_wt_mean_eyo, inherit.aes = FALSE, aes(x = age, ymin = lower_med_cdr, ymax = upper_med_cdr), fill = "#88419d", alpha = 0.2)
Ribbon3=geom_ribbon(data = combined_lines_pib_per_rel_wt_mean_eyo, inherit.aes = FALSE, aes(x = age, ymin = lower_lo_cdr, ymax = upper_lo_cdr), fill = "#980043", alpha = 0.2)
Line_int = geom_line(data=combined_lines_pib_per_rel_wt_mean_eyo, size=2,aes(x=age.1, y=median_lo_cdr ),colour = "#980043",show.legend=F) # median line for noncarriers. Linetype=2 makes it dashed
Line_lo = geom_line(data=combined_lines_pib_per_rel_wt_mean_eyo, size=2, aes(x=age.1, y=median_med_cdr),show.legend=F,colour = "#88419d") # median line for carriers
Line_NC = geom_line(data=combined_lines_pib_per_rel_wt_mean_eyo, size=2, aes(x=age.1, y=median_hi_cdr ),show.legend=F,colour = "#9ebcda") # median line for noncarriers. Linetype=2 makes it dashed
Labels=labs(x = "EYO", y= "Standardized pib") #Add in X and Y axis labels


Formatting_Scatter = #All of this just controls the physical look of the image
  # theme(axis.title = element_text(family = "Helvetica", color="#666666", face="bold", size=30)) + 
  theme_bw() +
  theme(axis.text = element_text(size = 20), axis.title.x=element_blank()) +
  theme(legend.title=element_blank()) +
  theme(legend.text=element_text(size=15, family="Helvetica")) +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size = 25, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size = 25, face="bold")) +
  theme(legend.position = "blank" )+
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent")) #Make background of legends transparent



plot_pib_per_rel_wt_25_75_eyo=Plot +Color  + Line_int  +Line_NC +Ribbon1 + Ribbon3 + Labels +Formatting_Scatter # Add all of the graph elements together
print(plot_pib_per_rel_wt_25_75_eyo)



Diff_Plot=ggplot(data=as.data.frame(diff_lines), aes(x=age, y=median))
Ribbon3=geom_ribbon(data = as.data.frame(diff_lines), inherit.aes = FALSE, aes(x = age, ymin = lower, ymax = upper), fill = "grey80", alpha = 0.6) # This plots upper and lower bounds
Line_Diff = geom_line(data=as.data.frame(diff_lines), size=2, aes(x=age, y=median),show.legend=F) # median line for difference
zero_line=geom_hline(yintercept=0, linetype="dashed", color = "red")
vline= geom_vline (xintercept= -12.3, linetype="dashed", size = 1)
cx= scale_x_continuous(breaks = c(-30, -20, -12.3, 0, 10), labels = c("-30", "-20", "-12.3", "0", "10"))
Labels=labs(x = "EYO", y= "25 v 75 difference in baseline pib") #Add in X and Y axis labels
rstan_lmer_diff_base_pib=Diff_Plot + Labels +cx+ Line_Diff+ Ribbon3 + zero_line +vline+ Labels  + Formatting_Scatter
print(rstan_lmer_diff_base_pib)



#####eyo and csf42 #####


dian_matched_bc_data_pib_mri_mmse2_latest_MC$dian_mutpar_eyo
splinefit=rcspline.eval(dian_matched_bc_data_pib_mri_mmse2_latest_MC$dian_mutpar_eyo, nk=3, norm=2, pc=FALSE, inclx=TRUE) #generate the cubic spline fit
colnames (splinefit) <- c("eyo_linear", "eyo_cubic") #add labels to the two columns
dian_matched_bc_data_pib_mri_mmse2_latest_MC = cbind(dian_matched_bc_data_pib_mri_mmse2_latest_MC, splinefit) # Add the spline fits back to the main data


stan_tert_zcsf42 <- stan_glm(z_cent_LUMIPULSE_CSF_AB42 ~ g_gsec_composite_rel_wt*eyo_linear + g_gsec_composite_rel_wt*eyo_cubic, 
                             data = dian_matched_bc_data_pib_mri_mmse2_latest_MC, family = gaussian(), 
                             algorithm = "sampling", adapt_delta = 0.99, seed=123456,
                             prior = cauchy(0,2.5), prior_intercept = cauchy(0,2.5),
                             chains = 8, cores = 1,  iter = 10000, thin = 10)
# chains is how many instances are being run in parallel, iterations means how many iterations are done within each chain.
# Thin indicates how many of the interations to retain. In this case I have it set to 1/10. This was their recommended default.
summary(stan_tert_zcsf42)


parameter_estimates_rate <- extract(stan_tert_zcsf42$stanfit) # Extract the model estimates from STAN
factorweights_rate <-cbind(parameter_estimates_rate$alpha, parameter_estimates_rate$beta) # This specifically pulls out the intercept (alpha) and beta (beta) parameters

age.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$eyo_linear),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$eyo_linear),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments
#age_cubic.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments


hv_25=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1)); #generate blank matrices for outputs for noncarriers
hv_50=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
hv_75=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
diff_log=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));

contrasts_25_hv=matrix(0,nrow = length(age.1), ncol = 6); contrasts_50_hv=matrix(0,nrow = length(age.1), ncol = 6);  contrasts_75_hv=matrix(0,nrow = length(age.1), ncol = 6) #Making a blank matrix for contrasts. ncol=6 as there are six model parameters as coded

for (j in 1:length(age.1)) 
{ # We take all of the values in the dummy matrix (age.1) and use the fit from our real data to get  the cubic term values for each dummy entry
  
  tempfit=rcspline.eval(age.1[j],attr(splinefit,'knots'),norm=2, pc=FALSE, inclx=TRUE) 
  
  #Contrast 
  contrasts = matrix(
    c(1,1,1,#intercept
      25,50,75, #cytoplasmic group
      tempfit[1,1],tempfit[1,1],tempfit[1,1],#eyo linear
      tempfit[1,2],tempfit[1,2],tempfit[1,2],#eyo cubic
      tempfit[1,1]*25,tempfit[1,1]*50,tempfit[1,1]*75, #eyolinear*CY group
      tempfit[1,2]*25,tempfit[1,2]*50,tempfit[1,2]*75),#eyo cubic*CY group
    
    nrow=3, 
    ncol=6)
  contrasts_25_hv[j,]=contrasts[1,]; contrasts_50_hv[j,]=contrasts[2,]; contrasts_75_hv[j,]=contrasts[3,] #Set up matrix of contrasts for each age point separately for carriers and noncarriers to be used below
}



for (i in 1:nrow(factorweights_rate)) {weights = factorweights_rate[i,] #for every iteration select the weights

for (j in 1:length(age.1)) # for every age value in our output what would the value be using this iterations model fits?
{
  contrasts=rbind(contrasts_25_hv[j,], contrasts_50_hv[j,], contrasts_75_hv[j,]) #select the appropriate contrast from the matrix made above
  
  weights_rate = contrasts %*% weights #Multiple beta weights by the contrast to get one value per group for this age and this iteration
  hv_25[i,j]=weights_rate[1,]; hv_50[i,j]=weights_rate[2,] ; hv_75[i,j]=weights_rate[3,]  #Write the points out into a matrix for each group
  diff_log[i,j]=weights_rate[3,]-weights_rate[1,] #difference is equal to carriers - noncarriers. This is  used to determine the actual ge of deviation 
  
}
} 

lines_HV_25=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_50=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_75=matrix(0, ncol=3, nrow=length(age.1));diff_lines=matrix(0, ncol=3, nrow=length(age.1));#generate blank matrices we will use for lines

for (i in 1:ncol(hv_25)) # For every age point (i)
{ 
  temp_hv_25=quantile(hv_25[,i], probs = c(0.05, .5, .95)); temp_hv_25=unname(temp_hv_25) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_50=quantile(hv_50[,i], probs = c(0.05, .5, .95)); temp_hv_50=unname(temp_hv_50) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_75=quantile(hv_75[,i], probs = c(0.05, .5, .95)); temp_hv_75=unname(temp_hv_75) #get the 0.05, median, and 99.5 percentile values for each age
  temp_diff=quantile(diff_log[,i], probs = c(0.05, .5, .95)); temp_diff=unname(temp_diff) #get the 0.05, median, and 99.5 percentile values for each age
  
  lines_HV_25[i,1]=temp_hv_25[1];  lines_HV_25[i,2]=temp_hv_25[2];  lines_HV_25[i,3]=temp_hv_25[3] #Write out the 99% Credible intervals and median for each EYO (i) 
  lines_HV_50[i,1]=temp_hv_50[1];  lines_HV_50[i,2]=temp_hv_50[2];  lines_HV_50[i,3]=temp_hv_50[3] # Carriers
  lines_HV_75[i,1]=temp_hv_75[1];  lines_HV_75[i,2]=temp_hv_75[2];  lines_HV_75[i,3]=temp_hv_75[3] # Carriers
  diff_lines[i,1]=temp_diff[1]; diff_lines[i,2]=temp_diff[2]; diff_lines[i,3]=temp_diff[3]; # Difference line. Where the 0.05% (lower) value is >0 the groups are different
  
}


# Add age term and column names to matrices for plotting
flipped_eyo_hv_quant=t(age.1); flipped_eyo_hv_quant=t(flipped_eyo_hv_quant); combined_lines_csf42_per_rel_wt_mean_eyo=cbind(flipped_eyo_hv_quant,lines_HV_25, lines_HV_50, lines_HV_75); # Combined the matrices for noncarriers and carriers and have the age values
colnames(combined_lines_csf42_per_rel_wt_mean_eyo)<- c("age","lower_lo_cdr", "median_lo_cdr", "upper_lo_cdr", "lower_med_cdr", "median_med_cdr", "upper_med_cdr", "lower_hi_cdr", "median_hi_cdr", "upper_hi_cdr") #combine carriers and noncarriers
combined_lines_csf42_per_rel_wt_mean_eyo=as.data.frame(combined_lines_csf42_per_rel_wt_mean_eyo) #Make it a data frame or none of the plotting works
diff_lines=cbind(flipped_eyo_hv_quant,diff_lines); colnames(diff_lines)<- c("age","lower", "median", "upper") #separate matrix for difference, honest this could be combined with the previous one

#980043","#88419d", "#9ebcda" 
##### Plot Images #####


Plot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC, aes(x=eyo_linear, y=z_cent_LUMIPULSE_CSF_AB42)) #scatter plot of the actual data
#Point = geom_point(size=5,aes(color=alpha=.7),show.legend=F) # change the size of the dots and have the shapes map carrier and noncarrier designations
Color= scale_colour_gradient2(low = ("#980043"),mid = "#88419d",high = ("#9ebcda"), midpoint = .6)
cx= scale_x_continuous(breaks = c(30, 40, 50,60), labels = c("30","40","50", "60"))
Ribbon1=geom_ribbon(data = combined_lines_csf42_per_rel_wt_mean_eyo, inherit.aes = FALSE, aes(x = age, ymin = lower_hi_cdr, ymax = upper_hi_cdr), fill = "#9ebcda", alpha = 0.2) # This plots upper and lower bounds
Ribbon2=geom_ribbon(data = combined_lines_csf42_per_rel_wt_mean_eyo, inherit.aes = FALSE, aes(x = age, ymin = lower_med_cdr, ymax = upper_med_cdr), fill = "#88419d", alpha = 0.2)
Ribbon3=geom_ribbon(data = combined_lines_csf42_per_rel_wt_mean_eyo, inherit.aes = FALSE, aes(x = age, ymin = lower_lo_cdr, ymax = upper_lo_cdr), fill = "#980043", alpha = 0.2)
Line_int = geom_line(data=combined_lines_csf42_per_rel_wt_mean_eyo, size=2,aes(x=age.1, y=median_lo_cdr ),colour = "#980043",show.legend=F) # median line for noncarriers. Linetype=2 makes it dashed
Line_lo = geom_line(data=combined_lines_csf42_per_rel_wt_mean_eyo, size=2, aes(x=age.1, y=median_med_cdr),show.legend=F,colour = "#88419d") # median line for carriers
Line_NC = geom_line(data=combined_lines_csf42_per_rel_wt_mean_eyo, size=2, aes(x=age.1, y=median_hi_cdr ),show.legend=F,colour = "#9ebcda") # median line for noncarriers. Linetype=2 makes it dashed
Labels=labs(x = "EYO", y= "Standardized CSF abeta42") #Add in X and Y axis labels

Formatting_Scatter = #All of this just controls the physical look of the image
  # theme(axis.title = element_text(family = "Helvetica", color="#666666", face="bold", size=30)) + 
  theme_bw() +
  theme(axis.text = element_text(size = 20), axis.title.x=element_blank()) +
  theme(legend.title=element_blank()) +
  theme(legend.text=element_text(size=15, family="Helvetica")) +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size = 25, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size = 25, face="bold")) +
  theme(legend.position = "blank" )+
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent")) #Make background of legends transparent

plot_csf42_per_rel_wt_25_75_eyo=Plot+Color  + Line_int  +Line_NC +Ribbon1 + Ribbon3 + Labels +Formatting_Scatter # Add all of the graph elements together
print(plot_csf42_per_rel_wt_25_75_eyo)


#####eyo and fdg #####

stan_tert_zfdg <- stan_glm(z_cent_FDG_POSTMID ~ g_gsec_composite_rel_wt*eyo_linear + g_gsec_composite_rel_wt*eyo_cubic, 
                           data = dian_matched_bc_data_pib_mri_mmse2_latest_MC, family = gaussian(), 
                           algorithm = "sampling", adapt_delta = 0.99, seed=123456,
                           prior = cauchy(0,2.5), prior_intercept = cauchy(0,2.5),
                           chains = 8, cores = 1,  iter = 10000, thin = 10)
# chains is how many instances are being run in parallel, iterations means how many iterations are done within each chain.
# Thin indicates how many of the interations to retain. In this case I have it set to 1/10. This was their recommended default.
summary(stan_tert_zfdg)

parameter_estimates_rate <- extract(stan_tert_zfdg$stanfit) # Extract the model estimates from STAN
factorweights_rate <-cbind(parameter_estimates_rate$alpha, parameter_estimates_rate$beta) # This specifically pulls out the intercept (alpha) and beta (beta) parameters

age.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$eyo_linear),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$eyo_linear),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments
#age_cubic.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments


hv_25=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1)); #generate blank matrices for outputs for noncarriers
hv_50=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
hv_75=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));


hv_25=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1)); #generate blank matrices for outputs for noncarriers
hv_50=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
hv_75=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
diff_log=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));

contrasts_25_hv=matrix(0,nrow = length(age.1), ncol = 6); contrasts_50_hv=matrix(0,nrow = length(age.1), ncol = 6);  contrasts_75_hv=matrix(0,nrow = length(age.1), ncol = 6) #Making a blank matrix for contrasts. ncol=6 as there are six model parameters as coded

for (j in 1:length(age.1)) 
{ # We take all of the values in the dummy matrix (age.1) and use the fit from our real data to get  the cubic term values for each dummy entry
  
  tempfit=rcspline.eval(age.1[j],attr(splinefit,'knots'),norm=2, pc=FALSE, inclx=TRUE) 
  
  #Contrast 
  contrasts = matrix(
    c(1,1,1,#intercept
      25,50,75, #cytoplasmic group
      tempfit[1,1],tempfit[1,1],tempfit[1,1],#eyo linear
      tempfit[1,2],tempfit[1,2],tempfit[1,2],#eyo cubic
      tempfit[1,1]*25,tempfit[1,1]*50,tempfit[1,1]*75, #eyolinear*CY group
      tempfit[1,2]*25,tempfit[1,2]*50,tempfit[1,2]*75),#eyo cubic*CY group
    
    nrow=3, 
    ncol=6)
  contrasts_25_hv[j,]=contrasts[1,]; contrasts_50_hv[j,]=contrasts[2,]; contrasts_75_hv[j,]=contrasts[3,] #Set up matrix of contrasts for each age point separately for carriers and noncarriers to be used below
}



for (i in 1:nrow(factorweights_rate)) {weights = factorweights_rate[i,] #for every iteration select the weights

for (j in 1:length(age.1)) # for every age value in our output what would the value be using this iterations model fits?
{
  contrasts=rbind(contrasts_25_hv[j,], contrasts_50_hv[j,], contrasts_75_hv[j,]) #select the appropriate contrast from the matrix made above
  
  weights_rate = contrasts %*% weights #Multiple beta weights by the contrast to get one value per group for this age and this iteration
  hv_25[i,j]=weights_rate[1,]; hv_50[i,j]=weights_rate[2,] ; hv_75[i,j]=weights_rate[3,]  #Write the points out into a matrix for each group
  diff_log[i,j]=weights_rate[3,]-weights_rate[1,] #difference is equal to carriers - noncarriers. This is  used to determine the actual ge of deviation 
  
}
} 

lines_HV_25=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_50=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_75=matrix(0, ncol=3, nrow=length(age.1));diff_lines=matrix(0, ncol=3, nrow=length(age.1));#generate blank matrices we will use for lines

for (i in 1:ncol(hv_25)) # For every age point (i)
{ 
  temp_hv_25=quantile(hv_25[,i], probs = c(0.05, .5, .95)); temp_hv_25=unname(temp_hv_25) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_50=quantile(hv_50[,i], probs = c(0.05, .5, .95)); temp_hv_50=unname(temp_hv_50) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_75=quantile(hv_75[,i], probs = c(0.05, .5, .95)); temp_hv_75=unname(temp_hv_75) #get the 0.05, median, and 99.5 percentile values for each age
  temp_diff=quantile(diff_log[,i], probs = c(0.05, .5, .95)); temp_diff=unname(temp_diff) #get the 0.05, median, and 99.5 percentile values for each age
  
  lines_HV_25[i,1]=temp_hv_25[1];  lines_HV_25[i,2]=temp_hv_25[2];  lines_HV_25[i,3]=temp_hv_25[3] #Write out the 99% Credible intervals and median for each EYO (i) 
  lines_HV_50[i,1]=temp_hv_50[1];  lines_HV_50[i,2]=temp_hv_50[2];  lines_HV_50[i,3]=temp_hv_50[3] # Carriers
  lines_HV_75[i,1]=temp_hv_75[1];  lines_HV_75[i,2]=temp_hv_75[2];  lines_HV_75[i,3]=temp_hv_75[3] # Carriers
  diff_lines[i,1]=temp_diff[1]; diff_lines[i,2]=temp_diff[2]; diff_lines[i,3]=temp_diff[3]; # Difference line. Where the 0.05% (lower) value is >0 the groups are different
  
}


# Add age term and column names to matrices for plotting
flipped_eyo_hv_quant=t(age.1); flipped_eyo_hv_quant=t(flipped_eyo_hv_quant); combined_lines_fdg_per_rel_wt_mean_eyo=cbind(flipped_eyo_hv_quant,lines_HV_25, lines_HV_50, lines_HV_75); # Combined the matrices for noncarriers and carriers and have the age values
colnames(combined_lines_fdg_per_rel_wt_mean_eyo)<- c("age","lower_lo_cdr", "median_lo_cdr", "upper_lo_cdr", "lower_med_cdr", "median_med_cdr", "upper_med_cdr", "lower_hi_cdr", "median_hi_cdr", "upper_hi_cdr") #combine carriers and noncarriers
combined_lines_fdg_per_rel_wt_mean_eyo=as.data.frame(combined_lines_fdg_per_rel_wt_mean_eyo) #Make it a data frame or none of the plotting works
diff_lines=cbind(flipped_eyo_hv_quant,diff_lines); colnames(diff_lines)<- c("age","lower", "median", "upper") #separate matrix for difference, honest this could be combined with the previous one

#980043","#88419d", "#9ebcda" 
##### Plot Images #####


##### Plot Images #####
Plot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC, aes(x=eyo_linear, y=z_cent_FDG_POSTMID)) #scatter plot of the actual data
#Point = geom_point(size=5,aes(color=alpha=.7),show.legend=F) # change the size of the dots and have the shapes map carrier and noncarrier designations
Color= scale_colour_gradient2(low = ("#980043"),mid = "#88419d",high = ("#9ebcda"), midpoint = .6)
cx= scale_x_continuous(breaks = c(30, 40, 50,60), labels = c("30","40","50", "60"))
Ribbon1=geom_ribbon(data = combined_lines_fdg_per_rel_wt_mean_eyo, inherit.aes = FALSE, aes(x = age, ymin = lower_hi_cdr, ymax = upper_hi_cdr), fill = "#9ebcda", alpha = 0.2) # This plots upper and lower bounds
Ribbon2=geom_ribbon(data = combined_lines_fdg_per_rel_wt_mean_eyo, inherit.aes = FALSE, aes(x = age, ymin = lower_med_cdr, ymax = upper_med_cdr), fill = "#88419d", alpha = 0.2)
Ribbon3=geom_ribbon(data = combined_lines_fdg_per_rel_wt_mean_eyo, inherit.aes = FALSE, aes(x = age, ymin = lower_lo_cdr, ymax = upper_lo_cdr), fill = "#980043", alpha = 0.2)
Line_int = geom_line(data=combined_lines_fdg_per_rel_wt_mean_eyo, size=2,aes(x=age.1, y=median_lo_cdr ),colour = "#980043",show.legend=F) # median line for noncarriers. Linetype=2 makes it dashed
Line_lo = geom_line(data=combined_lines_fdg_per_rel_wt_mean_eyo, size=2, aes(x=age.1, y=median_med_cdr),show.legend=F,colour = "#88419d") # median line for carriers
Line_NC = geom_line(data=combined_lines_fdg_per_rel_wt_mean_eyo, size=2, aes(x=age.1, y=median_hi_cdr ),show.legend=F,colour = "#9ebcda") # median line for noncarriers. Linetype=2 makes it dashed
Labels=labs(x = "EYO", y= "Standardized FDG-PET") #Add in X and Y axis labels


Formatting_Scatter = #All of this just controls the physical look of the image
  # theme(axis.title = element_text(family = "Helvetica", color="#666666", face="bold", size=30)) + 
  theme_bw() +
  theme(axis.text = element_text(size = 20), axis.title.x=element_blank()) +
  theme(legend.title=element_blank()) +
  theme(legend.text=element_text(size=15, family="Helvetica")) +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size = 25, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size = 25, face="bold")) +
  theme(legend.position = "blank" )+
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent")) #Make background of legends transparent

plot_fdg_per_rel_wt_25_75_eyo=Plot +Color  + Line_int  +Line_NC +Ribbon1 + Ribbon3 + Labels +Formatting_Scatter # Add all of the graph elements together
print(plot_fdg_per_rel_wt_25_75_eyo)


stan_tert_zcsftau <- stan_glm(z_cent_LUMIPULSE_CSF_ptau ~ g_gsec_composite_rel_wt*eyo_linear + g_gsec_composite_rel_wt*eyo_cubic, 
                              data = dian_matched_bc_data_pib_mri_mmse2_latest_MC, family = gaussian(), 
                              algorithm = "sampling", adapt_delta = 0.99, seed=123456,
                              prior = cauchy(0,2.5), prior_intercept = cauchy(0,2.5),
                              chains = 8, cores = 1,  iter = 10000, thin = 10)
# chains is how many instances are being run in parallel, iterations means how many iterations are done within each chain.
# Thin indicates how many of the interations to retain. In this case I have it set to 1/10. This was their recommended default.
summary(stan_tert_zcsftau)


parameter_estimates_rate <- extract(stan_tert_zcsftau$stanfit) # Extract the model estimates from STAN
factorweights_rate <-cbind(parameter_estimates_rate$alpha, parameter_estimates_rate$beta) # This specifically pulls out the intercept (alpha) and beta (beta) parameters

age.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$eyo_linear),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$eyo_linear),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments
#age_cubic.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments


hv_25=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1)); #generate blank matrices for outputs for noncarriers
hv_50=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
hv_75=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
diff_log=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));

contrasts_25_hv=matrix(0,nrow = length(age.1), ncol = 6); contrasts_50_hv=matrix(0,nrow = length(age.1), ncol = 6);  contrasts_75_hv=matrix(0,nrow = length(age.1), ncol = 6) #Making a blank matrix for contrasts. ncol=6 as there are six model parameters as coded

for (j in 1:length(age.1)) 
{ # We take all of the values in the dummy matrix (age.1) and use the fit from our real data to get  the cubic term values for each dummy entry
  
  tempfit=rcspline.eval(age.1[j],attr(splinefit,'knots'),norm=2, pc=FALSE, inclx=TRUE) 
  
  #Contrast 
  contrasts = matrix(
    c(1,1,1,#intercept
      25,50,75, #cytoplasmic group
      tempfit[1,1],tempfit[1,1],tempfit[1,1],#eyo linear
      tempfit[1,2],tempfit[1,2],tempfit[1,2],#eyo cubic
      tempfit[1,1]*25,tempfit[1,1]*50,tempfit[1,1]*75, #eyolinear*CY group
      tempfit[1,2]*25,tempfit[1,2]*50,tempfit[1,2]*75),#eyo cubic*CY group
    
    nrow=3, 
    ncol=6)
  contrasts_25_hv[j,]=contrasts[1,]; contrasts_50_hv[j,]=contrasts[2,]; contrasts_75_hv[j,]=contrasts[3,] #Set up matrix of contrasts for each age point separately for carriers and noncarriers to be used below
}



for (i in 1:nrow(factorweights_rate)) {weights = factorweights_rate[i,] #for every iteration select the weights

for (j in 1:length(age.1)) # for every age value in our output what would the value be using this iterations model fits?
{
  contrasts=rbind(contrasts_25_hv[j,], contrasts_50_hv[j,], contrasts_75_hv[j,]) #select the appropriate contrast from the matrix made above
  
  weights_rate = contrasts %*% weights #Multiple beta weights by the contrast to get one value per group for this age and this iteration
  hv_25[i,j]=weights_rate[1,]; hv_50[i,j]=weights_rate[2,] ; hv_75[i,j]=weights_rate[3,]  #Write the points out into a matrix for each group
  diff_log[i,j]=weights_rate[3,]-weights_rate[1,] #difference is equal to carriers - noncarriers. This is  used to determine the actual ge of deviation 
  
}
} 

lines_HV_25=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_50=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_75=matrix(0, ncol=3, nrow=length(age.1));diff_lines=matrix(0, ncol=3, nrow=length(age.1));#generate blank matrices we will use for lines

for (i in 1:ncol(hv_25)) # For every age point (i)
{ 
  temp_hv_25=quantile(hv_25[,i], probs = c(0.05, .5, .95)); temp_hv_25=unname(temp_hv_25) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_50=quantile(hv_50[,i], probs = c(0.05, .5, .95)); temp_hv_50=unname(temp_hv_50) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_75=quantile(hv_75[,i], probs = c(0.05, .5, .95)); temp_hv_75=unname(temp_hv_75) #get the 0.05, median, and 99.5 percentile values for each age
  temp_diff=quantile(diff_log[,i], probs = c(0.05, .5, .95)); temp_diff=unname(temp_diff) #get the 0.05, median, and 99.5 percentile values for each age
  
  lines_HV_25[i,1]=temp_hv_25[1];  lines_HV_25[i,2]=temp_hv_25[2];  lines_HV_25[i,3]=temp_hv_25[3] #Write out the 99% Credible intervals and median for each EYO (i) 
  lines_HV_50[i,1]=temp_hv_50[1];  lines_HV_50[i,2]=temp_hv_50[2];  lines_HV_50[i,3]=temp_hv_50[3] # Carriers
  lines_HV_75[i,1]=temp_hv_75[1];  lines_HV_75[i,2]=temp_hv_75[2];  lines_HV_75[i,3]=temp_hv_75[3] # Carriers
  diff_lines[i,1]=temp_diff[1]; diff_lines[i,2]=temp_diff[2]; diff_lines[i,3]=temp_diff[3]; # Difference line. Where the 0.05% (lower) value is >0 the groups are different
  
}




# Add age term and column names to matrices for plotting
flipped_eyo_hv_quant=t(age.1); flipped_eyo_hv_quant=t(flipped_eyo_hv_quant); combined_lines_csftau_per_rel_wt_mean_eyo=cbind(flipped_eyo_hv_quant,lines_HV_25, lines_HV_50, lines_HV_75); # Combined the matrices for noncarriers and carriers and have the age values
colnames(combined_lines_csftau_per_rel_wt_mean_eyo)<- c("age","lower_lo_cdr", "median_lo_cdr", "upper_lo_cdr", "lower_med_cdr", "median_med_cdr", "upper_med_cdr", "lower_hi_cdr", "median_hi_cdr", "upper_hi_cdr") #combine carriers and noncarriers
combined_lines_csftau_per_rel_wt_mean_eyo=as.data.frame(combined_lines_csftau_per_rel_wt_mean_eyo) #Make it a data frame or none of the plotting works
diff_lines=cbind(flipped_eyo_hv_quant,diff_lines); colnames(diff_lines)<- c("age","lower", "median", "upper") #separate matrix for difference, honest this could be combined with the previous one


##### Plot Images #####
Plot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC, aes(x=eyo_linear, y=z_cent_LUMIPULSE_CSF_ptau)) #scatter plot of the actual data
#Point = geom_point(size=5,aes(color=alpha=.7),show.legend=F) # change the size of the dots and have the shapes map carrier and noncarrier designations
Color= scale_colour_gradient2(low = ("#980043"),mid = "#88419d",high = ("#9ebcda"), midpoint = .6)
#Color=scale_color_manual(values=c('black', '#00798c', '#A50026'), labels=c('NC', 'PSEN1 Cytoplasmic','PSEN1 Transmembrane')) #set colors and labels
Ribbon1=geom_ribbon(data = combined_lines_csftau_per_rel_wt_mean_eyo, inherit.aes = FALSE, aes(x = age, ymin = lower_hi_cdr, ymax = upper_hi_cdr), fill = "#9ebcda", alpha = 0.2) # This plots upper and lower bounds
Ribbon3=geom_ribbon(data = combined_lines_csftau_per_rel_wt_mean_eyo, inherit.aes = FALSE, aes(x = age, ymin = lower_lo_cdr, ymax = upper_lo_cdr), fill = "#980043", alpha = 0.2)
Line_int = geom_line(data=combined_lines_csftau_per_rel_wt_mean_eyo, size=2,aes(x=age.1, y=median_lo_cdr ),colour = "#980043",show.legend=F) # median line for noncarriers. Linetype=2 makes it dashed
Line_NC = geom_line(data=combined_lines_csftau_per_rel_wt_mean_eyo, size=2, aes(x=age.1, y=median_hi_cdr ),show.legend=F,colour = "#9ebcda") # median line for noncarriers. Linetype=2 makes it dashed
Labels=labs(x = "EYO", y= "Standardized CSF pTau") #Add in X and Y axis labels
#Ticks=scale_x_continuous(breaks=seq(-35,25,5), limits=c(-35,22)) #Manually set x-axis tick spacing
cx= scale_x_continuous(breaks = c(-30, -20, -10,0,10,20), labels = c("-30", "-20", "-10","0","10","20"))


Formatting_Scatter = #All of this just controls the physical look of the image
  # theme(axis.title = element_text(family = "Helvetica", color="#666666", face="bold", size=30)) + 
  theme_bw() +
  theme(axis.text = element_text(size = 20), axis.title.x=element_blank()) +
  theme(legend.title=element_blank()) +
  theme(legend.text=element_text(size=15, family="Helvetica")) +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size = 25, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size = 25, face="bold")) +
  theme(legend.position = "blank" )+
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent")) #Make background of legends transparent

plot_csftau_per_rel_wt_25_75_eyo=Plot  +cx +Color+ Line_int  +Line_NC +Ribbon1 + Ribbon3 + Labels +Formatting_Scatter # Add all of the graph elements together
print(plot_csftau_per_rel_wt_25_75_eyo)



###EYO
age_abcomposite_per_rel_wt_mean_lo <- as.data.frame(cbind(combined_lines_cdr_perc_rel_wt_mean_eyo$age, combined_lines_cdr_perc_rel_wt_mean_eyo$median_lo_cdr,combined_lines_csf42_per_rel_wt_mean_eyo$median_lo_cdr, combined_lines_csftau_per_rel_wt_mean_eyo$median_lo_cdr, combined_lines_fdg_per_rel_wt_mean_eyo$median_lo_cdr, combined_lines_hv_per_rel_wt_mean_eyo$median_lo_cdr,combined_lines_pib_per_rel_wt_mean_eyo$median_lo_cdr) )
colnames(age_abcomposite_per_rel_wt_mean_lo)<- c("age", "lo_cdr", "lo_csfab42", "lo_csfttau", "lo_fdg", "lo_hv", "lo_pib")



COLS = c("#1167B1", "#43A047", "#607D8B", "#673AB7", "#ffd276", "#FFE800")
tert_lo <- ggplot(age_abcomposite_per_rel_wt_mean_lo) + 
  geom_hline(yintercept=0, linetype='dotted')+
  geom_line(data=age_abcomposite_per_rel_wt_mean_lo, size=2, aes(x=age, y=lo_hv),show.legend=F,colour = "#0571af") +# median line for carriers
  geom_line(data=age_abcomposite_per_rel_wt_mean_lo, size=2, aes(x=age, y=lo_pib),show.legend=F,colour = "#e08726")+ # median line for carriers
  geom_line(data=age_abcomposite_per_rel_wt_mean_lo, size=2, aes(x=age, y=lo_csfab42),show.legend=F,colour = "#e5c218")+ # median line for carriers
  geom_line(data=age_abcomposite_per_rel_wt_mean_lo, size=2, aes(x=age, y=lo_csfttau),show.legend=F,colour = "#499364")+ # median line for carriers
  geom_line(data=age_abcomposite_per_rel_wt_mean_lo, size=2, aes(x=age, y=lo_fdg),show.legend=F,colour = "#8c7aba") +# median line for carriers
  geom_line(data=age_abcomposite_per_rel_wt_mean_lo, size=2, aes(x=age, y=lo_cdr),show.legend=F,colour = "black") +# median line for carriers
  scale_color_manual(name="Meausure", values = COLS, labels = c("HV", "CSF tTau", "CDR-SB","FDG", "PiB", "CSF ab42") )+
  scale_fill_manual(name="Meausure", values = COLS, labels = c("HV", "CSF tTau", "CDR-SB","FDG", "PiB", "CSF ab42"))+
  labs( x="EYO", y=expression(bold(paste("Standardized Difference"))), title = expression(bold(paste("Gamma-Secretase Composite score 25% relative to wt"))))+
  theme_bw() +
  coord_cartesian(ylim = c(-3, 3), xlim=c(-20,15)) +
  theme(axis.text = element_text(size = 20))+
  theme(legend.title=element_text(size = 15, face="bold")) +
  theme(legend.text=element_text(size= 15, face = "bold",family="Helvetica")) +
  theme(axis.title.y = element_text(size = 25, face="bold")) + 
  theme(axis.title.x = element_text(size = 25, face="bold")) +
  theme(legend.position = "right")

##EYO
age_abcomposite_per_rel_wt_mean_hi <- as.data.frame(cbind(combined_lines_cdr_perc_rel_wt_mean_eyo$age, combined_lines_cdr_perc_rel_wt_mean_eyo$median_hi_cdr,combined_lines_csf42_per_rel_wt_mean_eyo$median_hi_cdr, combined_lines_csftau_per_rel_wt_mean_eyo$median_hi_cdr, combined_lines_fdg_per_rel_wt_mean_eyo$median_hi_cdr, combined_lines_hv_per_rel_wt_mean_eyo$median_hi_cdr,combined_lines_pib_per_rel_wt_mean_eyo$median_hi_cdr) )
colnames(age_abcomposite_per_rel_wt_mean_hi)<- c("age","hi_cdr", "hi_csfab42", "hi_csfttau", "hi_fdg", "hi_hv", "hi_pib")


tert_hi <- ggplot(age_abcomposite_per_rel_wt_mean_hi) + 
  geom_hline(yintercept=0, linetype='dotted')+
  geom_line(data=age_abcomposite_per_rel_wt_mean_hi, size=2, aes(x=age, y=hi_hv),colour = "#0571af") +# median line for carriers
  geom_line(data=age_abcomposite_per_rel_wt_mean_hi, size=2, aes(x=age, y=hi_pib),colour = "#e08726")+ # median line for carriers
  geom_line(data=age_abcomposite_per_rel_wt_mean_hi, size=2, aes(x=age, y=hi_csfab42),colour = "#e5c218")+ # median line for carriers
  geom_line(data=age_abcomposite_per_rel_wt_mean_hi, size=2, aes(x=age, y=hi_csfttau),colour = "#499364")+ # median line for carriers
  geom_line(data=age_abcomposite_per_rel_wt_mean_hi, size=2, aes(x=age, y=hi_fdg),colour = "#8c7aba") +# median line for carriers
  geom_line(data=age_abcomposite_per_rel_wt_mean_hi, size=2, aes(x=age, y=hi_cdr),colour = "black") + # median line for carriers
  scale_color_manual(name="Meausure", values = COLS, labels = c("HV", "CSF tTau", "CDR-SB","FDG", "PiB", "CSF ab42"))+
  scale_fill_manual(name="Meausure", values = COLS, labels = c("HV", "CSF tTau", "CDR-SB","FDG", "PiB", "CSF ab42"))+
  labs( x="EYO", y=expression(bold(paste("Standardized Difference"))), title = expression(bold(paste("Gamma-Secretase Composite score 75% relative to wt"))))+
  theme_bw() +
  coord_cartesian(ylim = c(-3, 3),xlim=c(-20,15)) +
  theme(axis.text = element_text(size = 20))+
  theme(legend.title=element_text(size = 15, face="bold")) +
  theme(legend.text=element_text(size= 15, face = "bold",family="Helvetica")) +
  theme(axis.title.y = element_text(size = 25, face="bold")) + 
  theme(axis.title.x = element_text(size = 25, face="bold")) +
  theme(legend.position = "right")
jackcurves_bl_abtert_per_rel_wt_combined<-plot_grid( tert_lo, tert_hi  ,nrow = 1, label_size=30)


dev.new()
pdf("/Users/sws59/Dropbox (Partners HealthCare)/Stephanie (1)/BC_analyses/Lancet_neuro/Revision_1/jackcurves_bl_abtert_per_rel_wt_combined_lo_hi_only_eyo2.pdf", width=22, height=10)
jackcurves_bl_abtert_per_rel_wt_combined
dev.off()




```

```{r}


stan_tert_zpib <- stan_glm(j_PIB_RSF_CORTMEAN ~ g_gsec_composite_rel_wt*age_linear + g_gsec_composite_rel_wt*age_cubic, 
                          data = dian_matched_bc_data_pib_mri_mmse2_latest_MC, family = gaussian(), 
                          algorithm = "sampling", adapt_delta = 0.99, seed=123456,
                          prior = cauchy(0,2.5), prior_intercept = cauchy(0,2.5),
                          chains = 8, cores = 1,  iter = 10000, thin = 10)
# chains is how many instances are being run in parallel, iterations means how many iterations are done within each chain.
# Thin indicates how many of the interations to retain. In this case I have it set to 1/10. This was their recommended default.
summary(stan_tert_zpib)


#### % rel to wt #### 


parameter_estimates_rate <- rstan::extract(stan_tert_zpib$stanfit) # Extract the model estimates from STAN
factorweights_rate <-cbind(parameter_estimates_rate$alpha, parameter_estimates_rate$beta) # This specifically pulls out the intercept (alpha) and beta (beta) parameters

age.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_linear),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_linear),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments
#age_cubic.1=seq(min(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),max(dian_matched_bc_data_pib_mri_mmse2_latest_MC$age_cubic),by=.1) #create a vector that ranges from 8 (min age) to 75 (max age) in 0.1 increments


hv_25=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1)); #generate blank matrices for outputs for noncarriers
hv_50=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
hv_75=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));
diff_log=matrix(0,nrow = nrow(factorweights_rate), ncol = length(age.1));

contrasts_25_hv=matrix(0,nrow = length(age.1), ncol = 6); contrasts_50_hv=matrix(0,nrow = length(age.1), ncol = 6);  contrasts_75_hv=matrix(0,nrow = length(age.1), ncol = 6) #Making a blank matrix for contrasts. ncol=6 as there are six model parameters as coded

for (j in 1:length(age.1)) 
{ # We take all of the values in the dummy matrix (age.1) and use the fit from our real data to get  the cubic term values for each dummy entry
  
  tempfit=rcspline.eval(age.1[j],attr(splinefit,'knots'),norm=2, pc=FALSE, inclx=TRUE) 
  
  #Contrast 
  contrasts = matrix(
    c(1,1,1,#intercept
      25,50,75, #cytoplasmic group
      tempfit[1,1],tempfit[1,1],tempfit[1,1],#eyo linear
      tempfit[1,2],tempfit[1,2],tempfit[1,2],#eyo cubic
      tempfit[1,1]*25,tempfit[1,1]*50,tempfit[1,1]*75, #eyolinear*CY group
      tempfit[1,2]*25,tempfit[1,2]*50,tempfit[1,2]*75),#eyo cubic*CY group
    
    nrow=3, 
    ncol=6)
  contrasts_25_hv[j,]=contrasts[1,]; contrasts_50_hv[j,]=contrasts[2,]; contrasts_75_hv[j,]=contrasts[3,] #Set up matrix of contrasts for each age point separately for carriers and noncarriers to be used below
}



for (i in 1:nrow(factorweights_rate)) {weights = factorweights_rate[i,] #for every iteration select the weights

for (j in 1:length(age.1)) # for every age value in our output what would the value be using this iterations model fits?
{
  contrasts=rbind(contrasts_25_hv[j,], contrasts_50_hv[j,], contrasts_75_hv[j,]) #select the appropriate contrast from the matrix made above
  
  weights_rate = contrasts %*% weights #Multiple beta weights by the contrast to get one value per group for this age and this iteration
  hv_25[i,j]=weights_rate[1,]; hv_50[i,j]=weights_rate[2,] ; hv_75[i,j]=weights_rate[3,]  #Write the points out into a matrix for each group
  diff_log[i,j]=weights_rate[3,]-weights_rate[1,] #difference is equal to carriers - noncarriers. This is  used to determine the actual ge of deviation 
  
}
} 

lines_HV_25=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_50=matrix(0, ncol=3, nrow=length(age.1)); lines_HV_75=matrix(0, ncol=3, nrow=length(age.1));diff_lines=matrix(0, ncol=3, nrow=length(age.1));#generate blank matrices we will use for lines

for (i in 1:ncol(hv_25)) # For every age point (i)
{ 
  temp_hv_25=quantile(hv_25[,i], probs = c(0.05, .5, .95)); temp_hv_25=unname(temp_hv_25) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_50=quantile(hv_50[,i], probs = c(0.05, .5, .95)); temp_hv_50=unname(temp_hv_50) #get the 0.05, median, and 99.5 percentile values for each age
  temp_hv_75=quantile(hv_75[,i], probs = c(0.05, .5, .95)); temp_hv_75=unname(temp_hv_75) #get the 0.05, median, and 99.5 percentile values for each age
  temp_diff=quantile(diff_log[,i], probs = c(0.05, .5, .95)); temp_diff=unname(temp_diff) #get the 0.05, median, and 99.5 percentile values for each age
  
  lines_HV_25[i,1]=temp_hv_25[1];  lines_HV_25[i,2]=temp_hv_25[2];  lines_HV_25[i,3]=temp_hv_25[3] #Write out the 99% Credible intervals and median for each EYO (i) 
  lines_HV_50[i,1]=temp_hv_50[1];  lines_HV_50[i,2]=temp_hv_50[2];  lines_HV_50[i,3]=temp_hv_50[3] # Carriers
  lines_HV_75[i,1]=temp_hv_75[1];  lines_HV_75[i,2]=temp_hv_75[2];  lines_HV_75[i,3]=temp_hv_75[3] # Carriers
  diff_lines[i,1]=temp_diff[1]; diff_lines[i,2]=temp_diff[2]; diff_lines[i,3]=temp_diff[3]; # Difference line. Where the 0.05% (lower) value is >0 the groups are different
  
}

# Add age term and column names to matrices for plotting
flipped_eyo_hv_quant=t(age.1); flipped_eyo_hv_quant=t(flipped_eyo_hv_quant); combined_lines_pib_per_rel_wt_mean=cbind(flipped_eyo_hv_quant,lines_HV_25, lines_HV_50, lines_HV_75); # Combined the matrices for noncarriers and carriers and have the age values
colnames(combined_lines_pib_per_rel_wt_mean)<- c("age","lower_lo_cdr", "median_lo_cdr", "upper_lo_cdr", "lower_med_cdr", "median_med_cdr", "upper_med_cdr", "lower_hi_cdr", "median_hi_cdr", "upper_hi_cdr") #combine carriers and noncarriers
combined_lines_pib_per_rel_wt_mean=as.data.frame(combined_lines_pib_per_rel_wt_mean) #Make it a data frame or none of the plotting works
diff_lines=cbind(flipped_eyo_hv_quant,diff_lines); colnames(diff_lines)<- c("age","lower", "median", "upper") #separate matrix for difference, honest this could be combined with the previous one

##### Plot Images #####
Plot=ggplot(dian_matched_bc_data_pib_mri_mmse2_latest_MC, aes(x=age_linear, y=z_cent_PIB_RSF_CORTMEAN)) #scatter plot of the actual data
Point = geom_point(size=3,aes(alpha=.1),show.legend=F) # change the size of the dots and have the shapes map carrier and noncarrier designations
Color= scale_colour_gradient2(low = ("#980043"),mid = "#88419d",high = ("#9ebcda"), midpoint = .6)
cx= scale_x_continuous(breaks = c(30, 40, 50,60), labels = c("30","40","50", "60"))
Ribbon1=geom_ribbon(data = combined_lines_pib_per_rel_wt_mean, inherit.aes = FALSE, aes(x = age, ymin = lower_hi_cdr, ymax = upper_hi_cdr), fill = "#9ebcda", alpha = 0.2) # This plots upper and lower bounds
Ribbon2=geom_ribbon(data = combined_lines_pib_per_rel_wt_mean, inherit.aes = FALSE, aes(x = age, ymin = lower_med_cdr, ymax = upper_med_cdr), fill = "#88419d", alpha = 0.2)
Ribbon3=geom_ribbon(data = combined_lines_pib_per_rel_wt_mean, inherit.aes = FALSE, aes(x = age, ymin = lower_lo_cdr, ymax = upper_lo_cdr), fill = "#980043", alpha = 0.2)
Line_int = geom_line(data=combined_lines_pib_per_rel_wt_mean, size=2,aes(x=age.1, y=median_lo_cdr ),colour = "#980043",show.legend=F) # median line for noncarriers. Linetype=2 makes it dashed
Line_lo = geom_line(data=combined_lines_pib_per_rel_wt_mean, size=2, aes(x=age.1, y=median_med_cdr),show.legend=F,colour = "#88419d") # median line for carriers
Line_NC = geom_line(data=combined_lines_pib_per_rel_wt_mean, size=2, aes(x=age.1, y=median_hi_cdr ),show.legend=F,colour = "#9ebcda") # median line for noncarriers. Linetype=2 makes it dashed
Labels=labs(x = "Age", y= "Standardized pib") #Add in X and Y axis labels


Formatting_Scatter = #All of this just controls the physical look of the image
  # theme(axis.title = element_text(family = "Helvetica", color="#666666", face="bold", size=30)) + 
  theme_bw() +
  theme(axis.text = element_text(size = 20), axis.title.x=element_blank()) +
  theme(legend.title=element_blank()) +
  theme(legend.text=element_text(size=15, family="Helvetica")) +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size = 25, face="bold")) + #moves the axis text slightly out to create more space
  theme(axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size = 25, face="bold")) +
  theme(legend.position = "blank" )+
  theme(legend.background = element_rect(fill="transparent"), legend.key = element_rect(fill = "transparent")) #Make background of legends transparent

plot_pib_per_rel_wt_mean=Plot +Color +cx  + Line_int + Line_lo +Line_NC +Ribbon1 +Ribbon2+ Ribbon3 + Labels +Formatting_Scatter # Add all of the graph elements together
print(plot_pib_per_rel_wt_mean)
```


